---
title: "Somalogic Benchmark"
author: "Sophia Soriano"
date: "`r Sys.Date()`"
output: 
  html_document:
    toc: true
    toc_float: true
    toc_depth: 4
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning=FALSE, message=FALSE)
library(dplyr)
library(tidyr)
library(tidyverse)
library(ggplot2)
library(ggpubr)
library(readr)
library(readxl)
library(SomaDataIO)
library(stringr)
library(EnvStats)
library(plotly)
library(htmlwidgets)
library(ggrepel)
library(truncnorm)
library(scales)
library(knitr)
library(naniar)
library(eList)
library(stats)
library(toolbox)
library(arsenal)
```

## Initial Data Exploration

```{r Data_Load}
#Load somalogic adat data file - mouse
mouse_soma_raw_df = read_adat("/Users/sorianos/Code/SomaLogic/OHS-2227720_2022-10-17/SS-2227720_v4.1_other.hybNorm.medNormInt.plateScale.medNormSMP.adat")
mouse_soma_alt_raw_df = read_adat("/Users/sorianos/Code/SomaLogic/OHS-2227720_2022-10-17/SS-2227720_v4.1_other.hybNorm.medNormInt.plateScale.adat")
#Load somalogic adat data file - human
human_soma_raw_df = read_adat("/Users/sorianos/Code/SomaLogic/OHS-2227721_2022-10-18/SS-2227721_v4.1_EDTAPlasma.hybNorm.medNormInt.plateScale.calibration.anmlQC.qcCheck.anmlSMP.adat")
human_soma_alt_raw_df = read_adat("/Users/sorianos/Code/SomaLogic/OHS-2227721_2022-10-18/SS-2227721_v4.1_EDTAPlasma.hybNorm.medNormInt.plateScale.calibration.anmlQC.qcCheck.adat")

#Load project disney tsv annotation file - mouse
mouse_anno_raw_df = read_tsv("/Users/sorianos/Code/SomaLogic/sample_NP_annotation_mouse.tsv")
#Load project disney tsv annotation file - human
human_anno_raw_df = read_csv("/Users/sorianos/Code/SomaLogic/sample_annotation_human.csv")

#Load xlsx mouse sample submission file
mouse_submission_df = read_excel("/Users/sorianos/Code/SomaLogic/SubmissionForms/SS-2227720 OHSU-Flory-EDTA plasma 1 of 2 SSF.xlsx", sheet = "Electronic Submission Form")
#Load xlsx human sample submission file
human_submission_df = read_excel("/Users/sorianos/Code/SomaLogic/SubmissionForms/SS-2227721 OHSU-Flory-Human EDTA plasma 2 of 2 SSF.xlsx", sheet = "Electronic Submission Form")

#Load SomaDataIO aptamer anno table
apt_anno_mouse_df = getAnalyteInfo(mouse_soma_raw_df)
apt_anno_human_df = getAnalyteInfo(human_soma_raw_df)

#Load mouse data with SomaLogic mouse signaling metric annotations for aptamer quality in mouse samples
#Load xlsx mouse sample submission file - needed to reformat excel file to read in to R
mouse_sigmetric_df = read_excel("/Users/sorianos/Code/SomaLogic/SomaLogic7KMousePlasmaSignalingMetrics220926_reformated.xlsx", sheet = "Mouse Metrics")
colnames(mouse_sigmetric_df) = c("SeqId","Target","TargetFullName","UniProtID","Type","Organism","F_Statistic_Mouse_Plasma","Quality")
mouse_sigmetric_df = mouse_sigmetric_df[-1,]
mouse_sigmetric_df = mouse_sigmetric_df %>% mutate(F_Statistic_Mouse_Plasma = round(as.numeric(F_Statistic_Mouse_Plasma),2)) %>%
                                              dplyr::select(c(SeqId, F_Statistic_Mouse_Plasma, Quality))

#Merge sigmetrics with aptamer annotations
combined_apt_anno_mouse_df = left_join(apt_anno_mouse_df, mouse_sigmetric_df, by="SeqId")
combined_apt_anno_human_df = left_join(apt_anno_human_df, mouse_sigmetric_df, by="SeqId")
```

```{r Combine_Data_Anno_Mouse}
#Combine mouse annotation and data df

#Change format of SampleId column in data df to match annotation df column
colnames(mouse_soma_raw_df)[colnames(mouse_soma_raw_df) == "SampleId"] = "Sample_ID"

#Combine proteograph filenames in annotation file into list per sample - not part of SomaLogic analysis
mouse_anno_df = mouse_anno_raw_df %>%
  group_by(SampleName) %>%
  mutate(File_Names = list(File.Name)) %>%
  dplyr::select(-c("File.Name")) %>%
  unique()

#Remove " A" from Sample_ID column in data df
mouse_soma_df = mouse_soma_raw_df %>% mutate(Sample_ID = ifelse(str_detect(Sample_ID, " A$"),
                                                                str_replace(Sample_ID," A",""),
                                                                Sample_ID))

#Left-join df - merge is deprecated for soma_adat objects
combined_mouse_df = left_join(mouse_soma_df, mouse_anno_df, by="Sample_ID")

#Create output annotation file for SomaLogic DataDelve
DD_mouse_anno1_df = mouse_soma_raw_df %>% mutate(Sample_ID_Anno = ifelse(str_detect(Sample_ID, " A$"),
                                                                str_replace(Sample_ID," A",""),
                                                                Sample_ID)) %>%
                                          dplyr::select(c("Sample_ID","Sample_ID_Anno"))
DD_mouse_anno2_df = mouse_anno_df %>% dplyr::select(c(Sample_ID,Group_final,Subgroup))
colnames(DD_mouse_anno2_df)[colnames(DD_mouse_anno2_df) == "Sample_ID"] = "Sample_ID_Anno"
DD_mouse_anno_df = left_join(DD_mouse_anno1_df, DD_mouse_anno2_df, by="Sample_ID_Anno")
rownames(DD_mouse_anno_df) = NULL
write.table(DD_mouse_anno_df,"SomaDD_mouse_annotation.txt", sep = "\t", row.names = FALSE)

#Create list of mouse samples for Matt (excluding controls and pools)
mouse_ID_df = DD_mouse_anno_df %>% filter(!is.na(SampleName)) %>% dplyr::select(-Sample_ID)
write.table(mouse_ID_df, "SomaLogic_Benchmark_MouseIDs.tsv", sep = "\t", row.names = FALSE)

#Reorganizing df to start w/ all annotations, then data columns, also removing any annotation columns irrelevant to analysis or with all NA
select_mouse_df = combined_mouse_df %>% dplyr::select(PlateId:SampleMatrix, SampleName.y:Specimen_group_ID, everything()) %>%
                                          dplyr::select(-c(Type, Note, SampleMatrix, ...16, ...18, AliquotingNotes, TimePoint, SampleGroup, SiteId, SubjectID, PlateId, File_Names, PlateRunDate, ScannerID))

#Long-version of select mouse df
long_select_mouse_df = select_mouse_df %>% pivot_longer(names_to = "AptName",
                                                      values_to = "Intensity",
                                                      cols = starts_with("seq"))

long_combo_mouse_df = left_join(long_select_mouse_df,combined_apt_anno_mouse_df, by="AptName")
```

```{r Combine_Data_Anno_Mouse_Alt}
#Combine mouse annotation and alt data df

#Change format of SampleId column in data df to match annotation df column
colnames(mouse_soma_alt_raw_df)[colnames(mouse_soma_alt_raw_df) == "SampleId"] = "Sample_ID"

#Remove " A" from Sample_ID column in data df
mouse_soma_alt_df = mouse_soma_alt_raw_df %>% mutate(Sample_ID = ifelse(str_detect(Sample_ID, " A$"),
                                                                str_replace(Sample_ID," A",""),
                                                                Sample_ID))

#Left-join df - merge is deprecated for soma_adat objects
combined_mouse_alt_df = left_join(mouse_soma_alt_df, mouse_anno_df, by="Sample_ID")

#Reorganizing df to start w/ all annotations, then data columns, also removing any annotation columns irrelevant to analysis or with all NA
select_mouse_alt_df = combined_mouse_alt_df %>% dplyr::select(PlateId:SampleMatrix, SampleName.y:Specimen_group_ID, everything()) %>%
                                          dplyr::select(-c(Type, Note, SampleMatrix, ...16, ...18, AliquotingNotes, TimePoint, SampleGroup, SiteId, SubjectID, PlateId, File_Names, PlateRunDate, ScannerID))

#Long-version of select mouse df
long_select_mouse_alt_df = select_mouse_alt_df %>% pivot_longer(names_to = "AptName",
                                                      values_to = "Intensity",
                                                      cols = starts_with("seq"))

long_combo_mouse_alt_df = left_join(long_select_mouse_alt_df,combined_apt_anno_mouse_df, by="AptName")
```

```{r Combine_Data_Anno_Human}
#Combine human annotation and data df

#Change format of SampleId column in data df to match annotation df column
colnames(human_soma_raw_df)[colnames(human_soma_raw_df) == "SampleId"] = "Sample_ID"

#Change format of SampleId column in data df to match annotation df column
colnames(human_anno_raw_df)[colnames(human_anno_raw_df) == "Subject.ID"] = "Sample_ID"

#No proteograph specific columns found in the human anno df

#Remove " A" from Sample_ID column in data df
human_soma_df = human_soma_raw_df %>% mutate(Sample_ID = ifelse(str_detect(Sample_ID, " A$"),
                                                                str_replace(Sample_ID," A",""),
                                                                Sample_ID))

#Left-join df - merge is deprecated for soma_adat objects
combined_human_df = left_join(human_soma_df, human_anno_raw_df, by="Sample_ID")

#Create output annotation file for SomaLogic DataDelve
DD_human_anno1_df = human_soma_raw_df %>% mutate(Sample_ID_Anno = ifelse(str_detect(Sample_ID, " A$"),
                                                                str_replace(Sample_ID," A",""),
                                                                Sample_ID)) %>%
                                          dplyr::select(c("Sample_ID","Sample_ID_Anno"))
DD_human_anno2_df = human_anno_raw_df %>% dplyr::select(c(Sample_ID,Clinical.Diagnosis,Screening.Outcome))
colnames(DD_human_anno2_df)[colnames(DD_human_anno2_df) == "Sample_ID"] = "Sample_ID_Anno"
DD_human_anno_df = left_join(DD_human_anno1_df, DD_human_anno2_df, by="Sample_ID_Anno")
rownames(DD_human_anno_df) = NULL
write.table(DD_human_anno_df,"SomaDD_human_annotation.txt", sep = "\t", row.names = FALSE)

#Reorganizing df to start w/ all annotations, then data columns, also removing any annotation columns irrelevant to analysis or with all NA
select_human_df = combined_human_df %>% dplyr::select(PlateId:SampleMatrix, ...1:Sample.Num, everything()) %>%
                                          dplyr::select(-c(PlateId,PlateRunDate,ScannerID,AliquotingNotes,TimePoint,SampleGroup,SiteId,SubjectID))

#Long-version of select human df
long_select_human_df = select_human_df %>% pivot_longer(names_to = "AptName",
                                                      values_to = "Intensity",
                                                      cols = starts_with("seq"))

long_combo_human_df = left_join(long_select_human_df,combined_apt_anno_human_df, by="AptName")
```

```{r Combine_Data_Anno_Human_Alt}
#Combine human annotation and alt data df

#Change format of SampleId column in data df to match annotation df column
colnames(human_soma_alt_raw_df)[colnames(human_soma_alt_raw_df) == "SampleId"] = "Sample_ID"

#No proteograph specific columns found in the human anno df

#Remove " A" from Sample_ID column in data df
human_soma_alt_df = human_soma_alt_raw_df %>% mutate(Sample_ID = ifelse(str_detect(Sample_ID, " A$"),
                                                                str_replace(Sample_ID," A",""),
                                                                Sample_ID))

#Left-join df - merge is deprecated for soma_adat objects
combined_human_alt_df = left_join(human_soma_alt_df, human_anno_raw_df, by="Sample_ID")

#Reorganizing df to start w/ all annotations, then data columns, also removing any annotation columns irrelevant to analysis or with all NA
select_human_alt_df = combined_human_alt_df %>% dplyr::select(PlateId:SampleMatrix, ...1:Sample.Num, everything()) %>%
                                          dplyr::select(-c(PlateId,PlateRunDate,ScannerID,AliquotingNotes,TimePoint,SampleGroup,SiteId,SubjectID))

#Long-version of select human df
long_select_human_alt_df = select_human_alt_df %>% pivot_longer(names_to = "AptName",
                                                      values_to = "Intensity",
                                                      cols = starts_with("seq"))

long_combo_human_alt_df = left_join(long_select_human_alt_df,combined_apt_anno_human_df, by="AptName")
```

## SomaLogic Data QC

### Aptamer Descriptions

**Table 1: List of aptamers grouped by organism and type categories, with number of aptamers per category shown in right-most column.**

```{r Aptamer_Anno_Tables}
#Create aptamer tables to assess types of aptamers in panel
aptamer_table = apt_anno_mouse_df %>% dplyr::select(c(AptName,TargetFullName,Target,Organism,Type)) %>%
                                      dplyr::group_by(Organism, Type) %>%
                                      summarise(Aptamer_Count = n()) %>%
                                      arrange(desc(Aptamer_Count))

# aptamer_mouse_table = long_combo_mouse_df %>% filter(Sample_ID == "AB_220329_10_PL") %>%
#                                                     dplyr::select(c(AptName,TargetFullName,Target,Organism,Type)) %>%
#                                                     dplyr::group_by(Organism, Type) %>%
#                                                     summarise(Aptamer_Count = n()) %>%
#                                                     arrange(desc(Aptamer_Count))
# 
# aptamer_human_table = long_combo_human_df %>% filter(Sample_ID == "RDY0016_31") %>%
#                                                     dplyr::select(c(AptName,TargetFullName,Target,Organism,Type)) %>%
#                                                     dplyr::group_by(Organism, Type) %>%
#                                                     summarise(Aptamer_Count = n()) %>%
#                                                     arrange(desc(Aptamer_Count))

#Count number of high/med/low quality human protien aptamers to detect proteins in mouse samples
qualitycount_mouse_df = combined_apt_anno_mouse_df %>% filter(Organism == "Human" & Type == "Protein") %>%
                                                        dplyr::group_by(Quality) %>%
                                                        summarise(Count = n())

knitr::kable(aptamer_table)
```

**Table 2: Number of human protein aptamers (7289, see Table 1) per quality category. Quality refers to the success with which the human protein-designed aptamer is able to quantify protein intensity in mouse plasma samples.**

```{r}
knitr::kable(qualitycount_mouse_df)
```

```{r Apt_Calc_eLoD_Mouse}
#Use buffers to calculate eLoD per aptamer - mouse (eLoD = M_blank - 4.9MAD_blank)

#Calculate buffer medians
buffer_mouse_df = long_combo_mouse_df %>% filter(SampleType == "Buffer") %>%
                                          dplyr::group_by(Sample_ID,SampleType,Target,TargetFullName,AptName) %>%
                                          summarize(Median_Intensity = median(Intensity),
                                                    MAD_Intensity = mad(Intensity),
                                                    eLoD = (Median_Intensity + 4.9*MAD_Intensity)) %>%
                                          ungroup()

#Filter out all columns except AptName (for join) and eLoD
eLoD_mouse_df = buffer_mouse_df %>% dplyr::select(c(AptName,eLoD))

#Merge eLoDs to main data by aptamer name
eLoD_combo_mouse_df = merge(long_combo_mouse_df, eLoD_mouse_df, by = "AptName")
```

```{r Apt_Calc_eLoD_Mouse_Alt}
#Use buffers to calculate eLoD per aptamer - mouse alt (eLoD = M_blank - 4.9MAD_blank)

#Merge eLoDs to main data by aptamer name
eLoD_combo_mouse_alt_df = merge(long_combo_mouse_alt_df, eLoD_mouse_df, by = "AptName")
```

```{r Apt_Calc_eLoD_Human}
#Use buffers to calculate eLoD per aptamer - human (eLoD = M_blank - 4.9MAD_blank)

#Calculate buffer medians
buffer_human_df = long_combo_human_df %>% filter(SampleType == "Buffer") %>%
                                          dplyr::group_by(Sample_ID,SampleType,Target,TargetFullName,AptName) %>%
                                          summarize(Median_Intensity = median(Intensity),
                                                    MAD_Intensity = mad(Intensity),
                                                    eLoD = (Median_Intensity + 4.9*MAD_Intensity)) %>%
                                          ungroup()

#Filter out all columns except AptName (for join) and eLoD
eLoD_human_df = buffer_human_df %>% dplyr::select(c(AptName,eLoD))

#Merge eLoDs to main data by aptamer name
eLoD_combo_human_df = merge(long_combo_human_df, eLoD_human_df, by = "AptName")
```

```{r Apt_Calc_eLoD_Human_Alt}
#Use buffers to calculate eLoD per aptamer - human alt (eLoD = M_blank - 4.9MAD_blank)

#Merge eLoDs to main data by aptamer name
eLoD_combo_human_alt_df = merge(long_combo_human_alt_df, eLoD_human_df, by = "AptName")
```

```{r eLoD_Filter_Mouse}
#eLoD filtering - mouse
filt_mouse_df = eLoD_combo_mouse_df %>% mutate(eLoD_Flag = ifelse(Intensity>=eLoD,
                                                                  TRUE,
                                                                  FALSE))
#Capture sample aptamers that do not exceed eLoD per sample
rem_mouse_df = filt_mouse_df %>% filter(eLoD_Flag == FALSE) %>%
                                  filter(SampleType == "Sample")
```

```{r eLoD_Filter_Mouse_Alt}
#eLoD filtering - mouse
filt_mouse_alt_df = eLoD_combo_mouse_alt_df %>% mutate(eLoD_Flag = ifelse(Intensity>=eLoD,
                                                                  TRUE,
                                                                  FALSE))

#Capture sample aptamers that do not exceed eLoD per sample
rem_mouse_alt_df = filt_mouse_alt_df %>% filter(eLoD_Flag == FALSE) %>%
                                  filter(SampleType == "Sample")
```

```{r eLoD_Filter_Human}
#eLoD filtering - mouse
filt_human_df = eLoD_combo_human_df %>% mutate(eLoD_Flag = ifelse(Intensity>=eLoD,
                                                                  TRUE,
                                                                  FALSE))

#Capture sample aptamers that do not exceed eLoD per sample
rem_human_df = filt_human_df %>% filter(eLoD_Flag == FALSE) %>%
                                  filter(SampleType == "Sample")
```

```{r eLoD_Filter_Human_Alt}
#eLoD filtering - mouse
filt_human_alt_df = eLoD_combo_human_alt_df %>% mutate(eLoD_Flag = ifelse(Intensity>=eLoD,
                                                                  TRUE,
                                                                  FALSE))

#Capture sample aptamers that do not exceed eLoD per sample
rem_human_alt_df = filt_human_alt_df %>% filter(eLoD_Flag == FALSE) %>%
                                  filter(SampleType == "Sample")
```

### Proteomic Depth of Coverage

```{r eLoD_Filter_AptCount}
#Number of aptamers/proteins detected per sample - mouse
aptcount_filt_mouse_df = filt_mouse_df %>% filter(SampleType == "Sample") %>%
                                            filter(eLoD_Flag==TRUE) %>%
                                            dplyr::group_by(Sample_ID) %>%
                                            mutate(numSigApt = n()) %>%
                                      dplyr::select(c(AptName,Sample_ID,SampleType,Group_final,Intensity,TargetFullName,Target,UniProt,Organism,Type,Quality,eLoD,eLoD_Flag,numSigApt)) %>%
                                            mutate(SampleType = ifelse(startsWith(Sample_ID,"Mouse 35"), "Pooled 35uL",
                                                                       ifelse(endsWith(Sample_ID,"hem"), "Hemolysis",
                                                                              ifelse(startsWith(Sample_ID,"Mouse A"), "Pooled 55uL",
                                                                                     ifelse(startsWith(Sample_ID, "Mouse B"), "Pooled Diluted 55uL",
                                                                                            SampleType))))) %>%
                                            mutate(SampleType = factor(SampleType, levels = c("Hemolysis", "Sample", "Pooled 35uL","Pooled 55uL","Pooled Diluted 55uL"))) %>%
                                            ungroup() %>%
                                            dplyr::arrange(SampleType) %>%
                                            mutate(row_ID = row_number())

#Number of aptamers/proteins detected per sample - human
aptcount_filt_human_df = filt_human_df %>% filter(SampleType == "Sample") %>%
                                            filter(eLoD_Flag==TRUE) %>%
                                            dplyr::group_by(Sample_ID) %>%
                                            mutate(numSigApt = n()) %>%
                                      dplyr::select(c(AptName,Sample_ID,SampleType,Clinical.Diagnosis,Screening.Outcome,Intensity,TargetFullName,Target,UniProt,Organism,Type,Quality,eLoD,eLoD_Flag,numSigApt)) %>%
                                            mutate(SampleType = ifelse(startsWith(Sample_ID,"Human 35"), "Pooled 35uL",
                                                                        ifelse(startsWith(Sample_ID,"Human A"), "Pooled 55uL",
                                                                               ifelse(startsWith(Sample_ID, "Human B"), "Pooled Diluted 55uL",
                                                                                      SampleType)))) %>%
                                            mutate(SampleType = factor(SampleType, levels = c("Hemolysis", "Sample", "Pooled 35uL","Pooled 55uL","Pooled Diluted 55uL"))) %>%
                                            ungroup() %>%
                                            dplyr::arrange(SampleType) %>%
                                            mutate(row_ID = row_number())

#Coverage graph - filtered mouse
aptcov_filtmouse_plot = aptcount_filt_mouse_df %>% 
  ggplot(aes(x = reorder(Sample_ID, row_ID), y = numSigApt, colour = SampleType)) +
  geom_point(size = 1) +
  labs(title = "Mouse - Normalized",
       x = "Sample ID",
       y = "Count") +
  theme(legend.position = "none") +
  theme_bw() +
  ylim(6350,8000) +
  geom_hline(yintercept=length(eLoD_mouse_df$AptName), colour = "blue") + #Max number of aptamers = 7596
  geom_hline(yintercept=6379, colour = "red") + #Number of high/medium human aptamers in mouse (expected minimum apt count per mouse sample) = 6379
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5)) +
  theme(axis.text.x = element_blank()) +
  theme(axis.text.y = element_text(size = 12)) +
  theme(axis.title = element_text(vjust = 0.5, size = 14)) +
  theme(title = element_text(size = 14)) +
  theme(legend.title = element_text(size = 14)) +
  theme(legend.text = element_text(size = 12)) +
  scale_colour_manual(values=c("#F8766D", "#A3A500", "#00BF7D", "#00B0F6", "#E76BF3"), drop = FALSE)
#ggsave("aptcov_filtmouse_plot.png")

#Coverage graph - filtered human
aptcov_filthuman_plot = aptcount_filt_human_df %>%
  ggplot(aes(x = reorder(Sample_ID, row_ID), y = numSigApt,color = SampleType)) +
  geom_point(size = 1) +
  labs(title = "Human - Normalized",
       x = "Sample ID",
       y = "Count") +
  theme(legend.position = "none") +
  theme_bw() +
  ylim(6350,8000) +
  geom_hline(yintercept=length(eLoD_human_df$AptName), colour = "blue") + #Max number of aptamers = 7596
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5)) +
  theme(axis.text.x = element_blank()) +
  theme(axis.text.y = element_text(size = 12)) +
  theme(axis.title = element_text(vjust = 0.5, size = 14)) +
  theme(title = element_text(size = 14)) +
  theme(legend.title = element_text(size = 14)) +
  theme(legend.text = element_text(size = 12)) +
  scale_colour_manual(values=c("#F8766D", "#A3A500", "#00BF7D", "#00B0F6", "#E76BF3"), drop = FALSE)
#ggsave("aptcov_filthuman_plot.png")

```

```{r Plot_eLoD_Filter_AptCount}
#Plot filtered aptamer counts per sample
fig_eLoDFilt_AptCount = ggarrange(aptcov_filtmouse_plot, aptcov_filthuman_plot, labels="AUTO", ncol = 1, nrow = 2, common.legend = TRUE, legend = "right")
annotate_figure(fig_eLoDFilt_AptCount, top = text_grob("Aptamer Count per Sample", size = 18))
ggsave("fig_eLoDFilt_AptCount.png", width = 7.29, height = 6.5, units = "in")
```

\newline

**Figure 1: Plots of aptamer counts per sample for A) mouse and B) human datasets above eLoD calculated for each aptamer (eLoD = Median_blank + 4.9MAD_blank). Blue horizontal line indicate the total number of aptamers in the panel (7596). Red horizontal line (mouse only) indicates the number of human protein aptamers assessed to perform with high/medium quality in mouse samples.**

```{r eLoD_Filter_Alt_AptCount}
#Number of aptamers/proteins detected per sample - mouse alt
aptcount_filt_mouse_alt_df = filt_mouse_alt_df %>% filter(SampleType == "Sample") %>%
                                            filter(eLoD_Flag==TRUE) %>%
                                            dplyr::group_by(Sample_ID) %>%
                                            mutate(numSigApt = n()) %>%
                                            dplyr::select(c(AptName,Sample_ID,SampleType,Group_final,Intensity,TargetFullName,Target,UniProt,Organism,Type,Quality,eLoD,eLoD_Flag,numSigApt)) %>%
                                            mutate(SampleType = ifelse(startsWith(Sample_ID,"Mouse 35"), "Pooled 35uL",
                                                                       ifelse(endsWith(Sample_ID,"hem"), "Hemolysis",
                                                                              ifelse(startsWith(Sample_ID,"Mouse A"), "Pooled 55uL",
                                                                                     ifelse(startsWith(Sample_ID, "Mouse B"), "Pooled Diluted 55uL",
                                                                                            SampleType))))) %>%
                                            mutate(SampleType = factor(SampleType, levels = c("Hemolysis", "Sample", "Pooled 35uL","Pooled 55uL","Pooled Diluted 55uL"))) %>%
                                            ungroup() %>%
                                            dplyr::arrange(SampleType) %>%
                                            mutate(row_ID = row_number())

#Number of aptamers/proteins detected per sample - human alt
aptcount_filt_human_alt_df = filt_human_alt_df %>% filter(SampleType == "Sample") %>%
                                            filter(eLoD_Flag==TRUE) %>%
                                            dplyr::group_by(Sample_ID) %>%
                                            mutate(numSigApt = n()) %>%
                                            dplyr::select(c(AptName,Sample_ID,SampleType,Clinical.Diagnosis,Screening.Outcome,Intensity,TargetFullName,Target,UniProt,Organism,Type,Quality,eLoD,eLoD_Flag,numSigApt)) %>%
                                            mutate(SampleType = ifelse(startsWith(Sample_ID,"Human 35"), "Pooled 35uL",
                                                                        ifelse(startsWith(Sample_ID,"Human A"), "Pooled 55uL",
                                                                               ifelse(startsWith(Sample_ID, "Human B"), "Pooled Diluted 55uL",
                                                                                      SampleType)))) %>%
                                            mutate(SampleType = factor(SampleType, levels = c("Hemolysis", "Sample", "Pooled 35uL","Pooled 55uL","Pooled Diluted 55uL"))) %>%
                                            ungroup() %>%
                                            dplyr::arrange(SampleType) %>%
                                            mutate(row_ID = row_number())

#Coverage graph - filtered mouse alt
aptcov_filtmouse_alt_plot = aptcount_filt_mouse_alt_df %>% 
  ggplot(aes(x = reorder(Sample_ID, row_ID), y = numSigApt, colour = SampleType)) +
  geom_point(size = 1) +
  labs(title = "Mouse (Pre-Normalization)",
       x = "",
       y = "Count") +
  theme(legend.position = "none") +
  theme_bw() +
  ylim(6350,8000) +
  geom_hline(yintercept=length(eLoD_mouse_df$AptName), colour = "blue") + #Max number of aptamers = 7596
  geom_hline(yintercept=6379, colour = "red") + #Number of high/medium human aptamers in mouse (expected minimum apt count per mouse sample) = 6379
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5)) +
  theme(axis.text.x = element_blank()) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(legend.title = element_text(size = 12)) +
  theme(title = element_text(size = 14)) +
  scale_colour_manual(values=c("#F8766D", "#A3A500", "#00BF7D", "#00B0F6", "#E76BF3"),
                      drop = FALSE)
#ggsave("aptcov_filtmouse_alt_plot.png")

#Coverage graph - filtered human alt
aptcov_filthuman_alt_plot = aptcount_filt_human_alt_df %>%
  ggplot(aes(x = reorder(Sample_ID,row_ID), y = numSigApt,color = SampleType)) +
  geom_point(size = 1) +
  labs(title = "Human (Pre-Normalization)",
       x = "Sample_ID",
       y = "Count") +
  theme(legend.position = "none") +
  theme_bw() +
  ylim(6350,8000) +
  geom_hline(yintercept=length(eLoD_human_df$AptName), colour = "blue") + #Max number of aptamers = 7596
  
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5)) +
  theme(axis.text.x = element_blank()) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(legend.title = element_text(size = 12)) +
  theme(title = element_text(size = 14)) +
  scale_colour_manual(values=c("#F8766D", "#A3A500", "#00BF7D", "#00B0F6", "#E76BF3"),
                      drop = FALSE)
# ggsave("aptcov_filthuman_alt_plot.png")
```

```{r Plot_eLoD_Filter_Alt_AptCount}
#Plot filtered aptamer counts per sample
fig_eLoDFilt_alt_AptCount = ggarrange(aptcov_filtmouse_alt_plot, aptcov_filthuman_alt_plot, labels="AUTO", ncol = 1, nrow = 2, common.legend = TRUE, legend = "right")
annotate_figure(fig_eLoDFilt_alt_AptCount, top = text_grob("Aptamer Count per Sample", size = 14))
ggsave("fig_eLoDFilt_alt_AptCount.png")
```

\newline

**Figure 2: Plots of aptamer counts per sample for A) mouse and B) human pre-normalization datasets above eLoD calculated for each aptamer (eLoD = Median_blank + 4.9MAD_blank). Blue horizontal line indicate the total number of aptamers in the panel (7596). Red horizontal line (mouse only) indicates the number of human protein aptamers assessed to perform with high/medium quality in mouse samples.**

### Assess Assay Variablity

```{r Int_Dist_Mouse}
#MOUSE

#Distribution of intensity values for each sample
sample_var_mouse_df = filt_mouse_df %>% filter(eLoD_Flag == TRUE) %>%
                                        filter(SampleType != "Buffer") %>%
                                        dplyr::select(c(Sample_ID,SampleType,Group_final,Intensity)) %>%
                                        mutate(Group_final = ifelse(is.na(Group_final),
                                                                    ifelse(SampleType == "Buffer",
                                                                           "Buffer",
                                                                           ifelse(SampleType == "Calibrator",
                                                                                  "Calibrator",
                                                                                  ifelse(startsWith(Sample_ID,"Mouse 35"),
                                                                                         "Pooled 35uL",
                                                                                         ifelse(endsWith(Sample_ID,"hem"),
                                                                                                "Hemolysis",
                                                                                                ifelse(startsWith(Sample_ID, "Mouse A"),
                                                                                                        "Pooled 55uL",
                                                                                                        "Pooled Diluted 55uL"))))),
                                                                    Group_final))
```

```{r Plot_Int_Dist_Mouse}
#Boxplot of intensity distributions per sample - mouse
sample_int_dist_mouse_plot = sample_var_mouse_df %>%
  mutate(`Mouse Condition Group` = Group_final) %>%
  ggplot(aes(x = Sample_ID, y = Intensity, fill = `Mouse Condition Group`)) +
  geom_boxplot(outlier.shape = NA) +
  labs(title = "Mouse - Normalized",
       x = "Samples") +
  theme_bw() +
  ylim(0,6000) +
  theme(axis.text.x = element_blank()) +
  theme(axis.title.y = element_blank()) +
  theme(axis.title = element_text(vjust = 0.5, size = 10)) +
  theme(title = element_text(size = 12)) +
  theme(legend.title = element_text(size = 10)) +
  theme(legend.text = element_text(size = 8)) 
ggsave("sample_int_dist_mouse_plot.png")
sample_int_dist_mouse_plot
```

```{r Int_Dist_Mouse_Alt}
#MOUSE

#Distribution of intensity values for each sample
sample_var_mouse_alt_df = filt_mouse_alt_df %>% filter(eLoD_Flag == TRUE) %>%
                                        filter(SampleType != "Buffer") %>%
                                        dplyr::select(c(Sample_ID,SampleType,Group_final,Intensity)) %>%
                                        mutate(Group_final = ifelse(is.na(Group_final),
                                                                    ifelse(SampleType == "Buffer",
                                                                           "Buffer",
                                                                           ifelse(SampleType == "Calibrator",
                                                                                  "Calibrator",
                                                                                  ifelse(startsWith(Sample_ID,"Mouse 35"),
                                                                                         "Pooled 35uL",
                                                                                         ifelse(endsWith(Sample_ID,"hem"),
                                                                                                "Hemolysis",
                                                                                                ifelse(startsWith(Sample_ID, "Mouse A"),
                                                                                                        "Pooled 55uL",
                                                                                                        "Pooled Diluted 55uL"))))),
                                                                    Group_final))
```

```{r Plot_Int_Dist_Mouse_Alt}
#Boxplot of intensity distributions per sample - mouse
sample_int_dist_mouse_alt_plot = sample_var_mouse_alt_df %>%
  mutate(`Mouse Condition Group` = Group_final) %>%
  ggplot(aes(x = Sample_ID, y = Intensity, fill = `Mouse Condition Group`)) +
  geom_boxplot(outlier.shape = NA) +
  labs(title = "Mouse - Pre-Normalization",
       x = "Samples") +
  theme_bw() +
  #stat_mean_sd_text(size = 2, y.pos = 6000) +
  ylim(0,6000) +
  theme(axis.text.x = element_blank()) +
  theme(axis.title = element_text(vjust = 0.5, size = 10)) +
  theme(title = element_text(size = 12)) +
  theme(legend.title = element_text(size = 10)) +
  theme(legend.text = element_text(size = 8)) 
ggsave("sample_int_dist_mouse_alt_plot.png")
sample_int_dist_mouse_alt_plot
```

```{r Int_Dist_Human}
#HUMAN

#Distribution of intensity values for each sample
sample_var_human_df = filt_human_df %>% filter(eLoD_Flag == TRUE) %>%
                                        filter(SampleType != "Buffer") %>%
                                        mutate(Clinical.Diagnosis = ifelse(is.na(Clinical.Diagnosis),
                                                                    ifelse(SampleType == "Calibrator",
                                                                           "Calibrator",
                                                                           ifelse(SampleType == "Buffer",
                                                                                  "Buffer",
                                                                                  ifelse(SampleType == "QC",
                                                                                         "QC",
                                                                                         ifelse(startsWith(Sample_ID,"Human 35"),
                                                                                                "Pooled 35uL",
                                                                                                ifelse(startsWith(Sample_ID,"Human A"),
                                                                                                        "Pooled 55uL",
                                                                                                        "Pooled Diluted 55uL"))))),
                                                                    Clinical.Diagnosis)) %>%
                                        mutate(Clinical.Diagnosis = factor(Clinical.Diagnosis,levels=c("Calibrator","QC","Benign","Benign prostatic hyperplasia","Prostatic intraepithelial neoplasia","Adenocarcinoma",
                                                               "Pooled 35uL","Pooled 55uL","Pooled Diluted 55uL"))) %>%
                                        dplyr::arrange(Clinical.Diagnosis) %>%
                                        mutate(row_ID = row_number())
```

```{r Plot_Int_Dist_Human}
#Boxplot of intensity distributions per sample - human
sample_int_dist_human_plot = sample_var_human_df %>%
  mutate(`Human Condition Group` = Clinical.Diagnosis) %>%
  ggplot(aes(x = reorder(Sample_ID,row_ID), y = Intensity, fill = `Human Condition Group`)) +
  geom_boxplot(outlier.shape = NA) +
  labs(title = "Human - Normalized",
       x = "Samples") +
  theme_bw() +
  #stat_mean_sd_text(size = 2, y.pos = 6000) +
  ylim(0,6000) +
  #geom_hline(yintercept=1000, colour="red", linetype = "dashed") +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 6))
  theme(axis.text.x = element_blank()) +
  theme(axis.title.y = element_blank()) +
  theme(axis.title = element_text(vjust = 0.5, size = 10)) +
  theme(title = element_text(size = 12)) +
  theme(legend.title = element_text(size = 10)) +
  theme(legend.text = element_text(size = 8)) 
ggsave("sample_int_dist_human_plot.png")
sample_int_dist_human_plot
```

```{r Int_Dist_Human_Alt}
#HUMAN

#Distribution of intensity values for each sample
sample_var_human_alt_df = filt_human_alt_df %>% filter(eLoD_Flag == TRUE) %>%
                                        filter(SampleType != "Buffer") %>%
                                        mutate(Clinical.Diagnosis = ifelse(is.na(Clinical.Diagnosis),
                                                                    ifelse(SampleType == "Calibrator",
                                                                           "Calibrator",
                                                                           ifelse(SampleType == "Buffer",
                                                                                  "Buffer",
                                                                                  ifelse(SampleType == "QC",
                                                                                         "QC",
                                                                                         ifelse(startsWith(Sample_ID,"Human 35"),
                                                                                                "Pooled 35uL",
                                                                                                ifelse(startsWith(Sample_ID,"Human A"),
                                                                                                        "Pooled 55uL",
                                                                                                        "Pooled Diluted 55uL"))))),
                                                                    Clinical.Diagnosis)) %>%
                                        mutate(Clinical.Diagnosis = factor(Clinical.Diagnosis,levels=c("Calibrator","QC","Benign","Benign prostatic hyperplasia","Prostatic intraepithelial neoplasia","Adenocarcinoma",
                                                               "Pooled 35uL","Pooled 55uL","Pooled Diluted 55uL"))) %>%
                                        dplyr::arrange(Clinical.Diagnosis) %>%
                                        mutate(row_ID = row_number())
```

```{r Plot_Int_Dist_Human_Alt}
#Boxplot of intensity distributions per sample - human
sample_int_dist_human_alt_plot = sample_var_human_alt_df %>%
  mutate(`Human Condition Group` = Clinical.Diagnosis) %>%
  ggplot(aes(x = reorder(Sample_ID,row_ID), y = Intensity, fill = `Human Condition Group`)) +
  geom_boxplot(outlier.shape = NA) +
  labs(title = "Human - Pre-Normalization",
       x = "Samples") +
  theme_bw() +
  ylim(0,6000) +
  theme(axis.text.x = element_blank()) +
  theme(axis.title = element_text(vjust = 0.5, size = 10)) +
  theme(title = element_text(size = 12)) +
  theme(legend.title = element_text(size = 10)) +
  theme(legend.text = element_text(size = 8)) 
ggsave("sample_int_dist_human_alt_plot.png")
sample_int_dist_human_alt_plot
```

\newline

**Figure 3: Boxplots showing distribution of aptamer intensities by sample for mouse and human datasets, final data and pre-normalization data for each. Plots are color coded by clinical group for both datasets. Most samples in each dataset have similar intensity distributions. Calibrator samples and hemolysis samples (mouse only) show slightly wider distribution ranges. The sole AK mouse sample has a slightly lower-shifted distribution compared to other samples.**

```{r Combo_Mouse_IntDist_Plots, include = FALSE}
#Plot combined mouse intensity distribution plots - normalized and pre-normalization
fig_ComboMouse_IntDist = ggarrange(sample_int_dist_mouse_alt_plot, sample_int_dist_mouse_plot, labels=c("E","F"), ncol = 2, nrow = 1, common.legend = TRUE, legend = "right")
annotate_figure(fig_ComboMouse_IntDist, top = text_grob("Intensity Distribution per Sample", size = 16))
ggsave("fig_ComboMouse_IntDist.png", width = 14, height = 4.51, units = "in")
```
```{r Combo_Human_IntDist_Plots}
#Plot combined mouse intensity distribution plots - normalized and pre-normalization
fig_ComboHuman_IntDist = ggarrange(sample_int_dist_human_alt_plot, sample_int_dist_human_plot, labels=c("G","H"), ncol = 2, nrow = 1, common.legend = TRUE, legend = "right")
#annotate_figure(fig_ComboHuman_IntDist, top = text_grob("Intensity Distribution per Sample", size = 20))
ggsave("fig_ComboHuman_IntDist.png", width = 14, height = 4.51, units = "in")
```

```{r Plot_Removed_Apt_Mouse}
#Plot removed aptamer count - mouse
sample_rem_mouse_df = rem_mouse_df %>% dplyr::group_by(Group_final,Sample_ID) %>% summarise(Apt_Count = n())

sample_rem_mouse_plot = sample_rem_mouse_df %>% 
  ggplot(aes(x = Sample_ID, y = Apt_Count, fill = Group_final)) +
  geom_bar(stat = "identity") +
  labs(title = "Normalized",
       subtitle = "Filter: Per Aptamer eLoD",
       x = "",
       y = "Aptamer Count") +
  theme_bw() +
  ylim(0,250) +
  rremove("xlab") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5))
#ggsave("sample_rem_mouse_plot.png")
```

```{r Plot_Removed_Apt_Mouse_Alt}
#Plot removed aptamer count - mouse alt
sample_rem_mouse_alt_df = rem_mouse_alt_df %>% dplyr::group_by(Group_final,Sample_ID) %>% summarise(Apt_Count = n())

sample_rem_mouse_alt_plot = sample_rem_mouse_alt_df %>% 
  ggplot(aes(x = Sample_ID, y = Apt_Count, fill = Group_final)) +
  geom_bar(stat = "identity") +
  labs(title = "Pre-Normalization",
       subtitle = "Filter: Per Aptamer eLoD",
       x = "",
       y = "Aptamer Count") +
  theme_bw() +
  ylim(0,250) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5))
#ggsave("sample_rem_mouse_alt_plot.png")
```

```{r Plot_ComboRemApt_Mouse}
fig_Rem_Apt_Mouse = ggarrange(sample_rem_mouse_plot, sample_rem_mouse_alt_plot, labels="AUTO", ncol = 1, nrow = 2, common.legend = TRUE, legend = "right", align = "hv")
annotate_figure(fig_Rem_Apt_Mouse, top = text_grob("Aptamers Removed per Sample: Mouse", size = 14))
```
\newline

**Figure 4: Bar graph showing number of aptamers removed by eLoD filter per sample for mouse datasets.**

```{r Plot_Removed_Apt_Quality_Mouse}
#Funciton to check for outliers (for labeling purposes)
is_outlier = function(x) {return(x < quantile(x, 0.25) - 1.5 * IQR(x) | x > quantile(x, 0.75) + 1.5 * IQR(x))}

#Plot removed aptamer count by quality per sample - mouse
quality_sample_rem_mouse_df = rem_mouse_df %>% filter(!is.na(Quality)) %>% dplyr::group_by(Group_final,Sample_ID, Quality) %>% summarise(Apt_Count = n()) %>% mutate(Quality = factor(Quality,levels = (c("Low","Medium","High"))))

quality_sample_rem_mouse_plot = quality_sample_rem_mouse_df %>% 
  group_by(Quality) %>%
  mutate(outlier = ifelse(is_outlier(Apt_Count), Sample_ID, as.numeric(NA))) %>%
  ggplot(aes(x = Quality, y = Apt_Count, fill = Quality)) +
  geom_boxplot() +
  geom_text_repel(aes(label = outlier), na.rm = TRUE, hjust = -0.2, size = 3, max.overlaps = 10) +
  labs(title = "Normalized",
       subtitle = "Filter: Per Aptamer eLoD",
       x = "",
       y = "") +
  theme_bw() +
  ylim(0,150) +
  rremove("xlab") +
  theme(legend.position = "none") +
  theme(axis.text.x = element_text(vjust = 0.5, size = 12)) +
  theme(axis.title = element_text(size = 12))
#ggsave("quality_sample_rem_mouse_plot.png")
```

```{r Plot_Removed_Apt_Quality_Mouse_Alt}
#Plot removed aptamer count by quality per sample - mouse alt
quality_sample_rem_mouse_alt_df = rem_mouse_alt_df %>% filter(!is.na(Quality)) %>% dplyr::group_by(Group_final,Sample_ID, Quality) %>% summarise(Apt_Count = n()) %>% mutate(Quality = factor(Quality,levels = (c("Low","Medium","High"))))

quality_sample_rem_mouse_alt_plot = quality_sample_rem_mouse_alt_df %>%
  group_by(Quality) %>%
  mutate(outlier = ifelse(is_outlier(Apt_Count), Sample_ID, as.numeric(NA))) %>%
  ggplot(aes(x = Quality, y = Apt_Count, fill = Quality)) +
  geom_boxplot() +
  geom_text_repel(aes(label = outlier), na.rm = TRUE, hjust = -0.2, size = 3, max.overlaps = 10) +
  labs(title = "Pre-Normalization",
       subtitle = "Filter: Per Aptamer eLoD",
       x = "",
       y = "Aptamer Count") +
  theme_bw() +
  ylim(0,150) +
  rremove("xlab") +
  theme(legend.position = "none") +
  theme(axis.text.x = element_text(vjust = 0.5, size = 12)) +
  #theme(axis.text.x = element_blank()) +
  theme(axis.title = element_text(size = 12))
#ggsave("quality_sample_rem_mouse_alt_plot.png")
```

```{r Plot_ComboQSRemApt_Mouse}
fig_QS_Rem_Apt_Mouse = ggarrange(quality_sample_rem_mouse_alt_plot, quality_sample_rem_mouse_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_QS_Rem_Apt_Mouse, top = text_grob("Aptamers Removed per Sample by Quality: Mouse", size = 14))
ggsave("fig_QS_Rem_Apt_Mouse.png")
```

\newline

**Figure 5: Boxplots showing distribution of removed aptamer counts per sample, grouped by aptamer quality in mouse. Outliers - samples with an irregular number of aptamers removed - are labeled.**

```{r Plot_Removed_Apt_Human}
#Plot removed aptamer count - human
sample_rem_human_df = rem_human_df %>% dplyr::group_by(Clinical.Diagnosis,Sample_ID) %>% summarise(Apt_Count = n())

sample_rem_human_plot = sample_rem_human_df %>% 
  ggplot(aes(x = Sample_ID, y = Apt_Count, fill = Clinical.Diagnosis)) +
  geom_bar(stat = "identity") +
  labs(title = "Normalized",
       subtitle = "Filter: Per Aptamer eLoD",
       x = "",
       y = "Aptamer Count") +
  theme_bw() +
  ylim(0,120) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5))
#ggsave("sample_rem_human_plot.png")
```

```{r Plot_Removed_Apt_Human_Alt}
#Plot removed aptamer count - human
sample_rem_human_alt_df = rem_human_alt_df %>% dplyr::group_by(Clinical.Diagnosis,Sample_ID) %>% summarise(Apt_Count = n())

sample_rem_human_alt_plot = sample_rem_human_alt_df %>% 
  ggplot(aes(x = Sample_ID, y = Apt_Count, fill = Clinical.Diagnosis)) +
  geom_bar(stat = "identity") +
  labs(title = "Pre-Normalization",
       subtitle = "Filter: Per Aptamer eLoD",
       x = "",
       y = "Aptamer Count") +
  theme_bw() +
  ylim(0,120) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5))
#ggsave("sample_rem_human_alt_plot.png")
```

```{r Plot_ComboRemApt_Human}
fig_Rem_Apt_Human = ggarrange(sample_rem_human_plot, sample_rem_human_alt_plot, labels="AUTO", ncol = 1, nrow = 2, common.legend = TRUE, legend = "right", align = "hv")
annotate_figure(fig_Rem_Apt_Human, top = text_grob("Aptamers Removed per Sample: Human", size = 14))
```

\newline

**Figure 6: Bar graph showing number of aptamers removed by eLoD filter per sample for human datasets.**

```{r Plot_Removed_Quality_Mouse}
#Plot removed aptamer counts by quality
quality_rem_mouse_df = rem_mouse_df %>% dplyr::group_by(Quality) %>% summarise(Apt_Count = n())

test_df = rem_mouse_df %>% filter(is.na(Quality))

quality_rem_mouse_plot = quality_rem_mouse_df %>% 
  ggplot(aes(x = Quality, y = Apt_Count, fill = Quality)) +
  geom_bar(stat = "identity") +
  labs(title = "Normalized",
     subtitle = "",
     x = "",
     y = "Aptamer Count") +
theme_bw() +
  ylim(0,3000) +
theme(legend.position = "none") +
theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5))
# ggsave("quality_rem_mouse_plot.png")
```

```{r Plot_Removed_Quality_Mouse_Alt}
#Plot removed aptamer counts by quality
quality_rem_mouse_alt_df = rem_mouse_alt_df %>% dplyr::group_by(Quality) %>% summarise(Apt_Count = n())

quality_rem_mouse_alt_plot = quality_rem_mouse_alt_df %>% 
  ggplot(aes(x = Quality, y = Apt_Count, fill = Quality)) +
  geom_bar(stat = "identity") +
  labs(title = "Pre-Normalization",
       subtitle = "",
       x = "",
       y = "") +
  theme_bw() +
  ylim(0,3000) +
  theme(legend.position = "none") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5))
#ggsave("quality_rem_mouse_alt_plot.png")
```

```{r Plot_ComboQRemApt_Mouse}
fig_QRem_Apt_Mouse = ggarrange(quality_rem_mouse_plot, quality_rem_mouse_alt_plot, labels = "AUTO", ncol = 2, nrow = 1, legend = "none")
annotate_figure(fig_QRem_Apt_Mouse, top = text_grob("Aptamers Removed per Quality Category: Mouse", size = 14))
```
\newline

**Figure 7: Bar graph showing distribution of aptamers removed using eLoD filter, categorized by aptamer quality score in mouse system.**

```{r Quality_ExApt_Mouse}
#Comparing intensity values across samples for a random aptamer selected per quality category

#Look for example aptamers - one per quality category
lowQ_df = filt_mouse_df %>% filter(Quality == "Low") %>% dplyr::group_by(AptName, TargetFullName, Target, Organism, Type) %>% summarise(Sample_Count = n())
medQ_df = filt_mouse_df %>% filter(Quality == "Medium") %>% dplyr::group_by(AptName, TargetFullName, Target, Organism, Type) %>% summarise(Sample_Count = n())
highQ_df = filt_mouse_df %>% filter(Quality == "High") %>% dplyr::group_by(AptName, TargetFullName, Target, Organism, Type) %>% summarise(Sample_Count = n())

#Select example aptamers to plot intensity distribution: low = seq.10044.12, med = seq.10003.15, high = seq.10000.28
exapt_mouse_df = filt_mouse_df %>% filter(AptName == "seq.10044.12" | AptName == "seq.10003.15" | AptName == "seq.10000.28") %>%
                                    mutate(Quality_ExAptamer_Target = paste(Quality,AptName,Target,sep="_"))

#Plot sample intensities across samples, different series/color for each quality category
quality_exapt_mouse_plot = exapt_mouse_df %>%
  mutate(Quality_ExAptamer_Target = factor(Quality_ExAptamer_Target,levels=c("Low_seq.10044.12_WN10A","Medium_seq.10003.15_ZNF41","High_seq.10000.28_CRBB2"))) %>%
  ggplot(aes(x = Sample_ID, y = Intensity,color = Quality_ExAptamer_Target)) +
  geom_point(size = 2) +
  labs(title = "Mouse",
       x = "Sample_ID",
       y = "Intensity (RFU)") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5))
# ggsave("quality_exapt_mouse_plot.png")
quality_exapt_mouse_plot
```
```{r Quality_ExApt_Mouse_Alt}
#Comparing intensity values across samples for a random aptamer selected per quality category

#Look for example aptamers - one per quality category
lowQ_alt_df = filt_mouse_alt_df %>% filter(Quality == "Low") %>% dplyr::group_by(AptName, TargetFullName, Target, Organism, Type) %>% summarise(Sample_Count = n())
medQ_alt_df = filt_mouse_alt_df %>% filter(Quality == "Medium") %>% dplyr::group_by(AptName, TargetFullName, Target, Organism, Type) %>% summarise(Sample_Count = n())
highQ_alt_df = filt_mouse_alt_df %>% filter(Quality == "High") %>% dplyr::group_by(AptName, TargetFullName, Target, Organism, Type) %>% summarise(Sample_Count = n())

#Select example aptamers to plot intensity distribution: low = seq.10044.12, med = seq.10003.15, high = seq.10000.28
exapt_mouse_alt_df = filt_mouse_alt_df %>% filter(AptName == "seq.10044.12" | AptName == "seq.10003.15" | AptName == "seq.10000.28") %>%
                                    mutate(Quality_ExAptamer_Target = paste(Quality,AptName,Target,sep="_"))

#Plot sample intensities across samples, different series/color for each quality category
quality_exapt_mouse_alt_plot = exapt_mouse_alt_df %>%
  mutate(Quality_ExAptamer_Target = factor(Quality_ExAptamer_Target,levels=c("Low_seq.10044.12_WN10A","Medium_seq.10003.15_ZNF41","High_seq.10000.28_CRBB2"))) %>%
  ggplot(aes(x = Sample_ID, y = Intensity,color = Quality_ExAptamer_Target)) +
  geom_point(size = 2) +
  labs(title = "Mouse (Pre-Normalization)",
       x = "Sample_ID",
       y = "Intensity (RFU)") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5))
# ggsave("quality_exapt_mouse_alt_plot.png")
quality_exapt_mouse_alt_plot
```

\newline

**Figure 8: Scatter plot showing the intensity distribution for three aptamers: seq.10000.28 (high quality aptamer in mouse), seq.10044.12 (low quality), and seq.10003.15 (medium quality). The low quality aptamer has higher variation then both the medium and high quality aptamers in both the normalized and pre-normalization mouse datasets, although the difference is more apparent in the normalized dataset.**

```{r Plot_QSInt_Mouse_Alt}
#Plot intensity boxplots per sample, divided into quality categories
#qsInt_mouse_dfs = filt_mouse_df %>% filter(eLoD_Flag == TRUE) %>% filter(!is.na(Quality) & SampleType == "Sample") %>% group_by(Sample_ID) %>% group_split(Sample_ID)

qsInt_low_mouse_plot = filt_mouse_alt_df %>% filter(eLoD_Flag == TRUE) %>% filter(SampleType == "Sample") %>% filter(Quality == "Low") %>%
  mutate(Sample_Quality = paste(Sample_ID,Quality,sep="_")) %>%
  #mutate(outlier = ifelse(is_outlier(Intensity), Sample_ID, as.numeric(NA))) %>%
  ggplot(aes(x = Sample_ID, y = log(Intensity,2))) +
  geom_boxplot(fill = "#F8766D", outlier.shape = NA) +
  #geom_text_repel(aes(label = outlier), na.rm = TRUE, hjust = -0.2, size = 3, max.overlaps = 10) +
  labs(title = "Low Quality Aptamers",
       subtitle = "",
       x = "Samples",
       y = "Log2(Intensity)") +
  theme_bw() +
  ylim(0,20) +
  theme(legend.position = "none") +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5))
  theme(axis.text.x = element_blank()) +
  theme(axis.title = element_text(size = 14)) +
  theme(title = element_text(size = 18))
ggsave("qsInt_low_mouse_plot.png", width = 2187, height = 1351, units = "px")
qsInt_low_mouse_plot

qsInt_med_mouse_plot = filt_mouse_alt_df %>% filter(eLoD_Flag == TRUE) %>% filter(SampleType == "Sample") %>% filter(Quality == "Medium") %>%
  mutate(Sample_Quality = paste(Sample_ID,Quality,sep="_")) %>%
  #mutate(outlier = ifelse(is_outlier(Intensity), Sample_ID, as.numeric(NA))) %>%
  ggplot(aes(x = Sample_ID, y = log(Intensity,2))) +
  geom_boxplot(fill="#00BF7D", outlier.shape = NA) +
  #geom_text_repel(aes(label = outlier), na.rm = TRUE, hjust = -0.2, size = 3, max.overlaps = 10) +
  labs(title = "Medium Quality Aptamers",
       subtitle = "",
       x = "Samples",
       y = "") +
  theme_bw() +
  ylim(0,20) +
  theme(legend.position = "none") +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5))
  theme(axis.text.x = element_blank()) +
  theme(axis.title = element_text(size = 14)) +
  theme(title = element_text(size = 18))
ggsave("qsInt_med_mouse_plot.png")
qsInt_med_mouse_plot

qsInt_high_mouse_plot = filt_mouse_alt_df %>% filter(eLoD_Flag == TRUE) %>% filter(SampleType == "Sample") %>% filter(Quality == "High") %>%
  mutate(Sample_Quality = paste(Sample_ID,Quality,sep="_")) %>%
  #mutate(outlier = ifelse(is_outlier(Intensity), Sample_ID, as.numeric(NA))) %>%
  ggplot(aes(x = Sample_ID, y = log(Intensity,2))) +
  geom_boxplot(fill = "#00B0F6", outlier.shape = NA) +
  #geom_text_repel(aes(label = outlier), na.rm = TRUE, hjust = -0.2, size = 3, max.overlaps = 10) +
  labs(title = "High Quality Aptamers",
       subtitle = "",
       x = "Samples",
       y = "") +
  theme_bw() +
  ylim(0,20) +
  theme(legend.position = "none") +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5)) +
  theme(axis.text.x = element_blank()) +
  theme(axis.title = element_text(size = 14)) +
  theme(title = element_text(size = 18))
ggsave("qsInt_high_mouse_plot.png")
qsInt_high_mouse_plot
```
**Figure 9: SomaLogic Log2 intensity boxplot distributions separated by mouse aptamer quality score. All three quality categories appear to have similar intensity distributions per sample.**

```{r AptamerCV_Mouse}
#Calculate CV of each aptamer - plot all CVs as boxplot
aptamer_CVall_mouse_df = filt_mouse_df %>%
  mutate(Log2_Intensity = log(Intensity,2)) %>%
  mutate(Group_final = ifelse(is.na(Group_final),
                              ifelse(SampleType == "Buffer",
                                     "Buffer",
                                     ifelse(SampleType == "Calibrator",
                                            "Calibrator",
                                            ifelse(startsWith(Sample_ID,"Mouse 35"),
                                                   "Pooled 35uL",
                                                   ifelse(endsWith(Sample_ID,"hem"),
                                                          "Hemolysis",
                                                          ifelse(startsWith(Sample_ID, "Mouse A"),
                                                                  "Pooled 55uL",
                                                                  "Pooled Diluted 55uL"))))),
                              Group_final)) %>%
  group_by(AptName) %>%
  summarise(Mean = mean(Log2_Intensity), SD = sd(Log2_Intensity), CV = 100*sqrt(2^(log(2)*(SD^2))-1))

aptamer_CVgroup_mouse_df = filt_mouse_df %>% 
  mutate(Log2_Intensity = log(Intensity,2)) %>%
  mutate(Group_final = ifelse(is.na(Group_final),
                              ifelse(SampleType == "Buffer",
                                     "Buffer",
                                     ifelse(SampleType == "Calibrator",
                                            "Calibrator",
                                            ifelse(startsWith(Sample_ID,"Mouse 35"),
                                                   "Pooled 35uL",
                                                   ifelse(endsWith(Sample_ID,"hem"),
                                                          "Hemolysis",
                                                          ifelse(startsWith(Sample_ID, "Mouse A"),
                                                                  "Pooled 55uL",
                                                                  "Pooled Diluted 55uL"))))),
                              Group_final)) %>%
  group_by(Group_final, AptName) %>% 
  summarize(Mean = mean(Log2_Intensity), SD = sd(Log2_Intensity), CV = 100*sqrt(2^(log(2)*(SD^2))-1))

#Plot CVs - all
aptamer_CVall_mouse_plot = aptamer_CVall_mouse_df %>%
  ggplot(aes(x = 0, y = CV)) +
  geom_boxplot(outlier.shape = NA, fill = "darkgrey") +
  labs(title = "Aptamer CV Distribution: Mouse",
       subtitle = "Outliers are hidden") +
  theme_bw() +
  ylim(0, 200) +
  stat_mean_sd_text(y.pos = 190, size = 4) +
  theme(legend.position = "none") + 
  rremove("xlab") +
  theme(axis.title = element_text(size = 14)) +
  theme(title = element_text(size = 14))
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5))
ggsave("aptamer_CVall_mouse_plot.png")
aptamer_CVall_mouse_plot

#Plot CVs - by group
aptamer_CVgroup_mouse_plot = aptamer_CVgroup_mouse_df %>%
  filter(!Group_final == "Buffer") %>% 
  mutate(`Condition Group` = Group_final) %>%
  ggplot(aes(x = Group_final, y = CV, fill = `Condition Group`)) +
  geom_boxplot(outlier.shape = NA) +
  labs(title = "Mouse - Normalized") +
  theme_bw() +
  ylim(0, 100) +
  stat_mean_sd_text(y.pos = 90, size = 3, angle=90) +
  theme(axis.title.x = element_blank()) +
  theme(axis.text.x = element_blank()) +
  theme(axis.text.y = element_text(size = 12)) +
  theme(axis.title = element_text(size = 14)) +
  theme(title = element_text(size = 14)) +
  theme(legend.title = element_text(size = 14)) +
  theme(legend.text = element_text(size = 12)) 
ggsave("aptamer_CVgroup_mouse_plot.png", width = 7.29, height = 4.51, units = "in")
aptamer_CVgroup_mouse_plot
```

```{r AptamerCV_Mouse_Alt}
#Calculate CV of each aptamer - plot all CVs as boxplot
aptamer_CVall_mouse_alt_df = filt_mouse_alt_df %>%
  mutate(Log2_Intensity = log(Intensity,2)) %>%
  mutate(Group_final = ifelse(is.na(Group_final),
                              ifelse(SampleType == "Buffer",
                                     "Buffer",
                                     ifelse(SampleType == "Calibrator",
                                            "Calibrator",
                                            ifelse(startsWith(Sample_ID,"Mouse 35"),
                                                   "Pooled 35uL",
                                                   ifelse(endsWith(Sample_ID,"hem"),
                                                          "Hemolysis",
                                                          ifelse(startsWith(Sample_ID, "Mouse A"),
                                                                  "Pooled 55uL",
                                                                  "Pooled Diluted 55uL"))))),
                              Group_final)) %>%
  group_by(AptName) %>%
  summarise(Mean = mean(Log2_Intensity), SD = sd(Log2_Intensity), CV = 100*sqrt(2^(log(2)*(SD^2))-1))

aptamer_CVgroup_mouse_alt_df = filt_mouse_alt_df %>% 
  mutate(Log2_Intensity = log(Intensity,2)) %>%
  mutate(Group_final = ifelse(is.na(Group_final),
                              ifelse(SampleType == "Buffer",
                                     "Buffer",
                                     ifelse(SampleType == "Calibrator",
                                            "Calibrator",
                                            ifelse(startsWith(Sample_ID,"Mouse 35"),
                                                   "Pooled 35uL",
                                                   ifelse(endsWith(Sample_ID,"hem"),
                                                          "Hemolysis",
                                                          ifelse(startsWith(Sample_ID, "Mouse A"),
                                                                  "Pooled 55uL",
                                                                  "Pooled Diluted 55uL"))))),
                              Group_final)) %>%
  group_by(Group_final, AptName) %>% 
  summarize(Mean = mean(Log2_Intensity), SD = sd(Log2_Intensity), CV = 100*sqrt(2^(log(2)*(SD^2))-1))

#Plot CVs - all
aptamer_CVall_mouse_alt_plot = aptamer_CVall_mouse_alt_df %>%
  ggplot(aes(x = 0, y = CV)) +
  geom_boxplot(outlier.shape = NA, fill = "darkgrey") +
  labs(title = "Aptamer CV Distribution: Mouse (Pre-Normalization)",
       subtitle = "Outliers are hidden") +
  theme_bw() +
  ylim(0, 200) +
  stat_mean_sd_text(y.pos = 190, size = 4) +
  theme(legend.position = "none") + 
  rremove("xlab") +
  theme(axis.title = element_text(size = 14)) +
  theme(title = element_text(size = 14))
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5))
ggsave("aptamer_CVall_mouse_alt_plot.png")
aptamer_CVall_mouse_alt_plot

#Plot CVs - by group
aptamer_CVgroup_mouse_alt_plot = aptamer_CVgroup_mouse_alt_df %>%
  ggplot(aes(x = Group_final, y = CV, fill = Group_final)) +
  geom_boxplot(outlier.shape = NA) +
  labs(title = "Aptamer CV Distribution (By Group): Mouse (Pre-Normalization)",
       subtitle = "Outliers are hidden",
       x = "") +
  theme_bw() +
  ylim(0, 100) +
  stat_mean_sd_text(y.pos = 90, size = 3, angle=90) +
  theme(legend.position = "none") + 
  rremove("xlab") +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.5, size = 8)) +
  theme(axis.title = element_text(size = 14)) +
  theme(title = element_text(size = 14))
ggsave("aptamer_CVgroup_mouse_alt_plot.png")
aptamer_CVgroup_mouse_alt_plot
```

```{r AptamerCV_Human}
#Calculate CV of each aptamer - plot all CVs as boxplot
aptamer_CVall_human_df = filt_human_df %>%
  mutate(Log2_Intensity = log(Intensity,2)) %>%
  mutate(Clinical.Diagnosis = ifelse(is.na(Clinical.Diagnosis),
                                      ifelse(SampleType == "Calibrator",
                                             "Calibrator",
                                             ifelse(SampleType == "Buffer",
                                                    "Buffer",
                                                    ifelse(SampleType == "QC",
                                                           "QC",
                                                           ifelse(startsWith(Sample_ID,"Human 35"),
                                                                  "Pooled 35uL",
                                                                  ifelse(startsWith(Sample_ID,"Human A"),
                                                                          "Pooled 55uL",
                                                                          "Pooled Diluted 55uL"))))),
                                      Clinical.Diagnosis)) %>%
  group_by(AptName) %>%
  summarise(Mean = mean(Log2_Intensity), SD = sd(Log2_Intensity), CV = 100*sqrt(2^(log(2)*(SD^2))-1))

aptamer_CVgroup_human_df = filt_human_df %>% 
  mutate(Log2_Intensity = log(Intensity,2)) %>%
  mutate(Clinical.Diagnosis = ifelse(is.na(Clinical.Diagnosis),
                                      ifelse(SampleType == "Calibrator",
                                             "Calibrator",
                                             ifelse(SampleType == "Buffer",
                                                    "Buffer",
                                                    ifelse(SampleType == "QC",
                                                           "QC",
                                                           ifelse(startsWith(Sample_ID,"Human 35"),
                                                                  "Pooled 35uL",
                                                                  ifelse(startsWith(Sample_ID,"Human A"),
                                                                          "Pooled 55uL",
                                                                          "Pooled Diluted 55uL"))))),
                                      Clinical.Diagnosis)) %>%
  group_by(Clinical.Diagnosis, AptName) %>%
  summarize(Mean = mean(Log2_Intensity), SD = sd(Log2_Intensity), CV = 100*sqrt(2^(log(2)*(SD^2))-1)) %>%
  mutate(Clinical.Diagnosis = factor(Clinical.Diagnosis,levels=c("Calibrator","Buffer","QC","Benign","Benign prostatic hyperplasia","Prostatic intraepithelial neoplasia","Adenocarcinoma", "Pooled 35uL","Pooled 55uL","Pooled Diluted 55uL"))) %>%
                                        ungroup() %>%
                                        dplyr::arrange(Clinical.Diagnosis) %>%
                                        mutate(row_ID = row_number())

#Plot CVs - all
aptamer_CVall_human_plot = aptamer_CVall_human_df %>%
  ggplot(aes(x = 0, y = CV)) +
  geom_boxplot(outlier.shape = NA, fill = "darkgrey") +
  labs(title = "Aptamer CV Distribution: Human",
       subtitle = "Outliers are hidden") +
  theme_bw() +
  ylim(0, 200) +
  stat_mean_sd_text(y.pos = 190, size = 4) +
  theme(legend.position = "none") + 
  rremove("xlab") +
  theme(axis.title = element_text(size = 14)) +
  theme(title = element_text(size = 14))
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5))
ggsave("aptamer_CVall_human_plot.png")
aptamer_CVall_human_plot

#Plot CVs - by group
aptamer_CVgroup_human_plot = aptamer_CVgroup_human_df %>%
  filter(!Clinical.Diagnosis == "Buffer") %>% 
  mutate(`Condition Group` = Clinical.Diagnosis) %>%
  ggplot(aes(x = reorder(Clinical.Diagnosis, row_ID), y = CV, fill = `Condition Group`)) +
  geom_boxplot(outlier.shape = NA) +
  labs(title = "Human - Normalized") +
  theme_bw() +
  ylim(0, 100) +
  stat_mean_sd_text(y.pos = 90, size = 3, angle=90) +
  theme(axis.title.x = element_blank()) +
  theme(axis.text.x = element_blank()) +
  theme(axis.text.y = element_text(size = 12)) +
  theme(axis.title = element_text(size = 14)) +
  theme(title = element_text(size = 14)) +
  theme(legend.title = element_text(size = 14)) +
  theme(legend.text = element_text(size = 12)) 
ggsave("aptamer_CVgroup_human_plot.png", width = 7.29, height = 4.51, units = "in")
aptamer_CVgroup_human_plot
```

```{r AptamerCV_Human_Alt}
#Calculate CV of each aptamer - plot all CVs as boxplot
aptamer_CVall_human_alt_df = filt_human_alt_df %>%
  mutate(Log2_Intensity = log(Intensity,2)) %>%
  mutate(Clinical.Diagnosis = ifelse(is.na(Clinical.Diagnosis),
                                      ifelse(SampleType == "Calibrator",
                                             "Calibrator",
                                             ifelse(SampleType == "Buffer",
                                                    "Buffer",
                                                    ifelse(SampleType == "QC",
                                                           "QC",
                                                           ifelse(startsWith(Sample_ID,"Human 35"),
                                                                  "Pooled 35uL",
                                                                  ifelse(startsWith(Sample_ID,"Human A"),
                                                                          "Pooled 55uL",
                                                                          "Pooled Diluted 55uL"))))),
                                      Clinical.Diagnosis)) %>%
  group_by(AptName) %>%
  summarise(Mean = mean(Log2_Intensity), SD = sd(Log2_Intensity), CV = 100*sqrt(2^(log(2)*(SD^2))-1))

aptamer_CVgroup_human_alt_df = filt_human_alt_df %>% 
  mutate(Log2_Intensity = log(Intensity,2)) %>%
  mutate(Clinical.Diagnosis = ifelse(is.na(Clinical.Diagnosis),
                                      ifelse(SampleType == "Calibrator",
                                             "Calibrator",
                                             ifelse(SampleType == "Buffer",
                                                    "Buffer",
                                                    ifelse(SampleType == "QC",
                                                           "QC",
                                                           ifelse(startsWith(Sample_ID,"Human 35"),
                                                                  "Pooled 35uL",
                                                                  ifelse(startsWith(Sample_ID,"Human A"),
                                                                          "Pooled 55uL",
                                                                          "Pooled Diluted 55uL"))))),
                                      Clinical.Diagnosis)) %>%
  group_by(Clinical.Diagnosis, AptName) %>% 
  summarize(Mean = mean(Log2_Intensity), SD = sd(Log2_Intensity), CV = 100*sqrt(2^(log(2)*(SD^2))-1))

#Plot CVs - all
aptamer_CVall_human_alt_plot = aptamer_CVall_human_alt_df %>%
  ggplot(aes(x = 0, y = CV)) +
  geom_boxplot(outlier.shape = NA, fill = "darkgrey") +
  labs(title = "Aptamer CV Distribution: Human (Pre-Normalization)",
       subtitle = "Outliers are hidden") +
  theme_bw() +
  ylim(0, 200) +
  stat_mean_sd_text(y.pos = 190, size = 4) +
  theme(legend.position = "none") + 
  rremove("xlab") +
  theme(axis.title = element_text(size = 14)) +
  theme(title = element_text(size = 14))
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5))
ggsave("aptamer_CVall_human_alt_plot.png")
aptamer_CVall_human_alt_plot

#Plot CVs - by group
aptamer_CVgroup_human_alt_plot = aptamer_CVgroup_human_alt_df %>%
  ggplot(aes(x = Clinical.Diagnosis, y = CV, fill = Clinical.Diagnosis)) +
  geom_boxplot(outlier.shape = NA) +
  labs(title = "Aptamer CV Distribution (By Group): Human (Pre-Normalization)",
       subtitle = "Outliers are hidden",
       x = "") +
  theme_bw() +
  ylim(0, 100) +
  stat_mean_sd_text(y.pos = 90, size = 3, angle=90) +
  theme(legend.position = "none") + 
  rremove("xlab") +
  theme(axis.text.x = element_text(angle = 55, vjust = 0.5, size = 8)) +
  theme(axis.title = element_text(size = 14)) +
  theme(title = element_text(size = 14))
ggsave("aptamer_CVgroup_human_alt_plot.png")
aptamer_CVgroup_human_alt_plot
```
**Figure 10: Aptamer CV distributions for murine and human data, grouped and colored by condition. Overall, low CVs are observed in both species datasets, human having slightly lower aptamer CVs than murine.**

```{r Combo_Plot_eLoD_Filter_AptCount, include = FALSE}
#Plot filtered aptamer counts per sample
fig_Combo_CVplots = ggarrange(aptamer_CVgroup_mouse_plot, aptamer_CVgroup_human_plot, labels=c("C","D"), ncol = 1, nrow = 2)
annotate_figure(fig_Combo_CVplots, top = text_grob("Aptamer CV Distribution per Condition", size = 18))
ggsave("fig_Combo_CVplots.png", width = 7.29, height = 6.5)
```

## SomaLogic Data Analysis

### Mouse (Project Disney) Analysis

#### Assess Limit of Detection (Dilution Series) - PRE-NORMALIZATION DATA ONLY

```{r Stat_DataTransform_Dilution_Mouse}
#Mouse Dilution Series: 35uL vs. 55uL vs. 55uL Diluted (USES ALT DATA - PRE-NORMALIZATION)

#Transform data to wide form for statistical test
dilution_mouse_wide_df = filt_mouse_alt_df %>% filter(startsWith(Sample_ID,"Mouse")) %>%
                                                filter(eLoD_Flag == TRUE) %>%
                                                mutate(Volume_uL = ifelse(startsWith(Sample_ID,"Mouse 35"),
                                                                          "35",
                                                                          ifelse(startsWith(Sample_ID,"Mouse A"),
                                                                                 "55",
                                                                                 "55_Diluted"))) %>%
                                                mutate(Log2_Intensity = log(Intensity,2)) %>%
                                                dplyr::group_by(AptName,Volume_uL) %>%
                                                summarize(data = list(Log2_Intensity), count = n(), .groups = "drop") %>%
                                                pivot_wider(names_from = Volume_uL, values_from = c(data,count)) %>%
                                                filter(count_35 > 1 & count_55 > 1 & count_55_Diluted > 1)

```

```{r WiT_Dilution_Mouse}
#Wilcox test - store pvalue and difference
WiT_res_mouse_df = dilution_mouse_wide_df %>%
  rowwise() %>%
  mutate(wt_pval_A = wilcox.test(unlist(data_35), unlist(data_55))$p.value) %>%
  mutate(Diff_A = (median(unlist(data_35)) - median(unlist(data_55)))) %>%
  ungroup() %>%
  mutate(wt_pval_A_BH = p.adjust(wt_pval_A, method = "BH")) %>%
  rowwise() %>%
  mutate(wt_pval_B = wilcox.test(unlist(data_35), unlist(data_55_Diluted))$p.value) %>%
  mutate(Diff_B = (median(unlist(data_35)) - median(unlist(data_55_Diluted)))) %>%
  ungroup() %>%
  mutate(wt_pval_B_BH = p.adjust(wt_pval_B, method = "BH")) %>%
  rowwise() %>%
  mutate(wt_pval_C = wilcox.test(unlist(data_55), unlist(data_55_Diluted))$p.value) %>%
  mutate(Diff_C = (median(unlist(data_55)) - median(unlist(data_55_Diluted)))) %>%
  ungroup() %>%
  mutate(wt_pval_C_BH = p.adjust(wt_pval_C, method = "BH"))

partial_apt_anno_df = apt_anno_mouse_df %>%
  dplyr::select(c("AptName", "Target", "TargetFullName", "UniProt"))

WiT_res_mouse_df = merge(WiT_res_mouse_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_res_mouse_df = WiT_res_mouse_df %>%
  mutate(p_val_A_sig = ifelse(wt_pval_A_BH < 0.05, "Signficant", "Non"),
         proteins_A_sig = ifelse(wt_pval_A_BH < 0.05, Target, ""),
         p_val_B_sig = ifelse(wt_pval_B_BH < 0.05, "Signficant", "Non"),
         proteins_B_sig = ifelse(wt_pval_B_BH < 0.05, Target, ""),
         p_val_C_sig = ifelse(wt_pval_C_BH < 0.05, "Signficant", "Non"),
         proteins_C_sig = ifelse(wt_pval_C_BH < 0.05, Target, "")
  )
```

```{r Plot_WiT_pval_Dilution_Mouse}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_A_dist_mouse_plot = WiT_res_mouse_df %>%
  ggplot(aes(wt_pval_A)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "",
       y = "Aptamer Count",
       title = "",
       subtitle = "35uL vs. 55uL") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_A_dist_mouse_plot.png")

WiT_pval_B_dist_mouse_plot = WiT_res_mouse_df %>%
  ggplot(aes(wt_pval_B)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "Wilcox Test Raw P-value",
       y = "",
       title = "",
       subtitle = "35uL vs. 55uL Diluted") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_B_dist_mouse_plot.png")

WiT_pval_C_dist_mouse_plot = WiT_res_mouse_df %>%
  ggplot(aes(wt_pval_C)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "",
       y = "",
       title = "",
       subtitle = "55uL vs. 55uL Diluted") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_C_dist_mouse_plot.png")

fig_pval_Dilution_Mouse = ggarrange(WiT_pval_A_dist_mouse_plot, WiT_pval_B_dist_mouse_plot, WiT_pval_C_dist_mouse_plot, labels="AUTO", ncol = 3, nrow = 1)
annotate_figure(fig_pval_Dilution_Mouse, top = text_grob("P-Value Distribution of Wilcox Test Results: Mouse", size = 14))
```
\newline

**Figure 11: P-value distributions for the wilcox tests performed on the mouse dilution series data (pre-normalization).**

```{r Plot_WiT_volc_Dilution_Mouse}
#Wilcoxon volcano Plot
WiT_A_mouse_volcano_plot = WiT_res_mouse_df %>%
  ggplot(aes(x = Diff_A, y = -log(wt_pval_A_BH, 10), group = p_val_A_sig)) +
  geom_point(aes(color = p_val_A_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed", color = "red") +
  geom_text_repel(aes(label = proteins_A_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "",
       x = "Log2(FC)",
       y = "-log10(p-value)",
       subtitle = "35uL vs. 55uL") +
  theme_bw() +
  xlim(-2,2) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic")) +
  theme(axis.text.x = element_text(vjust = 0.5, size = 12)) +
  theme(axis.text.y = element_text(vjust = 0.5, size = 12)) +
  theme(axis.title = element_text(vjust = 0.5, size = 14)) +
  theme(title = element_text(size = 14))
#ggsave("WiT_A_mouse_volcano_plot.png")
# WiT_A_mouse_volcano_plot

WiT_A_mouse_volcano_plotly = ggplotly(WiT_A_mouse_volcano_plot, tooltip = "Target")
saveWidget(WiT_A_mouse_volcano_plotly, "WiT_A_mouse_volcano_plotly.html", selfcontained = F, libdir = "lib")

WiT_B_mouse_volcano_plot = WiT_res_mouse_df %>%
  ggplot(aes(x = Diff_B, y = -log(wt_pval_B_BH, 10), group = p_val_B_sig)) +
  geom_point(aes(color = p_val_B_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed", color = "red") +
  geom_text_repel(aes(label = proteins_B_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "",
       x = "Log2(FC)",
       y = "",
       subtitle = "35uL vs. 55uL D") +
  theme_bw() +
  xlim(-2,2) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic")) +
  theme(axis.text.x = element_text(vjust = 0.5, size = 12)) +
  theme(axis.text.y = element_text(vjust = 0.5, size = 12)) +
  theme(axis.title = element_text(vjust = 0.5, size = 14)) +
  theme(title = element_text(size = 14))
#ggsave("WiT_B_mouse_volcano_plot.png")
# WiT_B_mouse_volcano_plot

WiT_B_mouse_volcano_plotly = ggplotly(WiT_B_mouse_volcano_plot, tooltip = "Target")
saveWidget(WiT_B_mouse_volcano_plotly, "WiT_B_mouse_volcano_plotly.html", selfcontained = F, libdir = "lib")

WiT_C_mouse_volcano_plot = WiT_res_mouse_df %>%
  ggplot(aes(x = Diff_C, y = -log(wt_pval_C_BH, 10), group = p_val_C_sig)) +
  geom_point(aes(color = p_val_C_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed", color = "red") +
  geom_text_repel(aes(label = proteins_C_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "",
       x = "Log2(FC)",
       y = "",
       subtitle = "55uL vs. 55uL D") +
  theme_bw() +
  xlim(-2,2) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic")) +
  theme(axis.text.x = element_text(vjust = 0.5, size = 12)) +
  theme(axis.text.y = element_text(vjust = 0.5, size = 12)) +
  theme(axis.title = element_text(vjust = 0.5, size = 14)) +
  theme(title = element_text(size = 14))
#ggsave("WiT_C_mouse_volcano_plot.png")
# WiT_C_mouse_volcano_plot

WiT_C_mouse_volcano_plotly = ggplotly(WiT_C_mouse_volcano_plot, tooltip = "Target")
saveWidget(WiT_C_mouse_volcano_plotly, "WiT_C_mouse_volcano_plotly.html", selfcontained = F, libdir = "lib")

fig_WiT_Dilution_Mouse = ggarrange(WiT_A_mouse_volcano_plot, WiT_B_mouse_volcano_plot, WiT_C_mouse_volcano_plot, ncol = 3, nrow = 1)
annotate_figure(fig_WiT_Dilution_Mouse, top = text_grob("SomaLogic Murine DE: Dilution Set", size = 18))
ggsave("fig_WiT_Dilution_Mouse.png", width = 6, height = 3, units = "in")
```

\newline

**Figure 12: Volcano plots showing significantly differentially expressed proteins between mouse dilution series samples A) 35uL vs. 55uL, B) 35uL vs. 55uL Diluted [35uL sample, 20uL PBS], C) 55uL vs. 55uL Diluted.**

**Table 2: Number of aptamers/proteins detected per mouse dilution series sample (pre-normalization). The 55uL Diluted samples have the lowest counts, but all are decently high (max 7596 aptamers).**

```{r Protein_Check_DilutionSeries_Mouse}
#Check protein ID counts across dilutions
dilutionprots_mouse_df = filt_mouse_alt_df %>% filter(startsWith(Sample_ID,"Mouse")) %>%
                                                filter(eLoD_Flag == TRUE) %>%
                                                mutate(Category = ifelse(startsWith(Sample_ID,"Mouse 35"),
                                                                          "35uL",
                                                                          ifelse(startsWith(Sample_ID,"Mouse A"),
                                                                                 "55uL",
                                                                                 "55uL Diluted"))) %>%
                                                group_by(Category,Sample_ID) %>%
                                                summarise(AptCount = n())

#dilutionprots_mouse_df = WiT_res_mouse_df %>% nrow()

knitr::kable(dilutionprots_mouse_df)
```

```{r Plot_WiT_box_Dilution_Mouse}
#Organize FC in long format
WiT_boxDS_mouse_df = WiT_res_mouse_df %>% pivot_longer(names_to = "Dilution_Comparison",
                                                        values_to = "FC",
                                                        cols = starts_with("Diff")) %>%
                                          mutate(Dilution_Comparison = ifelse(Dilution_Comparison == "Diff_A",
                                                                              "35uL-55uL",
                                                                              ifelse(Dilution_Comparison == "Diff_B",
                                                                                     "35uL-55uL_D",
                                                                                     "55uL-55uL_D")))

#Boxplot of intensity distributions per sample - mouse
WiT_boxDS_mouse_plot = WiT_boxDS_mouse_df %>%
  ggplot(aes(x = Dilution_Comparison, y = FC, fill = Dilution_Comparison)) +
  geom_boxplot(alpha = 0.5) +
  #geom_text(aes(label = median(FC))) +
  labs(title = "Dilution FC Distribution: Mouse",
       subtitle = "",
       y = "Log2(FC)") +
  theme_bw() +
  #stat_mean_sd_text(size = 2, y.pos = 6000) +
  #ylim(0,6000) +
  geom_hline(yintercept=log2(55/35), colour="purple4", linetype = "dashed") +
  theme(legend.position = "none") +
  theme(axis.text.x = element_text(vjust = 0.5, size = 12)) +
  theme(axis.text.y = element_text(vjust = 0.5, size = 12)) +
  theme(axis.title.x = element_blank()) +
  theme(axis.title = element_text(vjust = 0.5, size = 14)) +
  theme(title = element_text(size = 14))
ggsave("WiT_boxDS_mouse_plot.png", width = 4, height = 3, units = "in")
WiT_boxDS_mouse_plot
```
\newline

**Figure 13: Boxplot FC distributions of murine dilution series data. Median FC is on target for expected dilutions ratio from 35 to 55L (0.6x).**

#### Hemolysis Comparison

```{r Stat_DataTransform_Hem_Mouse}
#Mouse Hemolysis v. Pooled 55uL

#Transform data to wide form for statistical test
hem_mouse_wide_df = filt_mouse_df %>% filter(eLoD_Flag == TRUE) %>%
                                      filter(endsWith(Sample_ID,"hem") | startsWith(Sample_ID,"Mouse A")) %>%
                                      mutate(Group_final = ifelse(endsWith(Sample_ID,"hem"),
                                                                "Hemolysis",
                                                                "Pooled_55uL")) %>%
                                      dplyr::group_by(AptName,Target,TargetFullName,UniProt,Group_final) %>%
                                      mutate(Log2_Intensity = log(Intensity,2)) %>%
                                      summarize(data = list(Log2_Intensity), count = n()) %>%
                                      pivot_wider(names_from = Group_final, values_from = c(data,count)) %>%
                                      filter(count_Hemolysis > 1 & count_Pooled_55uL > 1)
```

```{r Stat_DataTransform_Hem_Mouse_Alt}
#Mouse Alt Hemolysis v. Pooled 55uL

#Transform data to wide form for statistical test
hem_mouse_alt_wide_df = filt_mouse_alt_df %>% filter(eLoD_Flag == TRUE) %>%
                                              filter(endsWith(Sample_ID,"hem") | startsWith(Sample_ID,"Mouse A")) %>%
                                              mutate(Group_final = ifelse(endsWith(Sample_ID,"hem"),
                                                                "Hemolysis",
                                                                "Pooled_55uL")) %>%
                                              dplyr::group_by(AptName,Target,TargetFullName,UniProt,Group_final) %>%
                                              mutate(Log2_Intensity = log(Intensity,2)) %>%
                                              summarize(data = list(Log2_Intensity), count = n(), .groups = "drop") %>%
                                              pivot_wider(names_from = Group_final, values_from = c(data,count)) %>%
                                              filter(count_Hemolysis > 1 & count_Pooled_55uL > 1)
```

```{r WiT_Hem_Mouse}
#Wilcox test - store pvalue and difference
WiT_hem_mouse_df = hem_mouse_wide_df %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_Hemolysis), unlist(data_Pooled_55uL))$p.value) %>%
  mutate(Diff = (median(unlist(data_Hemolysis)) - median(unlist(data_Pooled_55uL)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

#WiT_hem_mouse_df = merge(WiT_hem_mouse_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_hem_mouse_df = WiT_hem_mouse_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r WiT_Hem_Mouse_Alt}
#Wilcox test - store pvalue and difference
WiT_hem_mouse_alt_df = hem_mouse_alt_wide_df %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_Hemolysis), unlist(data_Pooled_55uL))$p.value) %>%
  mutate(Diff = (median(unlist(data_Hemolysis)) - median(unlist(data_Pooled_55uL)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

#WiT_hem_mouse_alt_df = merge(WiT_hem_mouse_alt_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_hem_mouse_alt_df = WiT_hem_mouse_alt_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r Plot_WiT_pval_Hem_Mouse}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_hem_mouse_plot = WiT_hem_mouse_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "Aptamer Count",
       title = "Mouse",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_hem_mouse_plot.png")
```

```{r Plot_WiT_pval_Hem_Mouse_Alt}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_hem_mouse_alt_plot = WiT_hem_mouse_alt_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "",
       title = "Mouse (Pre-Normalization)",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_hem_mouse_alt_plot.png")
```

```{r Plot_PvalCombo_Hem_Mouse}
fig_pval_hem_Mouse = ggarrange(WiT_pval_dist_hem_mouse_plot, WiT_pval_dist_hem_mouse_alt_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_pval_hem_Mouse, top = text_grob("P-Value Distribution (Wilcox Test): Hemolysis v. Pooled 55uL", size = 14))
```
\newline

**Figure 14: P-value distributions for the Wilcox tests performed on the mouse data comparing intensity (RFU) values between aptamers in the Hemolysis and Pooled 55uL groups. A) Normalized mouse data, B) Pre-normalization mouse data.**

```{r Plot_WiT_volc_Hem_Mouse}
#Wilcox volcano Plot
WiT_hem_mouse_volcano_plot = WiT_hem_mouse_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Mouse",
       x = "Diff(Hemolysis - Pooled_55uL)",
       y = "-log10(p-value)",
       subtitle = "") +
  theme_bw() +
  xlim(-10,10) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_hem_mouse_volcano_plot.png")

WiT_hem_mouse_volcano_plotly = ggplotly(WiT_hem_mouse_volcano_plot, tooltip = "Target")
saveWidget(WiT_hem_mouse_volcano_plotly, "WiT_hem_mouse_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_WiT_volc_Hem_Mouse_Alt}
#Wilcox volcano Plot
WiT_hem_mouse_alt_volcano_plot = WiT_hem_mouse_alt_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 150) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Mouse (Pre-Normalization)",
       x = "Diff(Hemolysis - Pooled_55uL)",
       y = "",
       subtitle = "") +
  theme_bw() +
  xlim(-10,10) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_hem_mouse_alt_volcano_plot.png")

WiT_hem_mouse_alt_volcano_plotly = ggplotly(WiT_hem_mouse_alt_volcano_plot, tooltip = "Target")
saveWidget(WiT_hem_mouse_alt_volcano_plotly, "WiT_hem_mouse_alt_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_VolcCombo_Hem_Mouse}
fig_WiT_hem_Mouse = ggarrange(WiT_hem_mouse_volcano_plot, WiT_hem_mouse_alt_volcano_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_WiT_hem_Mouse, top = text_grob("Wilcox Test: Hemolysis v. Pooled 55uL", size = 14))
```

\newline

**Figure 15: Volcano plots showing significantly differentially expressed proteins between hemolysis and pooled 55uL groups in mouse data, A) normalized, B) pre-normalization.**

```{r Plot_HemoApts_Mouse}
#Plot intensity of hemoglobin and haptoglobin aptamer for hemolysis and pooled replicates separately
hemapts_mouse_df = hem_mouse_wide_df %>%
                    filter(grepl("hemoglobin", TargetFullName, ignore.case = TRUE)) %>%
                    mutate(Apt_Target = paste(AptName,Target,sep = "_")) %>%
                    rowid_to_column("Table_Key") %>%
                    pivot_longer(names_to = "Group",
                                  values_to = "Log2_Intensity",
                                  cols = starts_with("data")) %>%
                    mutate(Group = ifelse(Group == "data_Hemolysis",
                                          "Hemolysis",
                                          "Pooled 55uL")) %>%
                    rowwise() %>%
                    mutate(Mean_Log2_Intensity = mean(unlist(Log2_Intensity)), SD_Log2_Intensity = sd(unlist(Log2_Intensity)))

hapapts_mouse_df = hem_mouse_wide_df %>%
                    filter(grepl("haptoglobin", TargetFullName, ignore.case = TRUE)) %>%
                    mutate(Apt_Target = paste(AptName,Target,sep = "_")) %>%
                    rowid_to_column("Table_Key") %>%
                    pivot_longer(names_to = "Group",
                                  values_to = "Log2_Intensity",
                                  cols = starts_with("data")) %>%
                    mutate(Group = ifelse(Group == "data_Hemolysis",
                                          "Hemolysis",
                                          "Pooled 55uL")) %>%
                    rowwise() %>%
                    mutate(Mean_Log2_Intensity = mean(unlist(Log2_Intensity)), SD_Log2_Intensity = sd(unlist(Log2_Intensity)))

hemapts_mouse_alt_df = hem_mouse_alt_wide_df %>%
                    filter(grepl("hemoglobin", TargetFullName, ignore.case = TRUE)) %>%
                    mutate(Apt_Target = paste(AptName,Target,sep = "_")) %>%
                    rowid_to_column("Table_Key") %>%
                    pivot_longer(names_to = "Group",
                                  values_to = "Log2_Intensity",
                                  cols = starts_with("data")) %>%
                    mutate(Group = ifelse(Group == "data_Hemolysis",
                                          "Hemolysis",
                                          "Pooled 55uL")) %>%
                    rowwise() %>%
                    mutate(Mean_Log2_Intensity = mean(unlist(Log2_Intensity)), SD_Log2_Intensity = sd(unlist(Log2_Intensity)))

hapapts_mouse_alt_df = hem_mouse_alt_wide_df %>%
                    filter(grepl("haptoglobin", TargetFullName, ignore.case = TRUE)) %>%
                    mutate(Apt_Target = paste(AptName,Target,sep = "_")) %>%
                    rowid_to_column("Table_Key") %>%
                    pivot_longer(names_to = "Group",
                                  values_to = "Log2_Intensity",
                                  cols = starts_with("data")) %>%
                    mutate(Group = ifelse(Group == "data_Hemolysis",
                                          "Hemolysis",
                                          "Pooled 55uL")) %>%
                    rowwise() %>%
                    mutate(Mean_Log2_Intensity = mean(unlist(Log2_Intensity)), SD_Log2_Intensity = sd(unlist(Log2_Intensity)))
                                      
hemoglobin_mouse_plot = hemapts_mouse_df %>% 
  filter(grepl("hemoglobin",TargetFullName,ignore.case=TRUE)) %>%
  ggplot(aes(x = Group, y = Mean_Log2_Intensity, fill = Group)) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = Mean_Log2_Intensity-SD_Log2_Intensity, ymax = Mean_Log2_Intensity+SD_Log2_Intensity), alpha = 0.9) +
  labs(title = "Filtered, Normalized Data",
       x = "",
       y = "Mean Log2(Intensity)") +
  theme_bw() +
  ylim(0,20) +
  theme(legend.position = "bottom") +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 8)) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(axis.text.x = element_blank()) +
  facet_grid(~Table_Key)
#ggsave("hemoglobin_mouse_plot.png")
#hemoglobin_mouse_plot

hemoglobin_mouse_alt_plot = hemapts_mouse_alt_df %>% 
  filter(grepl("hemoglobin",TargetFullName,ignore.case=TRUE)) %>%
  ggplot(aes(x = Group, y = Mean_Log2_Intensity, fill = Group)) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = Mean_Log2_Intensity-SD_Log2_Intensity, ymax = Mean_Log2_Intensity+SD_Log2_Intensity), alpha = 0.9) +
  labs(title = "Filtered, Pre-Normalized Data",
       subtitle = "",
       x = "",
       y = "Mean Log2(Intensity)") +
  theme_bw() +
  ylim(0,20) +
  scale_fill_discrete(labels=c('Red/Pink (Hemolysis)', 'Pooled 55uL')) +
  theme(legend.position = "bottom") +
  rremove("xlab") +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 8)) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(axis.text.x = element_blank()) +
  facet_grid(~Table_Key)
ggsave("hemoglobin_mouse_alt_plot.png", width = 7.29, height = 4.51, units = "in")
#hemoglobin_mouse_alt_plot

haptoglobin_mouse_plot = hapapts_mouse_df %>% 
  filter(grepl("haptoglobin",TargetFullName,ignore.case=TRUE)) %>%
  ggplot(aes(x = Group, y = Mean_Log2_Intensity, fill = Group)) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = Mean_Log2_Intensity-SD_Log2_Intensity, ymax = Mean_Log2_Intensity+SD_Log2_Intensity), alpha = 0.9) +
  labs(title = "Filtered, Normalized Data",
       x = "",
       y = "Mean Log2(Intensity)") +
  theme_bw() +
  ylim(0,20) +
  theme(legend.position = "bottom") +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 8)) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(axis.text.x = element_blank()) +
  facet_grid(~Table_Key)
#ggsave("haptoglobin_mouse_plot.png")
#haptoglobin_mouse_plot

haptoglobin_mouse_alt_plot = hapapts_mouse_alt_df %>% 
  filter(grepl("haptoglobin",TargetFullName,ignore.case=TRUE)) %>%
  ggplot(aes(x = Group, y = Mean_Log2_Intensity, fill = Group)) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = Mean_Log2_Intensity-SD_Log2_Intensity, ymax = Mean_Log2_Intensity+SD_Log2_Intensity), alpha = 0.9) +
  labs(title = "Filtered, Pre-Normalization Data",
       x = "",
       y = "Mean Log2(Intensity)") +
  theme_bw() +
  ylim(0,20) +
  theme(legend.position = "bottom") +
  rremove("xlab") +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 8)) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(axis.text.x = element_blank()) +
  facet_grid(~Table_Key)
#ggsave("haptoglobin_mouse_alt_plot.png")
# haptoglobin_mouse_alt_plot
```

```{r Plot_HemCombo_Mouse}
fig_hemoglobin_Mouse = ggarrange(hemoglobin_mouse_alt_plot,hemoglobin_mouse_plot, labels="AUTO", ncol = 1, nrow = 2, common.legend = TRUE, legend = "bottom")
annotate_figure(fig_hemoglobin_Mouse, top = text_grob("Hemoglobin Intensities: Hemolysis vs. Pooled 55uL", size = 14))
ggsave("fig_hemoglobin_Mouse.png")
```
**Figure 16: Hemoglobin intensity results comparing each hemoglobin-targeting aptamer in the murine suspected hemolysis samples and murine pooled 55L samples**

```{r Table_HemoCombo_Mouse}
#Table of aptamer and target for hemoglobin plot
hemo_table_df = hemapts_mouse_df %>% ungroup() %>% dplyr::select(c(Table_Key,Apt_Target,TargetFullName)) %>% unique()
knitr::kable(hemo_table_df, col.names = c("Table_Key","Aptamer_Target","Target_Full_Name"))
```

**Table 3: Hemoglobin-targeting aptamers with the key used to ID Figure 16 sub-plots with aptamer descriptions.**

```{r AllSamples_Hem_Mouse, include = FALSE}
#Mouse: intensity across all samples for each of the hemoglobin aptamers

#Transform data to wide form for statistical test
mouse_allS_hemo_df = filt_mouse_df %>% filter(eLoD_Flag == TRUE) %>%
                                      filter(SampleType == "Sample" | SampleType == "Calibrator") %>%
                                      filter(AptName %in% hemapts_mouse_df$AptName) %>%
                                      mutate(Group_final = ifelse(endsWith(Sample_ID,"hem"),
                                                                "Hemolysis",
                                                                ifelse(startsWith(Sample_ID, "Mouse A"),
                                                                  "Pooled_55uL",
                                                                  ifelse(startsWith(Sample_ID, "Mouse B"),
                                                                                    "Pooled_55uL_Diluted",
                                                                                    ifelse(startsWith(Sample_ID, "Mouse 35"),
                                                                                           "Pooled_35uL",
                                                                                           ifelse(SampleType == "Calibrator",
                                                                                                  "Calibrator",
                                                                                                  Group_final)))))) %>%
                                      mutate(Log2_Intensity = log(Intensity,2)) %>%
                                      dplyr::select(AptName, Target, TargetFullName, UniProt, Group_final, Sample_ID, Log2_Intensity) %>%
                                      mutate(Type = ifelse(Group_final == "Hemolysis",
                                                                     "Hemolysis",
                                                                     ifelse(Group_final == "Calibrator",
                                                                     "Calibrator",
                                                                     "Other"))) %>%
                                      group_by(Sample_ID, AptName) %>%
                                      mutate(Log2_Intensity = mean(Log2_Intensity)) %>%
                                      unique()

mouse_allS_hemo_plot = mouse_allS_hemo_df %>% 
  filter(!grepl("Pooled",Group_final)) %>% 
  ggplot(aes(x = Sample_ID, y = Log2_Intensity, fill = Type)) +
  geom_bar(stat = "identity") +
  labs(title = "Hemoglobin Aptamers: Hemolysis v. Non-Pooled Samples",
       subtitle = "Filtered, Normalized Data",
       x = "",
       y = "Log2(Intensity)") +
  theme_bw() +
  ylim(0,20) +
  theme(legend.position = "bottom") +
  rremove("xlab") +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 8)) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(axis.text.x = element_blank()) +
  scale_fill_manual(values = c("#00BF7D", "#F8766D", "#00B0F6")) +
  facet_grid(~AptName)
mouse_allS_hemo_plot

mouse_allS_HBAT_plot = mouse_allS_hemo_df %>% 
  filter(!grepl("Pooled",Group_final)) %>% 
  filter(AptName == "seq.7965.25") %>%
  ggplot(aes(x = Sample_ID, y = Log2_Intensity, fill = Type)) +
  geom_bar(stat = "identity") +
  labs(title = "HBAT: Hemolysis v. Non-Pooled Samples",
       subtitle = "Filtered, Normalized Data, Aptamer = seq.7965.25",
       x = "",
       y = "Log2(Intensity)") +
  theme_bw() +
  ylim(0,15) +
  theme(legend.position = "bottom") +
  rremove("xlab") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 8)) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  scale_fill_manual(values = c("#00BF7D", "#F8766D", "#00B0F6")) #+
  # theme(axis.text.x = element_blank())
mouse_allS_HBAT_plot

mouse_allS_HBAZ_plot = mouse_allS_hemo_df %>% 
  filter(!grepl("Pooled",Group_final)) %>% 
  filter(AptName == "seq.6919.3") %>%
  ggplot(aes(x = Sample_ID, y = Log2_Intensity, fill = Type)) +
  geom_bar(stat = "identity") +
  labs(title = "HBAZ: Hemolysis v. Non-Pooled Samples",
       subtitle = "Filtered, Normalized Data, Aptamer = seq.6919.3",
       x = "",
       y = "Log2(Intensity)") +
  theme_bw() +
  ylim(0,15) +
  theme(legend.position = "bottom") +
  rremove("xlab") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 8)) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  scale_fill_manual(values = c("#00BF7D", "#F8766D", "#00B0F6")) #+
  # theme(axis.text.x = element_blank())
mouse_allS_HBAZ_plot
```

```{r AllSamples_Hem_Mouse_Alt, include = FALSE}
#Mouse: intensity across all samples for each of the hemoglobin aptamers

#Transform data to wide form for statistical test
mouse_alt_allS_hemo_df = filt_mouse_alt_df %>% filter(eLoD_Flag == TRUE) %>%
                                      filter(SampleType == "Sample" | SampleType == "Calibrator") %>%
                                      filter(AptName %in% hemapts_mouse_df$AptName) %>%
                                      mutate(Group_final = ifelse(endsWith(Sample_ID,"hem"),
                                                                "Hemolysis",
                                                                ifelse(startsWith(Sample_ID, "Mouse A"),
                                                                  "Pooled_55uL",
                                                                  ifelse(startsWith(Sample_ID, "Mouse B"),
                                                                                    "Pooled_55uL_Diluted",
                                                                                    ifelse(startsWith(Sample_ID, "Mouse 35"),
                                                                                           "Pooled_35uL",
                                                                                           ifelse(SampleType == "Calibrator",
                                                                                                  "Calibrator",
                                                                                                  Group_final)))))) %>%
                                      mutate(Log2_Intensity = log(Intensity,2)) %>%
                                      dplyr::select(AptName, Target, TargetFullName, UniProt, Group_final, Sample_ID, Log2_Intensity) %>%
                                      mutate(Type = ifelse(Group_final == "Hemolysis",
                                                                     "Hemolysis",
                                                                     ifelse(Group_final == "Calibrator",
                                                                     "Calibrator",
                                                                     "Other"))) %>%
                                      group_by(Sample_ID, AptName) %>%
                                      mutate(Log2_Intensity = mean(Log2_Intensity)) %>%
                                      unique()

mouse_alt_allS_hemo_plot = mouse_alt_allS_hemo_df %>% 
  filter(!grepl("Pooled",Group_final)) %>% 
  ggplot(aes(x = Sample_ID, y = Log2_Intensity, fill = Type)) +
  geom_bar(stat = "identity") +
  labs(title = "Hemoglobin Aptamers: Hemolysis v. Non-Pooled Samples",
       subtitle = "Filtered, Pre-Normalization Data",
       x = "",
       y = "Log2(Intensity)") +
  theme_bw() +
  ylim(0,20) +
  theme(legend.position = "bottom") +
  rremove("xlab") +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 8)) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(axis.text.x = element_blank()) +
  scale_fill_manual(values = c("#00BF7D", "#F8766D", "#00B0F6")) +
  facet_grid(~AptName)
mouse_alt_allS_hemo_plot

mouse_alt_allS_HBD_plot = mouse_alt_allS_hemo_df %>%
  filter(!grepl("Pooled",Group_final)) %>%
  filter(AptName == "seq.6992.67") %>%
  ggplot(aes(x = Sample_ID, y = Log2_Intensity, fill = Type)) +
  geom_bar(stat = "identity") +
  labs(title = "HBD: Hemolysis v. Non-Pooled Samples",
       subtitle = "Filtered, Normalized Data, Aptamer = seq.6992.67",
       x = "",
       y = "Log2(Intensity)") +
  theme_bw() +
  ylim(0,15) +
  theme(legend.position = "bottom") +
  rremove("xlab") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 8)) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  scale_fill_manual(values = c("#00BF7D", "#F8766D", "#00B0F6")) #+
  # theme(axis.text.x = element_blank())
mouse_alt_allS_HBD_plot

mouse_alt_allS_AHSP_plot = mouse_alt_allS_hemo_df %>%
  filter(!grepl("Pooled",Group_final)) %>%
  filter(AptName == "seq.9025.5") %>%
  ggplot(aes(x = Sample_ID, y = Log2_Intensity, fill = Type)) +
  geom_bar(stat = "identity") +
  labs(title = "AHSP: Hemolysis v. Non-Pooled Samples",
       subtitle = "Filtered, Normalized Data, Aptamer = seq.seq.9025.5",
       x = "",
       y = "Log2(Intensity)") +
  theme_bw() +
  ylim(0,15) +
  theme(legend.position = "bottom") +
  rremove("xlab") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 8)) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  scale_fill_manual(values = c("#00BF7D", "#F8766D", "#00B0F6")) #+
  # theme(axis.text.x = element_blank())
mouse_alt_allS_AHSP_plot
```

```{r Plot_HapCombo_Mouse, include = FALSE}
fig_haptoglobin_Mouse = ggarrange(haptoglobin_mouse_plot, haptoglobin_mouse_alt_plot, labels="AUTO", ncol = 1, nrow = 2, common.legend = TRUE, legend = "bottom")
annotate_figure(fig_haptoglobin_Mouse, top = text_grob("Haptoglobin Intensities: Hemolysis vs. Pooled 55uL", size = 14))
ggsave("fig_haptoglobin_Mouse.png")
```

```{r Table_HaptoCombo_Mouse, include = FALSE}
#Table of aptamer and target for hemoglobin plot
hapto_table_df = hapapts_mouse_df %>% ungroup() %>% dplyr::select(c(Table_Key,Apt_Target,TargetFullName)) %>% unique()
knitr::kable(hapto_table_df, col.names = c("Table_Key","Aptamer_Target","Target_Full_Name"))
```

#### KMC-PDAC v. KP Late:

  *Reveals differentially abundant proteins in autochthonous PDAC model*

```{r Stat_DataTransform_DE1_Mouse}
#View group annos for comparison
groups_mouse_df = filt_mouse_df %>% filter(!is.na(Group_final)) %>% dplyr::group_by(Group_final) %>% summarise(Group_final, Condition, Group, Subgroup) %>% unique()

#Mouse DE1: KMC-PDAC v. KP Late

#Transform data to wide form for statistical test
DE1_mouse_wide_df = filt_mouse_df %>% filter(!is.na(Group_final)) %>%
                                      filter(eLoD_Flag == TRUE) %>%
                                      filter(Group_final == "Healthy control - late" | Group_final == "KMC - PDAC") %>%
                                      mutate(Group_final = ifelse(Group_final == "Healthy control - late",
                                                                "Healthy_Control_Late/KP_Late",
                                                                "KMC_PDAC")) %>%
                                      dplyr::group_by(AptName,Group_final) %>%
                                      mutate(Log2_Intensity = log(Intensity,2)) %>%
                                      summarize(data = list(Log2_Intensity), count = n(), .groups = "drop") %>%
                                      pivot_wider(names_from = Group_final, values_from = c(data,count)) %>%
                                      filter(`count_Healthy_Control_Late/KP_Late` > 1 & count_KMC_PDAC > 1)
```

```{r Stat_DataTransform_DE1_Mouse_Alt}
#Mouse Alt DE1: KMC-PDAC v. KP Late

#Transform data to wide form for statistical test
DE1_mouse_alt_wide_df = filt_mouse_alt_df %>% filter(!is.na(Group_final)) %>%
                                              filter(eLoD_Flag == TRUE) %>%
                                              filter(Group_final == "Healthy control - late" | Group_final == "KMC - PDAC") %>%
                                              mutate(Group_final = ifelse(Group_final == "Healthy control - late",
                                                                        "Healthy_Control_Late/KP_Late",
                                                                        "KMC_PDAC")) %>%
                                              dplyr::group_by(AptName,Group_final) %>%
                                              mutate(Log2_Intensity = log(Intensity,2)) %>%
                                              summarize(data = list(Log2_Intensity), count = n(), .groups = "drop") %>%
                                              pivot_wider(names_from = Group_final, values_from = c(data,count)) %>%
                                              filter(`count_Healthy_Control_Late/KP_Late` > 1 & count_KMC_PDAC > 1)
```

```{r WiT_DE1_Mouse}
#Wilcox test - store pvalue and difference
WiT_DE1_mouse_df = DE1_mouse_wide_df %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_KMC_PDAC), unlist(`data_Healthy_Control_Late/KP_Late`))$p.value) %>%
  mutate(Diff = (median(unlist(data_KMC_PDAC)) - median(unlist(`data_Healthy_Control_Late/KP_Late`)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE1_mouse_df = merge(WiT_DE1_mouse_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE1_mouse_df = WiT_DE1_mouse_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r WiT_DE1_Mouse_Alt}
#Wilcox test - store pvalue and difference
WiT_DE1_mouse_alt_df = DE1_mouse_alt_wide_df %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_KMC_PDAC), unlist(`data_Healthy_Control_Late/KP_Late`))$p.value) %>%
  mutate(Diff = (median(unlist(data_KMC_PDAC)) - median(unlist(`data_Healthy_Control_Late/KP_Late`)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE1_mouse_alt_df = merge(WiT_DE1_mouse_alt_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE1_mouse_alt_df = WiT_DE1_mouse_alt_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r Plot_WiT_pval_DE1_Mouse}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE1_mouse_plot = WiT_DE1_mouse_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "Aptamer Count",
       title = "Mouse",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE1_mouse_plot.png")
```

```{r Plot_WiT_pval_DE1_Mouse_Alt}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE1_mouse_alt_plot = WiT_DE1_mouse_alt_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "",
       title = "Mouse (Pre-Normalization)",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE1_mouse_alt_plot.png")
```

```{r Plot_PvalCombo_DE1_Mouse}
fig_pval_DE1_Mouse = ggarrange(WiT_pval_dist_DE1_mouse_plot, WiT_pval_dist_DE1_mouse_alt_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_pval_DE1_Mouse, top = text_grob("P-Value Distribution (Wilcox Test): KMC-PDAC v. Healthy Control - Late/KP Late", size = 14))
```
\newline

**Figure 17: P-value distributions for the wilcox tests performed on the mouse data comparing intensity (RFU) values between aptamers in the KMC-PDAC and Healthy Control - Late/KP Late groups. A) Normalized mouse data, B) Pre-normalization mouse data.**

```{r Plot_WiT_volc_DE1_Mouse}
#Wilcox volcano Plot
WiT_DE1_mouse_volcano_plot = WiT_DE1_mouse_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Mouse",
       x = "Diff(KMC-PDAC - KP-Late)",
       y = "-log10(p-value)",
       subtitle = "") +
  theme_bw() +
  xlim(-10,10) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_DE1_mouse_volcano_plot.png")

WiT_DE1_mouse_volcano_plotly = ggplotly(WiT_DE1_mouse_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE1_mouse_volcano_plotly, "WiT_DE1_mouse_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_WiT_volc_DE1_Mouse_Alt}
#Wilcox volcano Plot
WiT_DE1_mouse_alt_volcano_plot = WiT_DE1_mouse_alt_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 150) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Mouse (Pre-Normalization)",
       x = "Diff(KMC-PDAC - KP-Late)",
       y = "",
       subtitle = "") +
  theme_bw() +
  xlim(-10,10) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_DE1_mouse_alt_volcano_plot.png")

WiT_DE1_mouse_alt_volcano_plotly = ggplotly(WiT_DE1_mouse_alt_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE1_mouse_alt_volcano_plotly, "WiT_DE1_mouse_alt_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_VolcCombo_DE1_Mouse}
fig_WiT_DE1_Mouse = ggarrange(WiT_DE1_mouse_volcano_plot, WiT_DE1_mouse_alt_volcano_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_WiT_DE1_Mouse, top = text_grob("Wilcox Test: KMC-PDAC v. Healthy Control - Late/KP Late", size = 14))
```

\newline

**Figure 18: Volcano plots showing significantly differentially expressed proteins between KMC - PDAC and Healthy Control - Late/KP Late groups in mouse data, A) normalized, B) pre-normalization.**

#### KPC-Lung v. KP Late:

  *Reveals differentially abundant proteins in autochthonous lung cancer model*
  
```{r Stat_DataTransform_DE2_Mouse}
#Mouse DE2: KMC-PDAC v. KP Late

#Transform data to wide form for statistical test
DE2_mouse_wide_df = filt_mouse_df %>% filter(!is.na(Group_final)) %>%
                                      filter(eLoD_Flag == TRUE) %>%
                                      filter(Group_final == "KPC - lung" | Group_final == "Healthy control - late") %>%
                                      mutate(Group_final = ifelse(Group_final == "Healthy control - late",
                                                                "Healthy_Control_Late/KP_Late",
                                                                "KPC_Lung")) %>%
                                      dplyr::group_by(AptName,Group_final) %>%
                                      mutate(Log2_Intensity = log(Intensity,2)) %>%
                                      summarize(data = list(Log2_Intensity), count = n(), .groups = "drop") %>%
                                      pivot_wider(names_from = Group_final, values_from = c(data,count)) %>%
                                      filter(`count_Healthy_Control_Late/KP_Late` > 1 & count_KPC_Lung > 1)
```

```{r Stat_DataTransform_DE2_Mouse_Alt}
#Mouse Alt DE2: KMC-PDAC v. KP Late

#Transform data to wide form for statistical test
DE2_mouse_alt_wide_df = filt_mouse_alt_df %>% filter(!is.na(Group_final)) %>%
                                              filter(eLoD_Flag == TRUE) %>%
                                              filter(Group_final == "KPC - lung" | Group_final == "Healthy control - late") %>%
                                              mutate(Group_final = ifelse(Group_final == "Healthy control - late",
                                                                        "Healthy_Control_Late/KP_Late",
                                                                        "KPC_Lung")) %>%
                                              dplyr::group_by(AptName,Group_final) %>%
                                              mutate(Log2_Intensity = log(Intensity,2)) %>%
                                              summarize(data = list(Log2_Intensity), count = n(), .groups = "drop") %>%
                                              pivot_wider(names_from = Group_final, values_from = c(data,count)) %>%
                                              filter(`count_Healthy_Control_Late/KP_Late` > 1 & count_KPC_Lung > 1)
```

```{r WiT_DE2_Mouse}
#Wilcox test - store pvalue and difference
WiT_DE2_mouse_df = DE2_mouse_wide_df %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(`data_KPC_Lung`), unlist(`data_Healthy_Control_Late/KP_Late`))$p.value) %>%
  mutate(Diff = (median(unlist(data_KPC_Lung)) - median(unlist(`data_Healthy_Control_Late/KP_Late`)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE2_mouse_df = merge(WiT_DE2_mouse_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE2_mouse_df = WiT_DE2_mouse_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r WiT_DE2_Mouse_Alt}
#Wilcox test - store pvalue and difference
WiT_DE2_mouse_alt_df = DE2_mouse_alt_wide_df %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(`data_KPC_Lung`), unlist(`data_Healthy_Control_Late/KP_Late`))$p.value) %>%
  mutate(Diff = (median(unlist(data_KPC_Lung)) - median(unlist(`data_Healthy_Control_Late/KP_Late`)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE2_mouse_alt_df = merge(WiT_DE2_mouse_alt_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE2_mouse_alt_df = WiT_DE2_mouse_alt_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r Plot_WiT_pval_DE2_Mouse}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE2_mouse_plot = WiT_DE2_mouse_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "Aptamer Count",
       title = "Mouse",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE2_mouse_plot.png")
```

```{r Plot_WiT_pval_DE2_Mouse_Alt}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE2_mouse_alt_plot = WiT_DE2_mouse_alt_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "",
       title = "Mouse (Pre-Normalization)",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE2_mouse_alt_plot.png")
```

```{r Plot_PvalCombo_DE2_Mouse}
fig_pval_DE2_Mouse = ggarrange(WiT_pval_dist_DE2_mouse_plot, WiT_pval_dist_DE2_mouse_alt_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_pval_DE2_Mouse, top = text_grob("P-Value Distribution (Wilcox Test): KP-Lung v. Healthy Control - Late/KP Late", size = 14))
```

\newline

**Figure 19: P-value distributions for the Wilcox tests performed on the mouse data comparing intensity (RFU) values between proteins in the KPC-Lung and KP Late groups. A) Normalized mouse data, B) Pre-normalization mouse data.**

```{r Plot_WiT_volc_DE2_Mouse}
#Wilcoxon volcano Plot
WiT_DE2_mouse_volcano_plot = WiT_DE2_mouse_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Mouse",
       x = "Diff(KPC-Lung - KP Late)",
       y = "-log10(p-value)",
       subtitle = "") +
  theme_bw() +
  xlim(-20,20) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_DE2_mouse_volcano_plot.png")

WiT_DE2_mouse_volcano_plotly = ggplotly(WiT_DE2_mouse_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE2_mouse_volcano_plotly, "WiT_DE2_mouse_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_WiT_volc_DE2_Mouse_Alt}
#Wilcoxon volcano Plot
WiT_DE2_mouse_alt_volcano_plot = WiT_DE2_mouse_alt_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Mouse (Pre-Normalization)",
       x = "Diff(KPC-Lung - KP Late)",
       y = "",
       subtitle = "") +
  theme_bw() +
  xlim(-20,20) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_DE2_mouse_alt_volcano_plot.png")

WiT_DE2_mouse_alt_volcano_plotly = ggplotly(WiT_DE2_mouse_alt_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE2_mouse_alt_volcano_plotly, "WiT_DE2_mouse_alt_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_VolcCombo_DE2_Mouse}}
fig_WiT_DE2_Mouse = ggarrange(WiT_DE2_mouse_volcano_plot, WiT_DE2_mouse_alt_volcano_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_WiT_DE2_Mouse, top = text_grob("Wilcox Test: KP-Lung v. Healthy Control - Late/KP Late", size = 14))
```

\newline

**Figure 20: Volcano plots showing significantly differentially expressed proteins between KPC-Lung and Healthy Control - Late/KP Late groups in mouse data, A) normalized, B) pre-normalization.**

#### KMC v. KMC-Control:

  *Reveals differentially abundant proteins in autochthonous Myc-driven PDAC model*
  
```{r Stat_DataTransform_DE3_Mouse}
#Mouse DE3: KMC v. KMC-Control

#Transform data to wide form for statistical test
DE3_mouse_wide_df = filt_mouse_df %>% filter(!is.na(Group_final)) %>%
                                      filter(eLoD_Flag == TRUE) %>%
                                      filter(Group_final == "KMC - PDAC" | Group_final == "KMC_control") %>%
                                      mutate(Group_final = ifelse(Group_final == "KMC - PDAC",
                                                                "KMC_PDAC",
                                                                Group_final)) %>%
                                      dplyr::group_by(AptName,Group_final) %>%
                                      mutate(Log2_Intensity = log(Intensity,2)) %>%
                                      summarize(data = list(Log2_Intensity), count = n(), .groups = "drop") %>%
                                      pivot_wider(names_from = Group_final, values_from = c(data,count)) %>%
                                      filter(count_KMC_PDAC > 1 & count_KMC_control > 1)
```

```{r Stat_DataTransform_DE3_Mouse_Alt}
#Mouse Alt DE3: KMC v. KMC-Control

#Transform data to wide form for statistical test
DE3_mouse_alt_wide_df = filt_mouse_alt_df %>% filter(!is.na(Group_final)) %>%
                                              filter(eLoD_Flag == TRUE) %>%
                                              filter(Group_final == "KMC - PDAC" | Group_final == "KMC_control") %>%
                                              mutate(Group_final = ifelse(Group_final == "KMC - PDAC",
                                                                        "KMC_PDAC",
                                                                        Group_final)) %>%
                                              dplyr::group_by(AptName,Group_final) %>%
                                              mutate(Log2_Intensity = log(Intensity,2)) %>%
                                              summarize(data = list(Log2_Intensity), count = n(), .groups = "drop") %>%
                                              pivot_wider(names_from = Group_final, values_from = c(data,count)) %>%
                                              filter(count_KMC_PDAC > 1 & count_KMC_control > 1)
```

```{r WiT_DE3_Mouse}
#Wilcox test - store pvalue and difference
WiT_DE3_mouse_df = DE3_mouse_wide_df %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_KMC_PDAC), unlist(data_KMC_control))$p.value) %>%
  mutate(Diff = (median(unlist(data_KMC_PDAC)) - median(unlist(data_KMC_control)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE3_mouse_df = merge(WiT_DE3_mouse_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE3_mouse_df = WiT_DE3_mouse_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r WiT_DE3_Mouse_Alt}
#Wilcox test - store pvalue and difference
WiT_DE3_mouse_alt_df = DE3_mouse_alt_wide_df %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_KMC_PDAC), unlist(data_KMC_control))$p.value) %>%
  mutate(Diff = (median(unlist(data_KMC_PDAC)) - median(unlist(data_KMC_control)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE3_mouse_alt_df = merge(WiT_DE3_mouse_alt_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE3_mouse_alt_df = WiT_DE3_mouse_alt_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r Plot_WiT_pval_DE3_Mouse}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE3_mouse_plot = WiT_DE3_mouse_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "Aptamer Count",
       title = "Mouse",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE3_mouse_plot.png")
```

```{r Plot_WiT_pval_DE3_Mouse_Alt}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE3_mouse_alt_plot = WiT_DE3_mouse_alt_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "",
       title = "Mouse (Pre-Normalization)",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE3_mouse_alt_plot.png")
```

```{r Plot_WiTPvalCombo_DE3_Mouse}
fig_pval_DE3_Mouse = ggarrange(WiT_pval_dist_DE3_mouse_plot, WiT_pval_dist_DE3_mouse_alt_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_pval_DE3_Mouse, top = text_grob("P-Value Distribution (Wilcox Test): KMC-PDAC v. KMC Control", size = 14))
```

\newline

**Figure 21: P-value distributions for the Wilcox tests performed on the mouse data comparing intensity (RFU) values between proteins in the KMC-PDAC and KMC Control groups. A) Normalized mouse data, B) Pre-normalization mouse data.**

```{r Plot_WiT_volc_DE3_Mouse}
#Wilcoxon volcano Plot
WiT_DE3_mouse_volcano_plot = WiT_DE3_mouse_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed", color = "red") +
  geom_vline(xintercept = 2, linetype = "dashed", color = "red") +
  geom_vline(xintercept = -2, linetype = "dashed", color = "red") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Mouse (Normalized)",
       x = "Log2(KMC-PDAC-KMC_Control)",
       y = "") +
  theme_bw() +
  xlim(-5,5) +
  theme(axis.text.x = element_text(vjust = 0.5, size = 14)) +
  theme(axis.text.y = element_text(vjust = 0.5, size = 14)) +
  theme(axis.title = element_text(vjust = 0.5, size = 14)) +
  theme(title = element_text(size = 14)) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
ggsave("WiT_DE3_mouse_volcano_plot.png")

WiT_DE3_mouse_volcano_plotly = ggplotly(WiT_DE3_mouse_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE3_mouse_volcano_plotly, "WiT_DE3_mouse_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_WiT_volc_DE3_Mouse_Alt}
#Wilcoxon volcano Plot
WiT_DE3_mouse_alt_volcano_plot = WiT_DE3_mouse_alt_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed", color = "red") +
  geom_vline(xintercept = 2, linetype = "dashed", color = "red") +
  geom_vline(xintercept = -2, linetype = "dashed", color = "red") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Mouse (Pre-Normalization)",
       x = "Log2(KMC-PDAC-KMC_Control)",
       y = "-log10(p-value)") +
  theme_bw() +
  theme(axis.text.x = element_text(vjust = 0.5, size = 14)) +
  theme(axis.text.y = element_text(vjust = 0.5, size = 14)) +
  theme(axis.title = element_text(vjust = 0.5, size = 14)) +
  theme(title = element_text(size = 14)) +
  xlim(-5,5) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_DE3_mouse_alt_volcano_plot.png")

WiT_DE3_mouse_alt_volcano_plotly = ggplotly(WiT_DE3_mouse_alt_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE3_mouse_alt_volcano_plotly, "WiT_DE3_mouse_alt_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_WiTVolcCombo_DE3_Mouse}
fig_WiT_DE3_Mouse = ggarrange(WiT_DE3_mouse_alt_volcano_plot, WiT_DE3_mouse_volcano_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_WiT_DE3_Mouse, top = text_grob("Wilcox Test: KMC-PDAC v. KMC Control", size = 18))
ggsave("fig_WiT_DE3_Mouse.png")
```

\newline

**Figure 22: Volcano plots showing significantly differentially expressed proteins between KMC-PDAC and KMC Control groups in mouse data, A) normalized, B) pre-normalization.**

```{r TT_DE3_Mouse, include = FALSE}
#t-test - store pvalue and difference
# TT_DE3_mouse_df = DE3_mouse_wide_df %>%
#   rowwise() %>%
#   mutate(tt_pval = t.test(unlist(data_KMC_PDAC), unlist(data_KMC_control), var.equal = FALSE)$p.value) %>%
#   mutate(Diff = (median(unlist(data_KMC_PDAC)) - median(unlist(data_KMC_control)))) %>%
#   ungroup() %>%
#   mutate(tt_pval_BH = p.adjust(tt_pval, method = "BH"))
# 
# TT_DE3_mouse_df = merge(TT_DE3_mouse_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)
# 
# TT_DE3_mouse_df = TT_DE3_mouse_df %>%
#   mutate(p_val_sig = ifelse(tt_pval_BH < 0.05, "Significant", "Non"),
#          proteins_sig = ifelse(tt_pval_BH < 0.05, Target, ""),
#   )
```

```{r TT_DE3_Mouse_Alt, include = FALSE}
#t-test - store pvalue and difference
# TT_DE3_mouse_alt_df = DE3_mouse_alt_wide_df %>%
#   rowwise() %>%
#   mutate(tt_pval = t.test(unlist(data_KMC_PDAC), unlist(data_KMC_control), var.equal = FALSE)$p.value) %>%
#   mutate(Diff = (median(unlist(data_KMC_PDAC)) - median(unlist(data_KMC_control)))) %>%
#   ungroup() %>%
#   mutate(tt_pval_BH = p.adjust(tt_pval, method = "BH"))
# 
# TT_DE3_mouse_alt_df = merge(TT_DE3_mouse_alt_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)
# 
# TT_DE3_mouse_alt_df = TT_DE3_mouse_alt_df %>%
#   mutate(p_val_sig = ifelse(tt_pval_BH < 0.05, "Significant", "Non"),
#          proteins_sig = ifelse(tt_pval_BH < 0.05, Target, ""),
#   )
```

```{r Plot_TT_pval_DE3_Mouse, include = FALSE}
#Plot WiT pvalue distribution - before multiple testing correction
# TT_pval_dist_DE3_mouse_plot = TT_DE3_mouse_df %>%
#   ggplot(aes(tt_pval)) +
#   theme_bw() +
#   theme(panel.grid = element_blank()) +
#   geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
#   labs(x = "p-value",
#        y = "Aptamer Count",
#        title = "Mouse",
#        subtitle = "") +
#   theme(plot.subtitle = element_text(face = "italic"))
#ggsave("TT_pval_dist_DE3_mouse_plot.png")
```

```{r Plot_TT_pval_DE3_Mouse_Alt, include = FALSE}
#Plot WiT pvalue distribution - before multiple testing correction
# TT_pval_dist_DE3_mouse_alt_plot = TT_DE3_mouse_alt_df %>%
#   ggplot(aes(tt_pval)) +
#   theme_bw() +
#   theme(panel.grid = element_blank()) +
#   geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
#   labs(x = "p-value",
#        y = "",
#        title = "Mouse (Pre-Normalization)",
#        subtitle = "") +
#   theme(plot.subtitle = element_text(face = "italic"))
#ggsave("TT_pval_dist_DE3_mouse_alt_plot.png")
```

```{r Plot_TTPvalCombo_DE3_Mouse, include = FALSE}
# fig_ttpval_DE3_Mouse = ggarrange(TT_pval_dist_DE3_mouse_plot, TT_pval_dist_DE3_mouse_alt_plot, labels="AUTO", ncol = 2, nrow = 1)
# annotate_figure(fig_ttpval_DE3_Mouse, top = text_grob("P-Value Distribution (T-Test): KMC-PDAC v. KMC Control", size = 14))

# **Figure X: P-value distributions for the t-tests performed on the mouse data comparing intensity (RFU) values between proteins in the KMC-PDAC and KMC Control groups. A) Normalized mouse data, B) Pre-normalization mouse data.**
```

```{r Plot_TT_volc_DE3_Mouse, include = FALSE}
#t-test volcano Plot
# TT_DE3_mouse_volcano_plot = TT_DE3_mouse_df %>%
#   ggplot(aes(x = Diff, y = -log(tt_pval_BH, 10), group = p_val_sig)) +
#   geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
#   geom_hline(yintercept = -log(0.05, 10), linetype = "dashed", color = "red") +
#   geom_vline(xintercept = 2, linetype = "dashed", color = "red") +
#   geom_vline(xintercept = -2, linetype = "dashed", color = "red") +
#   geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
#   scale_color_manual(values = c("black", "red")) +
#   labs(title = "Mouse",
#        x = "Log2(KMC-PDAC-KMC_control)",
#        y = "-log10(p-value)",
#        subtitle = "") +
#   theme_bw() +
#   xlim(-5,5) +
#   theme(legend.position = "none") +
#   theme(plot.subtitle = element_text(face = "italic"))
# #ggsave("TT_DE3_mouse_volcano_plot.png")
# 
# TT_DE3_mouse_volcano_plotly = ggplotly(TT_DE3_mouse_volcano_plot, tooltip = "Target")
# saveWidget(TT_DE3_mouse_volcano_plotly, "TT_DE3_mouse_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_TT_volc_DE3_Mouse_Alt, include = FALSE}
#t-test volcano Plot
# TT_DE3_mouse_alt_volcano_plot = TT_DE3_mouse_alt_df %>%
#   ggplot(aes(x = Diff, y = -log(tt_pval_BH, 10), group = p_val_sig)) +
#   geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
#   geom_hline(yintercept = -log(0.05, 10), linetype = "dashed", color = "red") +
#   geom_vline(xintercept = 2, linetype = "dashed", color = "red") +
#   geom_vline(xintercept = -2, linetype = "dashed", color = "red") +
#   geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
#   scale_color_manual(values = c("black", "red")) +
#   labs(title = "Mouse (Pre-Normalization)",
#        x = "Log2(KMC-PDAC-KMC_Control)",
#        y = "",
#        subtitle = "") +
#   theme_bw() +
#   xlim(-5,5) +
#   theme(legend.position = "none") +
#   theme(plot.subtitle = element_text(face = "italic"))
# #ggsave("TT_DE3_mouse_alt_volcano_plot.png")
# 
# TT_DE3_mouse_alt_volcano_plotly = ggplotly(TT_DE3_mouse_alt_volcano_plot, tooltip = "Target")
# saveWidget(TT_DE3_mouse_alt_volcano_plotly, "TT_DE3_mouse_alt_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_TTVolcCombo_DE3_Mouse, include = FALSE}
# fig_TT_DE3_Mouse = ggarrange(TT_DE3_mouse_volcano_plot, TT_DE3_mouse_alt_volcano_plot, labels="AUTO", ncol = 2, nrow = 1)
# annotate_figure(fig_TT_DE3_Mouse, top = text_grob("T-Test: KMC-PDAC v. KMC Control", size = 14))

# **Figure X: Volcano plots of t-test results showing significantly differentially expressed proteins between KMC-PDAC and KMC Control groups in mouse data, A) normalized, B) pre-normalization.**
```

#### KPC-Early v. KP Early:

  *Reveals differentially abundant proteins in early lethal PanIN model*
  
```{r Stat_DataTransform_DE4_Mouse}
#Mouse DE4: KPC-Early v. KP Early

#Transform data to wide form for statistical test
DE4_mouse_wide_df = filt_mouse_df %>% filter(!is.na(Group_final)) %>%
                                      filter(eLoD_Flag == TRUE) %>%
                                      filter(Group_final == "Lethal PanIN - early" | Group_final == "Healthy control - early") %>%
                                      mutate(Condition = ifelse(Group_final == "Lethal PanIN - early",
                                                                "KPC_early",
                                                                "KP_early")) %>%
                                      dplyr::group_by(AptName,Condition) %>%
                                      mutate(Log2_Intensity = log(Intensity,2)) %>%
                                      summarize(data = list(Log2_Intensity), count = n(), .groups = "drop") %>%
                                      pivot_wider(names_from = Condition, values_from = c(data,count)) %>%
                                      filter(count_KPC_early > 1 & count_KP_early > 1)
```

```{r Stat_DataTransform_DE4_Mouse_Alt}
#Mouse Alt DE4: KPC-Early v. KP Early

#Transform data to wide form for statistical test
DE4_mouse_alt_wide_df = filt_mouse_alt_df %>% filter(!is.na(Group_final)) %>%
                                              filter(eLoD_Flag == TRUE) %>%
                                              filter(Group_final == "Lethal PanIN - early" | Group_final == "Healthy control - early") %>%
                                              mutate(Condition = ifelse(Group_final == "Lethal PanIN - early",
                                                                        "KPC_early",
                                                                        "KP_early")) %>%
                                              dplyr::group_by(AptName,Condition) %>%
                                              mutate(Log2_Intensity = log(Intensity,2)) %>%
                                              summarize(data = list(Log2_Intensity), count = n(), .groups = "drop") %>%
                                              pivot_wider(names_from = Condition, values_from = c(data,count)) %>%
                                              filter(count_KPC_early > 1 & count_KP_early > 1)
```

```{r WiT_DE4_Mouse}
#Wilcox test - store pvalue and difference
WiT_DE4_mouse_df = DE4_mouse_wide_df %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_KPC_early), unlist(data_KP_early))$p.value) %>%
  mutate(Diff = (median(unlist(data_KPC_early)) - median(unlist(data_KP_early)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE4_mouse_df = merge(WiT_DE4_mouse_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE4_mouse_df = WiT_DE4_mouse_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r WiT_DE4_Mouse_Alt}
#Wilcox test - store pvalue and difference
WiT_DE4_mouse_alt_df = DE4_mouse_alt_wide_df %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_KPC_early), unlist(data_KP_early))$p.value) %>%
  mutate(Diff = (median(unlist(data_KPC_early)) - median(unlist(data_KP_early)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE4_mouse_alt_df = merge(WiT_DE4_mouse_alt_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE4_mouse_alt_df = WiT_DE4_mouse_alt_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r Plot_WiT_pval_DE4_Mouse}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE4_mouse_plot = WiT_DE4_mouse_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "Aptamer Count",
       title = "Mouse",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE4_mouse_plot.png")
```

```{r Plot_WiT_pval_DE4_Mouse_Alt}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE4_mouse_alt_plot = WiT_DE4_mouse_alt_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "",
       title = "Mouse (Pre-Normalization)",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE4_mouse_alt_plot.png")
```

```{r Plot_PvalCombo_DE4_Mouse}
fig_pval_DE4_Mouse = ggarrange(WiT_pval_dist_DE4_mouse_plot, WiT_pval_dist_DE4_mouse_alt_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_pval_DE4_Mouse, top = text_grob("P-Value Distribution (Wilcox Test): KPC-Early v. Healthy Control Early/KP Early", size = 14))
```

\newline

**Figure 23: P-value distributions for the Wilcox tests performed on the mouse data comparing intensity (RFU) values between proteins in the KPC - Early and Healthy control - early/KP-early groups. A) Normalized mouse data, B) Pre-normalization mouse data.**

```{r Plot_WiT_volc_DE4_Mouse}
#Wilcoxon volcano Plot
WiT_DE4_mouse_volcano_plot = WiT_DE4_mouse_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Mouse",
       x = "Diff(KPC_early - KP_early)",
       y = "-log10(p-value)",
       subtitle = "") +
  theme_bw() +
  xlim(-20,20) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_DE4_mouse_volcano_plot.png")

WiT_DE4_mouse_volcano_plotly = ggplotly(WiT_DE4_mouse_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE4_mouse_volcano_plotly, "WiT_DE4_mouse_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_WiT_volc_DE4_Mouse_Alt}
#Wilcoxon volcano Plot
WiT_DE4_mouse_alt_volcano_plot = WiT_DE4_mouse_alt_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Mouse (Pre-Normalization)",
       x = "Diff(KPC_early - KP_early)",
       y = "",
       subtitle = "") +
  theme_bw() +
  xlim(-20,20) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_DE4_mouse_alt_volcano_plot.png")

WiT_DE4_mouse_alt_volcano_plotly = ggplotly(WiT_DE4_mouse_alt_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE4_mouse_alt_volcano_plotly, "WiT_DE4_mouse_alt_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_VolcCombo_DE4_Mouse}
fig_WiT_DE4_Mouse = ggarrange(WiT_DE4_mouse_volcano_plot, WiT_DE4_mouse_alt_volcano_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_WiT_DE4_Mouse, top = text_grob("Wilcox Test: KPC-Early v. KP-Early", size = 14))
```  

\newline

**Figure 24: Volcano plots showing significantly differentially expressed proteins between KPC - early and Healthy control - early/KP - early groups in mouse data, A) normalized, B) pre-normalization.**
  
#### KPC-Late (Lethal PanIN  Late) v. KP Late:

  *Reveals differentially abundant proteins in late lethal PanIN model*
  
```{r Stat_DataTransform_DE5_Mouse}
#Mouse DE5: KPC-Late v. KP Late

#Transform data to wide form for statistical test
DE5_mouse_wide_df = filt_mouse_df %>% filter(!is.na(Group_final)) %>%
                                      filter(eLoD_Flag == TRUE) %>%
                                      filter(Group_final == "Lethal PanIN - late" | Group_final == "Healthy control - late") %>%
                                      mutate(Condition = ifelse(Group_final == "Lethal PanIN - late",
                                                                "KPC_late",
                                                                "KP_late")) %>%
                                      dplyr::group_by(AptName,Condition) %>%
                                      mutate(Log2_Intensity = log(Intensity,2)) %>%
                                      summarize(data = list(Log2_Intensity), count = n(), .groups = "drop") %>%
                                      pivot_wider(names_from = Condition, values_from = c(data,count)) %>%
                                      filter(count_KPC_late > 1 & count_KP_late > 1)
```

```{r Stat_DataTransform_DE5_Mouse_Alt}
#Mouse Alt DE5: KPC-Late v. KP Late

#Transform data to wide form for statistical test
DE5_mouse_alt_wide_df = filt_mouse_alt_df %>% filter(!is.na(Group_final)) %>%
                                              filter(eLoD_Flag == TRUE) %>%
                                              filter(Group_final == "Lethal PanIN - late" | Group_final == "Healthy control - late") %>%
                                              mutate(Condition = ifelse(Group_final == "Lethal PanIN - late",
                                                                        "KPC_late",
                                                                        "KP_late")) %>%
                                              dplyr::group_by(AptName,Condition) %>%
                                              mutate(Log2_Intensity = log(Intensity,2)) %>%
                                              summarize(data = list(Log2_Intensity), count = n(), .groups = "drop") %>%
                                              pivot_wider(names_from = Condition, values_from = c(data,count)) %>%
                                              filter(count_KPC_late > 1 & count_KP_late > 1)
```

```{r WiT_DE5_Mouse}
#Wilcox test - store pvalue and difference
WiT_DE5_mouse_df = DE5_mouse_wide_df %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_KPC_late), unlist(data_KP_late))$p.value) %>%
  mutate(Diff = (median(unlist(data_KPC_late)) - median(unlist(data_KP_late)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE5_mouse_df = merge(WiT_DE5_mouse_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE5_mouse_df = WiT_DE5_mouse_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r WiT_DE5_Mouse_Alt}
#Wilcox test - store pvalue and difference
WiT_DE5_mouse_alt_df = DE5_mouse_alt_wide_df %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_KPC_late), unlist(data_KP_late))$p.value) %>%
  mutate(Diff = (median(unlist(data_KPC_late)) - median(unlist(data_KP_late)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE5_mouse_alt_df = merge(WiT_DE5_mouse_alt_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE5_mouse_alt_df = WiT_DE5_mouse_alt_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r Plot_WiT_pval_DE5_Mouse}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE5_mouse_plot = WiT_DE5_mouse_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "Aptamer Count",
       title = "Mouse",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE5_mouse_plot.png")
```

```{r Plot_WiT_pval_DE5_Mouse_Alt}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE5_mouse_alt_plot = WiT_DE5_mouse_alt_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "",
       title = "Mouse (Pre-Normalization)",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE5_mouse_alt_plot.png")
```

```{r Plot_PvalCombo_DE5_Mouse}
fig_pval_DE5_Mouse = ggarrange(WiT_pval_dist_DE5_mouse_plot, WiT_pval_dist_DE5_mouse_alt_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_pval_DE5_Mouse, top = text_grob("P-Value Distribution (Wilcox Test): Lethal PanIN Late/KPC-Late v. Healthy Control Late/KP Late", size = 14))
```

\newline

**Figure 25: P-value distributions for the Wilcox tests performed on the mouse data comparing intensity (RFU) values between proteins in the Lethal PanIN - Late/KPC - Late and Healthy Control - Late/KP-Late groups. A) Normalized mouse data, B) Pre-normalization mouse data.**

```{r Plot_WiT_volc_DE5_Mouse}
#Wilcoxon volcano Plot
WiT_DE5_mouse_volcano_plot = WiT_DE5_mouse_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Mouse",
       x = "Diff(KPC_late - KP_late)",
       y = "-log10(p-value)",
       subtitle = "") +
  theme_bw() +
  xlim(-20,20) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_DE5_mouse_volcano_plot.png")

WiT_DE5_mouse_volcano_plotly = ggplotly(WiT_DE5_mouse_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE5_mouse_volcano_plotly, "WiT_DE5_mouse_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_WiT_volc_DE5_Mouse_Alt}
#Wilcoxon volcano Plot
WiT_DE5_mouse_alt_volcano_plot = WiT_DE5_mouse_alt_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Mouse (Pre-Normalization)",
       x = "Diff(KPC_late - KP_late)",
       y = "",
       subtitle = "") +
  theme_bw() +
  xlim(-20,20) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_DE5_mouse_alt_volcano_plot.png")

WiT_DE5_mouse_alt_volcano_plotly = ggplotly(WiT_DE5_mouse_alt_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE5_mouse_alt_volcano_plotly, "WiT_DE5_mouse_alt_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_VolcCombo_DE5_Mouse}
fig_WiT_DE5_Mouse = ggarrange(WiT_DE5_mouse_volcano_plot, WiT_DE5_mouse_alt_volcano_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_WiT_DE5_Mouse, top = text_grob("Wilcox Test: KPC-Late v. KP-Late", size = 14))
```  

\newline

**Figure 26: Volcano plots showing significantly differentially expressed proteins between Lethal PanIN - Late/KPC - Late and Healthy Control - Late/KP-Late groups in mouse data, A) normalized, B) pre-normalization.**
  
#### KC-Late v. KP Late:

  *Reveals differentially abundant proteins in non-lethal PanIN model*
  
```{r Stat_DataTransform_DE6_Mouse}
#Mouse DE6: KC-Late v. KP Late

#Transform data to wide form for statistical test
DE6_mouse_wide_df = filt_mouse_df %>% filter(!is.na(Group_final)) %>%
                                      filter(eLoD_Flag == TRUE) %>%
                                      filter(Group_final == "Non-lethal PanIN" | Group_final == "Healthy control - late") %>%
                                      mutate(Condition = ifelse(Group_final == "Non-lethal PanIN",
                                                                "KC_late",
                                                                "KP_late")) %>%
                                      dplyr::group_by(AptName,Condition) %>%
                                      mutate(Log2_Intensity = log(Intensity,2)) %>%
                                      summarize(data = list(Log2_Intensity), count = n(), .groups = "drop") %>%
                                      pivot_wider(names_from = Condition, values_from = c(data,count)) %>%
                                      filter(count_KC_late > 1 & count_KP_late > 1)
```

```{r Stat_DataTransform_DE6_Mouse_Alt}
#Mouse Alt DE6: KC-Late v. KP Late

#Transform data to wide form for statistical test
DE6_mouse_alt_wide_df = filt_mouse_alt_df %>% filter(!is.na(Group_final)) %>%
                                              filter(eLoD_Flag == TRUE) %>%
                                              filter(Group_final == "Non-lethal PanIN" | Group_final == "Healthy control - late") %>%
                                              mutate(Condition = ifelse(Group_final == "Non-lethal PanIN",
                                                                        "KC_late",
                                                                        "KP_late")) %>%
                                              dplyr::group_by(AptName,Condition) %>%
                                              mutate(Log2_Intensity = log(Intensity,2)) %>%
                                              summarize(data = list(Log2_Intensity), count = n(), .groups = "drop") %>%
                                              pivot_wider(names_from = Condition, values_from = c(data,count)) %>%
                                              filter(count_KC_late > 1 & count_KP_late > 1)
```

```{r WiT_DE6_Mouse}
#Wilcox test - store pvalue and difference
WiT_DE6_mouse_df = DE6_mouse_wide_df %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_KC_late), unlist(data_KP_late))$p.value) %>%
  mutate(Diff = (median(unlist(data_KC_late)) - median(unlist(data_KP_late)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE6_mouse_df = merge(WiT_DE6_mouse_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE6_mouse_df = WiT_DE6_mouse_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r WiT_DE6_Mouse_Alt}
#Wilcox test - store pvalue and difference
WiT_DE6_mouse_alt_df = DE6_mouse_alt_wide_df %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_KC_late), unlist(data_KP_late))$p.value) %>%
  mutate(Diff = (median(unlist(data_KC_late)) - median(unlist(data_KP_late)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE6_mouse_alt_df = merge(WiT_DE6_mouse_alt_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE6_mouse_alt_df = WiT_DE6_mouse_alt_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r Plot_WiT_pval_DE6_Mouse}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE6_mouse_plot = WiT_DE6_mouse_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "Aptamer Count",
       title = "Mouse",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE6_mouse_plot.png")
```

```{r Plot_WiT_pval_DE6_Mouse_Alt}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE6_mouse_alt_plot = WiT_DE6_mouse_alt_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "",
       title = "Mouse (Pre-Normalization)",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE6_mouse_alt_plot.png")
```

```{r Plot_PvalCombo_DE6_Mouse}
fig_pval_DE6_Mouse = ggarrange(WiT_pval_dist_DE6_mouse_plot, WiT_pval_dist_DE6_mouse_alt_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_pval_DE6_Mouse, top = text_grob("P-Value Distribution (Wilcox Test): Non-lethal PanIN/KC-Late v. Healthy Control Late/KP Late", size = 14))
```

\newline

**Figure 27: P-value distributions for the Wilcox tests performed on the mouse data comparing intensity (RFU) values between proteins in the Non-lethal PanIN - Late/KC - Late and Healthy Control - Late/KP-Late groups. A) Normalized mouse data, B) Pre-normalization mouse data.**

```{r Plot_WiT_volc_DE6_Mouse}
#Wilcoxon volcano Plot
WiT_DE6_mouse_volcano_plot = WiT_DE6_mouse_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Mouse",
       x = "Diff(KC_late - KP_late)",
       y = "-log10(p-value)",
       subtitle = "") +
  theme_bw() +
  xlim(-20,20) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_DE6_mouse_volcano_plot.png")

WiT_DE6_mouse_volcano_plotly = ggplotly(WiT_DE6_mouse_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE6_mouse_volcano_plotly, "WiT_DE6_mouse_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_WiT_volc_DE6_Mouse_Alt}
#Wilcoxon volcano Plot
WiT_DE6_mouse_alt_volcano_plot = WiT_DE6_mouse_alt_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Mouse (Pre-Normalization)",
       x = "Diff(KC_late - KP_late)",
       y = "",
       subtitle = "") +
  theme_bw() +
  xlim(-20,20) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_DE6_mouse_alt_volcano_plot.png")

WiT_DE6_mouse_alt_volcano_plotly = ggplotly(WiT_DE6_mouse_alt_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE6_mouse_alt_volcano_plotly, "WiT_DE6_mouse_alt_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_VolcCombo_DE6_Mouse}
fig_WiT_DE6_Mouse = ggarrange(WiT_DE6_mouse_volcano_plot, WiT_DE6_mouse_alt_volcano_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_WiT_DE6_Mouse, top = text_grob("Wilcox Test: KC_Late v. KP_Late", size = 14))
```  

\newline

**Figure 28: Volcano plots showing significantly differentially expressed proteins between Non-lethal PanIN - Late/KC - Late and Healthy Control - Late/KP-Late groups in mouse data, A) normalized, B) pre-normalization.**

### Human (Project Orion) Analysis

#### Assess Limit of Detection (Dilution Series) - PRE-NORMALIZATION DATA ONLY

```{r Stat_DataTransform_Dilution_Human}
#Mouse Dilution Series: 35uL vs. 55uL vs. 55uL Diluted (USES ALT DATA - PRE-NORMALIZATION)

#Transform data to wide form for statistical test
dilution_human_wide_df = filt_human_alt_df %>% filter(startsWith(Sample_ID,"Human")) %>%
                                                filter(eLoD_Flag == TRUE) %>%
                                                mutate(Volume_uL = ifelse(startsWith(Sample_ID,"Human 35"),
                                                                          "35",
                                                                          ifelse(startsWith(Sample_ID,"Human A"),
                                                                                 "55",
                                                                                 "55_Diluted"))) %>%
                                                dplyr::group_by(AptName,Volume_uL) %>%
                                                mutate(Log2_Intensity = log(Intensity,2)) %>%
                                                summarize(data = list(Log2_Intensity), count = n(), .groups = "drop") %>%
                                                pivot_wider(names_from = Volume_uL, values_from = c(data,count)) %>%
                                                filter(count_35 > 1 & count_55 > 1 & count_55_Diluted > 1)
```

```{r WiT_Dilution_Human}
#Wilcox test - store pvalue and difference
WiT_res_human_df = dilution_human_wide_df %>%
  rowwise() %>%
  mutate(wt_pval_A = wilcox.test(unlist(data_35), unlist(data_55))$p.value) %>%
  mutate(Diff_A = (median(unlist(data_35)) - median(unlist(data_55)))) %>%
  ungroup() %>%
  mutate(wt_pval_A_BH = p.adjust(wt_pval_A, method = "BH")) %>%
  rowwise() %>%
  mutate(wt_pval_B = wilcox.test(unlist(data_35), unlist(data_55_Diluted))$p.value) %>%
  mutate(Diff_B = (median(unlist(data_35)) - median(unlist(data_55_Diluted)))) %>%
  ungroup() %>%
  mutate(wt_pval_B_BH = p.adjust(wt_pval_B, method = "BH")) %>%
  rowwise() %>%
  mutate(wt_pval_C = wilcox.test(unlist(data_55), unlist(data_55_Diluted))$p.value) %>%
  mutate(Diff_C = (median(unlist(data_55)) - median(unlist(data_55_Diluted)))) %>%
  ungroup() %>%
  mutate(wt_pval_C_BH = p.adjust(wt_pval_C, method = "BH"))

partial_apt_anno_df = apt_anno_human_df %>%
  dplyr::select(c("AptName", "Target", "TargetFullName", "UniProt"))

WiT_res_human_df = merge(WiT_res_human_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_res_human_df = WiT_res_human_df %>%
  mutate(p_val_A_sig = ifelse(wt_pval_A_BH < 0.05, "Signficant", "Non"),
         proteins_A_sig = ifelse(wt_pval_A_BH < 0.05, Target, ""),
         p_val_B_sig = ifelse(wt_pval_B_BH < 0.05, "Signficant", "Non"),
         proteins_B_sig = ifelse(wt_pval_B_BH < 0.05, Target, ""),
         p_val_C_sig = ifelse(wt_pval_C_BH < 0.05, "Signficant", "Non"),
         proteins_C_sig = ifelse(wt_pval_C_BH < 0.05, Target, "")
  )
```

```{r Plot_WiT_pval_Dilution_Human}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_A_dist_human_plot = WiT_res_human_df %>%
  ggplot(aes(wt_pval_A)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "",
       y = "Aptamer Count",
       title = "",
       subtitle = "35uL vs. 55uL") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_A_dist_human_plot.png")

WiT_pval_B_dist_human_plot = WiT_res_human_df %>%
  ggplot(aes(wt_pval_B)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "Wilcox Test Raw P-value",
       y = "",
       title = "",
       subtitle = "35uL vs. 55uL Diluted") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_B_dist_human_plot.png")

WiT_pval_C_dist_human_plot = WiT_res_human_df %>%
  ggplot(aes(wt_pval_C)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "",
       y = "",
       title = "",
       subtitle = "55uL vs. 55uL Diluted") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_C_dist_human_plot.png")

fig_WiT_Dilution_Human = ggarrange(WiT_pval_A_dist_human_plot, WiT_pval_B_dist_human_plot, WiT_pval_C_dist_human_plot, labels="AUTO", ncol = 3, nrow = 1)
annotate_figure(fig_WiT_Dilution_Human, top = text_grob("Wilcox Test for Proteins in SomaLogic Study: Human", size = 14))
```

\newline

**Figure 29: P-value distributions for the wilcox tests performed on the human dilution series data (pre-normalization).**

```{r Plot_WiT_volc_Dilution_Human}
#Wilcoxon volcano Plot
WiT_A_human_volcano_plot = WiT_res_human_df %>%
  ggplot(aes(x = Diff_A, y = -log(wt_pval_A_BH, 10), group = p_val_A_sig)) +
  geom_point(aes(color = p_val_A_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed", color = "red") +
  geom_text_repel(aes(label = proteins_A_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "",
       x = "Log2(FC)",
       y = "-log10(p-value)",
       subtitle = "35uL vs. 55uL") +
  theme_bw() +
  xlim(-5,5) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic")) +
  theme(axis.text.x = element_text(vjust = 0.5, size = 12)) +
  theme(axis.text.y = element_text(vjust = 0.5, size = 12)) +
  theme(axis.title = element_text(vjust = 0.5, size = 14)) +
  theme(title = element_text(size = 14))
#ggsave("WiT_A_human_volcano_plot.png")
# WiT_A_human_volcano_plot

WiT_A_human_volcano_plotly = ggplotly(WiT_A_human_volcano_plot, tooltip = "Target")
saveWidget(WiT_A_human_volcano_plotly, "WiT_A_human_volcano_plotly.html", selfcontained = F, libdir = "lib")

WiT_B_human_volcano_plot = WiT_res_human_df %>%
  ggplot(aes(x = Diff_B, y = -log(wt_pval_B_BH, 10), group = p_val_B_sig)) +
  geom_point(aes(color = p_val_B_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed", color = "red") +
  geom_text_repel(aes(label = proteins_B_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "",
       x = "Log2(FC)",
       y = "",
       subtitle = "35uL vs. 55uL D") +
  theme_bw() +
  xlim(-5,5) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic")) +
  theme(axis.text.x = element_text(vjust = 0.5, size = 12)) +
  theme(axis.text.y = element_text(vjust = 0.5, size = 12)) +
  theme(axis.title = element_text(vjust = 0.5, size = 14)) +
  theme(title = element_text(size = 14))
#ggsave("WiT_B_human_volcano_plot.png")
# WiT_B_human_volcano_plot

WiT_B_human_volcano_plotly = ggplotly(WiT_B_human_volcano_plot, tooltip = "Target")
saveWidget(WiT_B_human_volcano_plotly, "WiT_B_human_volcano_plotly.html", selfcontained = F, libdir = "lib")

WiT_C_human_volcano_plot = WiT_res_human_df %>%
  ggplot(aes(x = Diff_C, y = -log(wt_pval_C_BH, 10), group = p_val_C_sig)) +
  geom_point(aes(color = p_val_C_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed", color = "red") +
  geom_text_repel(aes(label = proteins_C_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "",
       x = "Log2(FC)",
       y = "",
       subtitle = "55uL vs. 55uL D") +
  theme_bw() +
  xlim(-5,5) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic")) +
  theme(axis.text.x = element_text(vjust = 0.5, size = 12)) +
  theme(axis.text.y = element_text(vjust = 0.5, size = 12)) +
  theme(axis.title = element_text(vjust = 0.5, size = 14)) +
  theme(title = element_text(size = 14))
#ggsave("WiT_C_human_volcano_plot.png")
# WiT_C_human_volcano_plot

WiT_C_human_volcano_plotly = ggplotly(WiT_C_human_volcano_plot, tooltip = "Target")
saveWidget(WiT_C_human_volcano_plotly, "WiT_C_human_volcano_plotly.html", selfcontained = F, libdir = "lib")

fig_WiT_Dilution_Human = ggarrange(WiT_A_human_volcano_plot, WiT_B_human_volcano_plot, WiT_C_human_volcano_plot, ncol = 3, nrow = 1)
annotate_figure(fig_WiT_Dilution_Human, top = text_grob("SomaLogic Human DE: Dilution Set", size = 18))
ggsave("fig_WiT_Dilution_Human.png", width = 6, height = 3, units = "in")
```

\newline

**Figure 30: Volcano plots showing significantly differentially expressed proteins between human dilution series samples A) 35uL vs. 55uL, B) 35uL vs. 55uL Diluted [35uL sample, 20uL PBS], C) 55uL vs. 55uL Diluted.**

**Table 4: Number of aptamers/proteins detected per human dilution series sample (pre-normalization). The 55uL Diluted samples have the lowest counts, but all are decently high (max 7596 aptamers).**

```{r Protein_Check_DilutionSeries_Human}
#Check protein ID counts across dilutions
dilutionprots_human_df = filt_human_alt_df %>% filter(startsWith(Sample_ID,"Human")) %>%
                                                filter(eLoD_Flag == TRUE) %>%
                                                mutate(Category = ifelse(startsWith(Sample_ID,"Human 35"),
                                                                          "35uL",
                                                                          ifelse(startsWith(Sample_ID,"Human A"),
                                                                                 "55uL",
                                                                                 "55uL Diluted"))) %>%
                                                group_by(Category,Sample_ID) %>%
                                                summarise(AptCount = n())

# test_human = WiT_res_human_df %>% nrow()

knitr::kable(dilutionprots_human_df)
```

```{r Plot_WiT_box_Dilution_Human}
#Organize FC in long format
WiT_boxDS_human_df = WiT_res_human_df %>% pivot_longer(names_to = "Dilution_Comparison",
                                                        values_to = "FC",
                                                        cols = starts_with("Diff")) %>%
                                          mutate(Dilution_Comparison = ifelse(Dilution_Comparison == "Diff_A",
                                                                              "35uL-55uL",
                                                                              ifelse(Dilution_Comparison == "Diff_B",
                                                                                     "35uL-55uL_D",
                                                                                     "55uL-55uL_D")))

#Boxplot of intensity distributions per sample - human
WiT_boxDS_human_plot = WiT_boxDS_human_df %>%
  ggplot(aes(x = Dilution_Comparison, y = FC, fill = Dilution_Comparison)) +
  geom_boxplot(alpha = 0.5) +
  #geom_text(aes(label = median(FC))) +
  labs(title = "Dilution FC Distribution: Human",
       subtitle = "",
       y = "Log2(FC)") +
  theme_bw() +
  #stat_mean_sd_text(size = 2, y.pos = 6000) +
  #ylim(0,6000) +
  geom_hline(yintercept=log2(55/35), colour="purple4", linetype = "dashed") +
  theme(legend.position = "none") +
  theme(axis.text.x = element_text(vjust = 0.5, size = 12)) +
  theme(axis.title.x = element_blank()) +
  theme(axis.title = element_text(vjust = 0.5, size = 14)) +
  theme(title = element_text(size = 14))
ggsave("WiT_boxDS_human_plot.png", width = 4, height = 3, units = "in")
WiT_boxDS_human_plot

```

**Figure 31: Boxplot FC distributions of human dilution series data. Median FC is on target for expected dilutions ratio from 35 to 55L (0.6x).**

#### Case (Adenocarcinoma) v. All Controls

```{r Stat_DataTransform_DE1_Human}
#View group annos for comparison
groups_human_df = filt_human_df %>% filter(!is.na(Clinical.Diagnosis)) %>% dplyr::group_by(Clinical.Diagnosis) %>% summarise(Clinical.Diagnosis, Screening.Outcome) %>% unique()

#Human DE1: Case v. All Control

#Transform data to wide form for statistical test
DE1_human_wide_df = filt_human_df %>% filter(eLoD_Flag == TRUE) %>%
                                      filter(!is.na(Clinical.Diagnosis)) %>%
                                      dplyr::group_by(AptName,Screening.Outcome) %>%
                                      mutate(Log2_Intensity = log(Intensity,2)) %>%
                                      summarize(data = list(Log2_Intensity), count = n(), .groups = "drop") %>%
                                      pivot_wider(names_from = Screening.Outcome, values_from = c(data,count)) %>%
                                      filter(count_Case > 1 & count_Control > 1)
```

```{r Stat_DataTransform_DE1_Human_Alt}
#Mouse Alt DE1: Case v. All Control

#Transform data to wide form for statistical test
DE1_human_alt_wide_df = filt_human_alt_df %>% filter(eLoD_Flag == TRUE) %>%
                                              filter(!is.na(Clinical.Diagnosis)) %>%
                                              dplyr::group_by(AptName,Screening.Outcome) %>%
                                              mutate(Log2_Intensity = log(Intensity,2)) %>%
                                              summarize(data = list(Log2_Intensity), count = n(), .groups = "drop") %>%
                                              pivot_wider(names_from = Screening.Outcome, values_from = c(data,count)) %>%
                                              filter(count_Case > 1 & count_Control > 1)
```

```{r WiT_DE1_Human}
#Wilcox test - store pvalue and difference
WiT_DE1_human_df = DE1_human_wide_df %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_Case), unlist(data_Control))$p.value) %>%
  mutate(Diff = (median(unlist(data_Case)) - median(unlist(data_Control)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE1_human_df = merge(WiT_DE1_human_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE1_human_df = WiT_DE1_human_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r WiT_DE1_Human_Alt}
#Wilcox test - store pvalue and difference
WiT_DE1_human_alt_df = DE1_human_alt_wide_df %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_Case), unlist(data_Control))$p.value) %>%
  mutate(Diff = (median(unlist(data_Case)) - median(unlist(data_Control)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE1_human_alt_df = merge(WiT_DE1_human_alt_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE1_human_alt_df = WiT_DE1_human_alt_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r Plot_WiT_pval_DE1_Human}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE1_human_plot = WiT_DE1_human_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "Aptamer Count",
       title = "Human",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE1_human_plot.png")
```

```{r Plot_WiT_pval_DE1_Human_Alt}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE1_human_alt_plot = WiT_DE1_human_alt_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "",
       title = "Human (Pre-Normalization)",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE1_human_alt_plot.png")
```

```{r Plot_PvalCombo_DE1_Human}
fig_pval_DE1_Human = ggarrange(WiT_pval_dist_DE1_human_plot, WiT_pval_dist_DE1_human_alt_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_pval_DE1_Human, top = text_grob("P-Value Distribution (Wilcox Test): Case (Adenocarcinoma) v. All Controls", size = 14))
```

\newline

**Figure 32: P-value distributions for the Wilcox tests performed on the human data comparing intensity (RFU) values between proteins in the Case (Adenocarcinoma) and All Controls groups. A) Normalized human data, B) Pre-normalization human data.**

```{r Plot_WiT_volc_DE1_Human}
#Wilcoxon volcano Plot
WiT_DE1_human_volcano_plot = WiT_DE1_human_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Human",
       x = "Diff(Case - All Controls)",
       y = "-log10(p-value)",
       subtitle = "") +
  theme_bw() +
  xlim(-20,20) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_DE1_human_volcano_plot.png")

WiT_DE1_human_volcano_plotly = ggplotly(WiT_DE1_human_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE1_human_volcano_plotly, "WiT_DE1_human_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_WiT_volc_DE1_Human_Alt}
#Wilcoxon volcano Plot
WiT_DE1_human_alt_volcano_plot = WiT_DE1_human_alt_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Human (Pre-Normalization)",
       x = "Diff(Case - All Controls)",
       y = "",
       subtitle = "") +
  theme_bw() +
  xlim(-6,6) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_DE1_human_alt_volcano_plot.png")

WiT_DE1_human_alt_volcano_plotly = ggplotly(WiT_DE1_human_alt_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE1_human_alt_volcano_plotly, "WiT_DE1_human_alt_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_VolcCombo_DE1_Human}
fig_WiT_DE1_Human = ggarrange(WiT_DE1_human_volcano_plot, WiT_DE1_human_alt_volcano_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_WiT_DE1_Human, top = text_grob("Wilcox Test: Case v. Controls", size = 14))
```  

\newline

**Figure 33: Volcano plots showing significantly differentially expressed proteins between Case (Adenocarcinoma) and All Controls groups in human data, A) normalized, B) pre-normalization.**

#### Case (Adenocarcinoma) v. Control (Benign)

```{r Stat_DataTransform_DE2_Human}
#Human DE2: Case v. Control (Benign)

#Transform data to wide form for statistical test
DE2_human_wide_df = filt_human_df %>% filter(eLoD_Flag == TRUE) %>%
                                      filter(!is.na(Clinical.Diagnosis)) %>%
                                      dplyr::group_by(AptName,Clinical.Diagnosis) %>%
                                      mutate(Log2_Intensity = log(Intensity,2)) %>%
                                      summarize(data = list(Log2_Intensity), count = n(), .groups = "drop") %>%
                                      pivot_wider(names_from = Clinical.Diagnosis, values_from = c(data,count))
```

```{r Stat_DataTransform_DE2_Human_Alt}
#Mouse Alt DE2: Case v. Control (Benign)

#Transform data to wide form for statistical test
DE2_human_alt_wide_df = filt_human_alt_df %>% filter(eLoD_Flag == TRUE) %>%
                                              filter(!is.na(Clinical.Diagnosis)) %>%
                                              dplyr::group_by(AptName,Clinical.Diagnosis) %>%
                                              mutate(Log2_Intensity = log(Intensity,2)) %>%
                                              summarize(data = list(Log2_Intensity), count = n(), .groups = "drop") %>%
                                              pivot_wider(names_from = Clinical.Diagnosis, values_from = c(data,count))
```

```{r WiT_DE2_Human}
#Wilcox test - store pvalue and difference
WiT_DE2_human_df = DE2_human_wide_df %>%
  filter(count_Adenocarcinoma > 1 & count_Benign > 1) %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_Adenocarcinoma), unlist(data_Benign))$p.value) %>%
  mutate(Diff = (median(unlist(data_Adenocarcinoma)) - median(unlist(data_Benign)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE2_human_df = merge(WiT_DE2_human_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE2_human_df = WiT_DE2_human_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r WiT_DE2_Human_Alt}
#Wilcox test - store pvalue and difference
WiT_DE2_human_alt_df = DE2_human_alt_wide_df %>%
  filter(count_Adenocarcinoma > 1 & count_Benign > 1) %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_Adenocarcinoma), unlist(data_Benign))$p.value) %>%
  mutate(Diff = (median(unlist(data_Adenocarcinoma)) - median(unlist(data_Benign)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE2_human_alt_df = merge(WiT_DE2_human_alt_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE2_human_alt_df = WiT_DE2_human_alt_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r Plot_WiT_pval_DE2_Human}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE2_human_plot = WiT_DE2_human_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "Aptamer Count",
       title = "Human",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE2_human_plot.png")
```

```{r Plot_WiT_pval_DE2_Human_Alt}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE2_human_alt_plot = WiT_DE2_human_alt_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "",
       title = "Human (Pre-Normalization)",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE2_human_alt_plot.png")
```

```{r Plot_PvalCombo_DE2_Human}
fig_pval_DE2_Human = ggarrange(WiT_pval_dist_DE2_human_plot, WiT_pval_dist_DE2_human_alt_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_pval_DE2_Human, top = text_grob("P-Value Distribution (Wilcox Test): Case (Adenocarcinoma) v. Control (Benign)", size = 14))
```

\newline

**Figure 34: P-value distributions for the Wilcox tests performed on the human data comparing intensity (RFU) values between proteins in the Case (Adenocarcinoma) and Control (Benign) groups. A) Normalized human data, B) Pre-normalization human data.**

```{r Plot_WiT_volc_DE2_Human}
#Wilcoxon volcano Plot
WiT_DE2_human_volcano_plot = WiT_DE2_human_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Human",
       x = "Diff(Case - Control (Benign))",
       y = "-log10(p-value)",
       subtitle = "") +
  theme_bw() +
  xlim(-20,20) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_DE2_human_volcano_plot.png")

WiT_DE2_human_volcano_plotly = ggplotly(WiT_DE2_human_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE2_human_volcano_plotly, "WiT_DE2_human_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_WiT_volc_DE2_Human_Alt}
#Wilcoxon volcano Plot
WiT_DE2_human_alt_volcano_plot = WiT_DE2_human_alt_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Human (Pre-Normalization)",
       x = "Diff(Case - Control(Benign))",
       y = "",
       subtitle = "") +
  theme_bw() +
  xlim(-20,20) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_DE2_human_alt_volcano_plot.png")

WiT_DE2_human_alt_volcano_plotly = ggplotly(WiT_DE2_human_alt_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE2_human_alt_volcano_plotly, "WiT_DE2_human_alt_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_VolcCombo_DE2_Human}
fig_WiT_DE2_Human = ggarrange(WiT_DE2_human_volcano_plot, WiT_DE2_human_alt_volcano_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_WiT_DE2_Human, top = text_grob("Wilcox Test: Case v. Control (Benign)", size = 14))
```  

\newline

**Figure 35: Volcano plots showing significantly differentially expressed proteins between Case (Adenocarcinoma) and Control (Benign) groups in human data, A) normalized, B) pre-normalization.**

#### Case (Adenocarcinoma) v. Control (Benign Prostatic Hyperplasia)

```{r WiT_DE3_Human}
#Wilcox test - store pvalue and difference
WiT_DE3_human_df = DE2_human_wide_df %>%
  filter(count_Adenocarcinoma > 1 & `count_Benign prostatic hyperplasia` > 1) %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_Adenocarcinoma), unlist(`data_Benign prostatic hyperplasia`))$p.value) %>%
  mutate(Diff = (median(unlist(data_Adenocarcinoma)) - median(unlist(`data_Benign prostatic hyperplasia`)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE3_human_df = merge(WiT_DE3_human_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE3_human_df = WiT_DE3_human_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r WiT_DE3_Human_Alt}
#Wilcox test - store pvalue and difference
WiT_DE3_human_alt_df = DE2_human_alt_wide_df %>%
  filter(count_Adenocarcinoma > 1 & `count_Benign prostatic hyperplasia` > 1) %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_Adenocarcinoma), unlist(`data_Benign prostatic hyperplasia`))$p.value) %>%
  mutate(Diff = (median(unlist(data_Adenocarcinoma)) - median(unlist(`data_Benign prostatic hyperplasia`)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE3_human_alt_df = merge(WiT_DE3_human_alt_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE3_human_alt_df = WiT_DE3_human_alt_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r Plot_WiT_pval_DE3_Human}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE3_human_plot = WiT_DE3_human_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "Aptamer Count",
       title = "Human",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE3_human_plot.png")
```

```{r Plot_WiT_pval_DE3_Human_Alt}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE3_human_alt_plot = WiT_DE3_human_alt_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "",
       title = "Human (Pre-Normalization)",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE3_human_alt_plot.png")
```

```{r Plot_PvalCombo_DE3_Human}
fig_pval_DE3_Human = ggarrange(WiT_pval_dist_DE3_human_plot, WiT_pval_dist_DE3_human_alt_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_pval_DE3_Human, top = text_grob("P-Value Distribution (Wilcox Test): Case (Adenocarcinoma) v. Control (Benign Prostatic Hyperplasia)", size = 14))
```

\newline

**Figure 36: P-value distributions for the Wilcox tests performed on the human data comparing intensity (RFU) values between proteins in the Case (Adenocarcinoma) and Control (Benign Prostatic Hyperplasia) groups. A) Normalized human data, B) Pre-normalization human data.**

```{r Plot_WiT_volc_DE3_Human}
#Wilcoxon volcano Plot
WiT_DE3_human_volcano_plot = WiT_DE3_human_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Human",
       x = "Diff(Case - Control (Benign Prostatic Hyperplasia))",
       y = "-log10(p-value)",
       subtitle = "") +
  theme_bw() +
  xlim(-20,20) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_DE3_human_volcano_plot.png")

WiT_DE3_human_volcano_plotly = ggplotly(WiT_DE3_human_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE3_human_volcano_plotly, "WiT_DE3_human_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_WiT_volc_DE3_Human_Alt}
#Wilcoxon volcano Plot
WiT_DE3_human_alt_volcano_plot = WiT_DE3_human_alt_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Human (Pre-Normalization)",
       x = "Diff(Case - Control(Benign Prostatic Hyperplasia))",
       y = "",
       subtitle = "") +
  theme_bw() +
  xlim(-20,20) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_DE3_human_alt_volcano_plot.png")

WiT_DE3_human_alt_volcano_plotly = ggplotly(WiT_DE3_human_alt_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE3_human_alt_volcano_plotly, "WiT_DE3_human_alt_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_VolcCombo_DE3_Human}
fig_WiT_DE3_Human = ggarrange(WiT_DE3_human_volcano_plot, WiT_DE3_human_alt_volcano_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_WiT_DE3_Human, top = text_grob("Wilcox Test: Case v. Control (Benign Prostatic Hyperplasia)", size = 14))
```  

\newline

**Figure 37: Volcano plots showing significantly differentially expressed proteins between Case (Adenocarcinoma) and Control (Benign Prostatic Hyperplasia) groups in human data, A) normalized, B) pre-normalization.**

#### Case (Adenocarcinoma) v. Control (Prostatic intraepithelial neoplasia)

```{r WiT_DE4_Human}
#Wilcox test - store pvalue and difference
WiT_DE4_human_df = DE2_human_wide_df %>%
  filter(count_Adenocarcinoma > 1 & `count_Prostatic intraepithelial neoplasia` > 1) %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_Adenocarcinoma), unlist(`data_Prostatic intraepithelial neoplasia`))$p.value) %>%
  mutate(Diff = (median(unlist(data_Adenocarcinoma)) - median(unlist(`data_Prostatic intraepithelial neoplasia`)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE4_human_df = merge(WiT_DE4_human_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE4_human_df = WiT_DE4_human_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r WiT_DE4_Human_Alt}
#Wilcox test - store pvalue and difference
WiT_DE4_human_alt_df = DE2_human_alt_wide_df %>%
  filter(count_Adenocarcinoma > 1 & `count_Prostatic intraepithelial neoplasia` > 1) %>%
  rowwise() %>%
  mutate(wt_pval = wilcox.test(unlist(data_Adenocarcinoma), unlist(`data_Prostatic intraepithelial neoplasia`))$p.value) %>%
  mutate(Diff = (median(unlist(data_Adenocarcinoma)) - median(unlist(`data_Prostatic intraepithelial neoplasia`)))) %>%
  ungroup() %>%
  mutate(wt_pval_BH = p.adjust(wt_pval, method = "BH"))

WiT_DE4_human_alt_df = merge(WiT_DE4_human_alt_df, partial_apt_anno_df, by = "AptName", all.x = TRUE)

WiT_DE4_human_alt_df = WiT_DE4_human_alt_df %>%
  mutate(p_val_sig = ifelse(wt_pval_BH < 0.05, "Significant", "Non"),
         proteins_sig = ifelse(wt_pval_BH < 0.05, Target, ""),
  )
```

```{r Plot_WiT_pval_DE4_Human}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE4_human_plot = WiT_DE4_human_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "Aptamer Count",
       title = "Human",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE4_human_plot.png")
```

```{r Plot_WiT_pval_DE4_Human_Alt}
#Plot WiT pvalue distribution - before multiple testing correction
WiT_pval_dist_DE4_human_alt_plot = WiT_DE4_human_alt_df %>%
  ggplot(aes(wt_pval)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
  labs(x = "p-value",
       y = "",
       title = "Human (Pre-Normalization)",
       subtitle = "") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_pval_dist_DE4_human_alt_plot.png")
```

```{r Plot_PvalCombo_DE4_Human}
fig_pval_DE4_Human = ggarrange(WiT_pval_dist_DE4_human_plot, WiT_pval_dist_DE4_human_alt_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_pval_DE4_Human, top = text_grob("P-Value Distribution (Wilcox Test): Case (Adenocarcinoma) v. Control (Prostatic Intraepithelial Neoplasia)", size = 14))
```

\newline

**Figure 38: P-value distributions for the Wilcox tests performed on the human data comparing intensity (RFU) values between proteins in the Case (Adenocarcinoma) and Control (Prostatic Intraepithelial Neoplasia) groups. A) Normalized human data, B) Pre-normalization human data.**

```{r Plot_WiT_volc_DE4_Human}
#Wilcoxon volcano Plot
WiT_DE4_human_volcano_plot = WiT_DE4_human_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Human",
       x = "Diff(Case - Control (Prostatic Intraepithelial Neoplasia)",
       y = "-log10(p-value)",
       subtitle = "") +
  theme_bw() +
  xlim(-20,20) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_DE4_human_volcano_plot.png")

WiT_DE4_human_volcano_plotly = ggplotly(WiT_DE4_human_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE4_human_volcano_plotly, "WiT_DE4_human_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_WiT_volc_DE4_Human_Alt}
#Wilcoxon volcano Plot
WiT_DE4_human_alt_volcano_plot = WiT_DE4_human_alt_df %>%
  ggplot(aes(x = Diff, y = -log(wt_pval_BH, 10), group = p_val_sig)) +
  geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
  geom_hline(yintercept = -log(0.05, 10), linetype = "dashed") +
  geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
  scale_color_manual(values = c("black", "red")) +
  labs(title = "Human (Pre-Normalization)",
       x = "Diff(Case - Control(Prostatic Intraepithelial Neoplasia)",
       y = "",
       subtitle = "") +
  theme_bw() +
  xlim(-20,20) +
  theme(legend.position = "none") +
  theme(plot.subtitle = element_text(face = "italic"))
#ggsave("WiT_DE4_human_alt_volcano_plot.png")

WiT_DE4_human_alt_volcano_plotly = ggplotly(WiT_DE4_human_alt_volcano_plot, tooltip = "Target")
saveWidget(WiT_DE4_human_alt_volcano_plotly, "WiT_DE4_human_alt_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r Plot_VolcCombo_DE4_Human}
fig_WiT_DE4_Human = ggarrange(WiT_DE4_human_volcano_plot, WiT_DE4_human_alt_volcano_plot, labels="AUTO", ncol = 2, nrow = 1)
annotate_figure(fig_WiT_DE4_Human, top = text_grob("Wilcox Test: Case v. Control (Prostatic Intraepithelial Neoplasia)", size = 14))
```  

\newline

**Figure 39: Volcano plots showing significantly differentially expressed proteins between Case (Adenocarcinoma) and Control (Prostatic Intraepithelial Neoplasia) groups in human data, A) normalized, B) pre-normalization.**

## Somalogic vs. Seer Proteograph

### Protein ID Overlap & Fold Change Comparison - KMC-PDAC v. KMC Control

```{r KMC_WiT_Mouse_MethodComparison}
#Use biomart function to create merge column
source("NomPivot_MouseHuman_UniProtInput.R")
#Grab SOMA aptamer seq, target, targetfullname, intensity, columns from filt_mouse df (not using normalized data for now...)
soma_KMC_df = WiT_DE3_mouse_df %>% dplyr::select(-c(count_KMC_PDAC,count_KMC_control,data_KMC_PDAC,data_KMC_control)) %>%
                                    mutate(UniProt = ifelse(grepl("|", UniProt, fixed = TRUE), gsub("[|].*", "", UniProt), UniProt)) %>%
                                    mutate(Target = toupper(Target))
colnames(soma_KMC_df) = c("AptName","SOMA_wt_pval","SOMA_Diff_Median","SOMA_wt_pval_BH","SOMA_Target","SOMA_TargetFullName","UniProt","SOMA_p_val_sig","SOMA_prot_sig")
#Perform nomenclature pivot from Soma human uniprots to corresponding mouse uniprots
###NOTE: Depending on KEGG server status, the following function may result in a timeout error. If so, run again and continue when successful.
NCBI_KEGG_anno_df = nomConvert_Human_Mouse(soma_KMC_df$UniProt)
###
#Merge mouse uniprots and other mouse annotations to Soma data
soma_KMC_filt_df = merge(soma_KMC_df,NCBI_KEGG_anno_df, by.x = "UniProt", by.y = "HOM_UniProt", all.x = TRUE) %>% unique()
#Filter out human/soma genes with no mouse UniProt match
soma_KMC_filt_df = soma_KMC_filt_df %>% filter(!is.na(MUS_UniProt)) #%>% mutate(MUS_Gene = tolower(MUS_Gene))
#Load Seer KMC-KMC_control data
seer_KMC_df = read_tsv("~/Code/SomaLogic/Seer_MUS_KMC_WiT.tsv")
seer_KMC_df = seer_KMC_df %>% dplyr::select(-c(`count_KMC - PDAC`,count_KMC_control)) %>%
                                  mutate(UniProt = ifelse(grepl(";", Protein.Ids), sub(";.*", "", Protein.Ids), Protein.Ids)) %>%
                                  mutate(Proteins = ifelse(grepl(";", Protein.Names), sub(";.*", "", Protein.Names), Protein.Names)) %>%
                                  mutate(Feature = ifelse(grepl(";", Feature), sub(";.*", "", Feature), Feature)) %>%
                                  dplyr::select(-c("Protein.Ids","Protein.Names"))

#Merge Soma and Seer df for comparison
#seer_KMC_df = seer_KMC_df %>% filter(!is.na(Genes)) %>% mutate(Genes = tolower(Genes))
sscombo_KMC_df = merge(seer_KMC_df,soma_KMC_filt_df, by.x = "UniProt", by.y = "MUS_UniProt") #Intersecting UniProt IDs btwn Soma + Seer
colnames(sscombo_KMC_df)[colnames(sscombo_KMC_df) == "UniProt"] = "MUS_UniProt"
colnames(sscombo_KMC_df)[colnames(sscombo_KMC_df) == "UniProt.y"] = "HOM_UniProt"
colnames(sscombo_KMC_df)[colnames(sscombo_KMC_df) == "Proteins"] = "Seer_Proteins"
colnames(sscombo_KMC_df)[colnames(sscombo_KMC_df) == "SOMA_Target"] = "SOMA_Genes"

#Calculate soma --> seer protein overlap
sscombo_KMC_protID_overlap = sscombo_KMC_df %>% dplyr::select(c("MUS_UniProt","Seer_Proteins","MUS_Gene","HOM_UniProt","SOMA_Genes","HOM_Gene")) %>% unique()
#Export table of overlap proteins
write_tsv(sscombo_KMC_protID_overlap,"protID_overlap_KMC_SomatoSeer.tsv")
sscombo_KMC_sig_df = sscombo_KMC_df %>% filter(p_val_sig == "Significant")
sigprotID_KMC_overlap = sscombo_KMC_sig_df %>% dplyr::select(c("MUS_UniProt","Seer_Proteins","MUS_Gene","HOM_UniProt","SOMA_Genes","HOM_Gene")) %>% unique() #%>% nrow()

#Calculate seer --> soma protein overlap
###NOTE: Depending on KEGG server status, the following function may result in a timeout error. If so, run again and continue when successful.
NCBI_KEGG_Seer_anno_df = nomConvert_Mouse_Human(seer_KMC_df$UniProt)
###
seer_KMC_filt_df = merge(seer_KMC_df,NCBI_KEGG_Seer_anno_df, by.x = "UniProt", by.y = "MUS_UniProt", all.x = TRUE)
colnames(seer_KMC_filt_df)[colnames(seer_KMC_filt_df) == "Proteins"] = "Seer_Proteins"
colnames(seer_KMC_filt_df)[colnames(seer_KMC_filt_df) == "UniProt"] = "MUS_UniProt"
seer_KMC_filt_df = seer_KMC_filt_df %>% filter(!is.na(HOM_UniProt)) #%>% mutate(HOM_Gene = toupper(HOM_Gene))
seer_sscombo_KMC_df = merge(seer_KMC_filt_df,soma_KMC_df, by.x = "HOM_UniProt", by.y = "UniProt")
colnames(seer_sscombo_KMC_df)[colnames(seer_sscombo_KMC_df) == "SOMA_Target"] = "SOMA_Genes"
seer_sscombo_KMC_protein_overlap = seer_sscombo_KMC_df %>% dplyr::select(c("MUS_UniProt","Seer_Proteins","MUS_Gene","HOM_UniProt","SOMA_Genes","HOM_Gene")) %>%
                                                            unique() #%>% nrow()
#Export table of seer --> soma overlap proteins
write_tsv(seer_sscombo_KMC_protein_overlap,"protID_overlap_KMC_SeertoSoma.tsv")
seer_sscombo_KMC_sig_df = seer_sscombo_KMC_df %>% filter(p_val_sig == "Significant")
seer_sigprotID_KMC_overlap = seer_sscombo_KMC_sig_df %>% dplyr::select(c("MUS_UniProt","Seer_Proteins","MUS_Gene","HOM_UniProt","SOMA_Genes","HOM_Gene")) %>% unique() #%>% nrow()
#Find number of protein IDs (removing NP duplicates) in seer KMC data
seer_KMC_proteins = seer_KMC_df %>% dplyr::select(c(Proteins,UniProt)) %>% unique() %>% nrow()
#Find number of protein IDs (removing NP duplicates) in seer KMC data
sig_seer_KMC_proteins = seer_KMC_df %>% filter(p_val_sig == "Significant") %>% dplyr::select(c(Proteins,UniProt)) %>% unique() %>% nrow()

#Check for difference btwn soma --> seer and seer --> soma overlaps - all Seer/SOMA and MUS/HOM labels
#NOTE ON OVERLAP:
    #From a single submitted uniprot ID (either species), it is possible for my pivot functions to return multiple mapped uniprots because of how the
    #KEGG database is organized for the orthology query response. Depending on the direction I went (Seer murine --> human pivot for comparison to Soma
    #human, or Soma human --> murine pivot for comparison to Seer murine), I was getting very different overlap numbers due at least partially to these
    #cases where multiple mapping matches were found for a single uniprot. In an effort to get to a single overlap number (without just arbitrarily
    #choosing a pivot direction), I calculated the overall overlap by using the intersect() function to make a list of all uniprot pairs
    #(UniProts_MUS_HOM  column in the picture above) in common between the pivot beginning with Soma (human) and the pivot beginning with Seer (murine).
    #I then used that list to filter the combined results of both pivots to only include proteins where the murine UniProt and human uniprot were paired
    #together in both pivots.
#Proteins unique to SOMA --> Seer overlap
overlap_method_diff_SoSe = dplyr::setdiff(sscombo_KMC_protID_overlap,seer_sscombo_KMC_protein_overlap)
#Proteins unique to Seer --> SOMA overlap
overlap_method_diff_SeSo = dplyr::setdiff(seer_sscombo_KMC_protein_overlap, sscombo_KMC_protID_overlap)
#Proteins shared in both overlap sets (SOMA --> Seer & Seer --> SOMA intersect)
overlap_method_intersect = dplyr::intersect(sscombo_KMC_protID_overlap,seer_sscombo_KMC_protein_overlap)
overlap_method_intersect = overlap_method_intersect %>% mutate(UniProts_MUS_HOM = paste(MUS_UniProt,HOM_UniProt,sep = "_"))
#summary(comparedf(sscombo_KMC_protID_overlap,seer_sscombo_KMC_protein_overlap))
```

```{r KMC_WiT_Mouse_Alt_MethodComparison, include = FALSE}
# #Grab SOMA alt aptamer seq, target, targetfullname, intensity, columns from filt_mouse df (not using normalized data for now...)
# soma_KMC_alt_df = WiT_DE3_mouse_alt_df %>% dplyr::select(-c(count_KMC_PDAC,count_KMC_control,data_KMC_PDAC,data_KMC_control)) %>%
#                                     mutate(UniProt = ifelse(grepl("|", UniProt, fixed = TRUE), gsub("[|].*", "", UniProt), UniProt)) %>%
#                                     mutate(Gene_Merge = tolower(Target))
# colnames(soma_KMC_alt_df) = c("AptName","SOMA_wt_pval","SOMA_Diff_Median","SOMA_wt_pval_BH","SOMA_Target","SOMA_TargetFullName","UniProt","SOMA_p_val_sig","SOMA_prot_sig","Gene_Merge")
# 
# #Merge Soma alt and Seer df for comparison
# sscombo_KMC_alt_df = merge(soma_KMC_alt_df,seer_KMC_df, by="Gene_Merge") #Intersecting UniProt IDs btwn Soma + Seer
# 
# #Generate table of protein overlaps (removing NP replicates)
# # protID_alt_overlap = sscombo_KMC_alt_df %>% dplyr::select(c(Gene_Merge,SOMA_Target,SOMA_TargetFullName,UniProt.x,Genes,UniProt.y)) %>% unique()
# # colnames(protID_alt_overlap) = c("Gene_Merge","SOMA_Target","SOMA_TargetFullName","SOMA_UniProt","Seer_Target","Seer_UniProt")
# # #Export table of overlap proteins
# # write_tsv(protID_alt_overlap,"protID_alt_overlap.tsv")
# 
# sscombo_KMC_alt_sig_df = sscombo_KMC_alt_df %>% filter(Significance_wt == "Y")
# # sigprotID_KMC_alt_overlap = sscombo_KMC_alt_sig_df %>% dplyr::select(c(Genes)) %>% unique() %>% nrow()
```

```{r KMC_TT_Mouse_MethodComparison, include = FALSE}
#Grab SOMA aptamer seq, target, targetfullname, intensity, columns from filt_mouse df (not using normalized data for now...)
# soma_KMC_TT_df = TT_DE3_mouse_df %>% dplyr::select(-c(data_KMC_PDAC,data_KMC_control))
# colnames(soma_KMC_TT_df) = c("SOMA_count_KMC_PDAC","SOMA_count_KMC_control","SOMA_tt_pval","SOMA_Diff_Median","SOMA_tt_pval_BH","SOMA_Target","SOMA_TargetFullName","UniProt","SOMA_p_val_sig","SOMA_prot_sig")
# #Load Seer KMC-KMC_control data
# seer_KMC_TT_df = seer_KMC_df 
# # %>% dplyr::select(-c(`data_KMC - PDAC`,data_KMC_control,diff_mean,wt_pval,wt_pval_BH,Significance_wt,Sig_Gene_wt)) %>%
# #                                   mutate(UniProt = ifelse(grepl(";", Protein.Group), sub(";.*", "", Protein.Group), Protein.Group)) %>%
# #                                   mutate(Genes = ifelse(grepl(";", Genes), sub(";.*", "", Genes), Genes)) %>%
# #                                   mutate(feature = ifelse(grepl(";", feature), sub(";.*", "", feature), feature)) %>%
# #                                   dplyr::select(-Protein.Group)
# 
# sscombo_KMC_TT_df = merge(soma_KMC_TT_df,seer_KMC_TT_df, by="UniProt") #Intersecting UniProt IDs btwn Soma + Seer
```

```{r KMC_TT_Mouse_Alt_MethodComparison, include = FALSE}
# #Grab SOMA aptamer seq, target, targetfullname, intensity, columns from filt_mouse df (not using normalized data for now...)
# soma_KMC_TT_alt_df = TT_DE3_mouse_alt_df %>% dplyr::select(-c(data_KMC_PDAC,data_KMC_control,AptName))
# colnames(soma_KMC_TT_alt_df) = c("SOMA_count_KMC_PDAC","SOMA_count_KMC_control","SOMA_tt_pval","SOMA_Diff_Median","SOMA_tt_pval_BH","SOMA_Target","SOMA_TargetFullName","UniProt","SOMA_p_val_sig","SOMA_prot_sig")
# 
# sscombo_KMC_TT_alt_df = merge(soma_KMC_TT_alt_df,seer_KMC_TT_df, by="UniProt") #Intersecting UniProt IDs btwn Soma + Seer
```

```{r KMC_WiT_Mouse_Corr}
#Calculate per row (single protein Soma vs. Seer NP) correlation coefficient (r) for Soma and Seer fold changes

sscombo_KMC_corr_df = sscombo_KMC_df %>% mutate(UniProts_MUS_HOM = paste(MUS_UniProt,HOM_UniProt,sep = "_")) %>%
                                          #ONLY KEEP OVERLAP PROTEINS IN INTERSECT OF BOTH DIRECTION PIVOTS (767 unique)
                                          filter(UniProts_MUS_HOM %in% overlap_method_intersect$UniProts_MUS_HOM) %>%
                                          rowwise() %>% mutate(data_FC = list(c(SOMA_Diff_Median,Diff))) %>%
                                          mutate(FC_SD = sd(data_FC), FC_CV = 100*sqrt(2^(log(2)*(FC_SD^2))-1)) %>%
                                          mutate(AptName_SOMATarget = paste(AptName, SOMA_Genes, sep = "|")) %>%
                                          mutate(AptName_SeerFeature = paste(AptName_SOMATarget, Feature, sep = "_")) %>%
                                          ungroup() %>%
                                          mutate(SOMA_Sig_FC_Prot = ifelse(SOMA_Diff_Median >= 2 | SOMA_Diff_Median <= -2, AptName_SeerFeature, NA)) %>%
                                          mutate(Seer_Sig_FC_Prot = ifelse(Diff >= 2 | Diff <= -2, AptName_SeerFeature, NA)) %>%
                                          mutate(SigFC_Flag = ifelse(!is.na(SOMA_Sig_FC_Prot) & !is.na(Seer_Sig_FC_Prot),
                                                                      "Both Methods",
                                                                      ifelse(!is.na(SOMA_Sig_FC_Prot),
                                                                             "SOMA Only",
                                                                             ifelse(!is.na(Seer_Sig_FC_Prot),
                                                                                    "Seer Only",
                                                                                    "Neither Method"))))
colnames(sscombo_KMC_corr_df)[colnames(sscombo_KMC_corr_df) == "SOMA_Diff_Median"] = "Soma_FC"
colnames(sscombo_KMC_corr_df)[colnames(sscombo_KMC_corr_df) == "Diff"] = "Seer_FC"

#Seer-sig proteins in intersect-filtered KMC overlap
sigprotID_sscombo_KMC_intersectfilt_df = sscombo_KMC_corr_df %>% filter(p_val_sig == "Significant")
sigprotID_sscombo_KMC_intersectfilt = sigprotID_sscombo_KMC_intersectfilt_df$UniProts_MUS_HOM %>% unique() %>% length()

#Correlation calculation
cor_sscombo_KMC = cor(sscombo_KMC_corr_df$Soma_FC,sscombo_KMC_corr_df$Seer_FC,method="pearson")

#Adding in mouse aptamer quality terms to KMC corr df
subset_apt_anno_mouse_df = combined_apt_anno_mouse_df %>% dplyr::select(AptName,Quality)
sscombo_KMC_corr_df = merge(sscombo_KMC_corr_df,subset_apt_anno_mouse_df, by = "AptName")
sscombo_KMC_corr_overlap_df = sscombo_KMC_corr_df %>% dplyr::select(c("AptName","AptName_SOMATarget","HOM_UniProt","HOM_Gene","SOMA_Genes","MUS_UniProt","MUS_Gene","Seer_Proteins","Quality")) %>% unique()

#Plot distribution of quality in overlap protein aptamers
quality_overlap_plot = sscombo_KMC_corr_overlap_df %>%
  mutate(Quality = factor(Quality, levels = c("Low", "Medium", "High"))) %>%
  dplyr::group_by(Quality) %>%
  summarise(Apt_Count = n(), Percent = round((Apt_Count/811)*100,digits = 0), Perc_Label = paste(as.character(Percent),"%",sep = "")) %>% #Total number of rows in sscombo_KMC_corr_overlap_df, higher than 476 because of 2+ aptamers targeting single protein
  ggplot(aes(x = Quality, y = Apt_Count, fill = Quality)) +
  geom_bar(stat = "identity") +
  geom_text(aes(label = Perc_Label), position=position_dodge(width=0.9),vjust=-0.25) +
  labs(title = "Aptamer Quality of Overlap Proteins: KMC-PDAC v. KMC Control",
       subtitle = "Overlap = Proteins appearing in both SOMA and Seer results",
       x = "",
       y = "Aptamer Count") +
  theme_bw() +
  scale_y_continuous(limits = c(0,850), breaks = c(100,200,300,400,500,600,700,800,900)) +
  geom_hline(yintercept = 811, color = "purple4") +
  geom_text(aes(2,811,label = "Total Aptamers in Overlap: 811", vjust = -1),color = "purple4") +
  theme(legend.position = "none")
ggsave("quality_overlap_plot.png")
quality_overlap_plot

#FC correlation graph - filtered mouse
corr_sscombo_KMC_plot = sscombo_KMC_corr_df %>%
  ggplot() +
  geom_point(data = sscombo_KMC_corr_df, aes(x = AptName_SeerFeature, y = Soma_FC, colour = "SomaLogic"), size = 1) +
  geom_text_repel(aes(x = AptName_SeerFeature, y = Soma_FC,label = SOMA_Sig_FC_Prot), max.overlaps = 8, size = 2) +
  geom_point(data = sscombo_KMC_corr_df, aes(x = AptName_SeerFeature, y = Seer_FC, colour = "Seer MS"), shape = "triangle", size = 1) +
  geom_text_repel(aes(x = AptName_SeerFeature, y = Seer_FC,label = Seer_Sig_FC_Prot), max.overlaps = 8, size = 2) +
  labs(title = "Fold Change (FC) Magnitude: SomaLogic vs. Seer",
       subtitle = "Mouse KMC-PDAC v. KMC Control",
       x = "Protein IDs (SomaAptamer_SeerFeature)",
       y = "Log2(FC)",
       colour = "Platform") +
  theme_bw() +
  ylim(-3,3) +
  rremove("x.grid") +
  geom_hline(yintercept = 2, colour = "black", linetype = "dashed") +
  geom_hline(yintercept = -2, colour = "black", linetype = "dashed") +
  theme(axis.text.x = element_blank()) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(legend.title = element_text(size = 12)) +
  theme(title = element_text(size = 14))
ggsave("corr_sscombo_KMC_plot.png", width = 7.29, height = 4.51, units = "in")
corr_sscombo_KMC_plot

sscombo_KMC_corr_sig_df = sscombo_KMC_corr_df %>% filter(p_val_sig == "Significant")

#FC correlation graph - filtered mouse, only significant Seer features
corr_sscombo_KMC_sig_plot = sscombo_KMC_corr_sig_df %>%
  ggplot() +
  geom_point(aes(x = AptName_SeerFeature, y = Soma_FC, colour = "SomaLogic"), size = 1) +
  geom_text_repel(aes(x = AptName_SeerFeature, y = Soma_FC,label = SOMA_Sig_FC_Prot), max.overlaps = 10, size = 2) +
  geom_point(aes(x = AptName_SeerFeature, y = Seer_FC, colour = "Seer MS"), shape = "triangle", size = 1) +
  geom_text_repel(aes(x = AptName_SeerFeature, y = Seer_FC,label = Seer_Sig_FC_Prot), max.overlaps = 10, size = 2) +
  labs(title = "SomaLogic vs. Seer Fold Change (FC)",
       subtitle = "KMC-PDAC v. KMC Control, Seer MS-Significant Features",
       x = "Protein IDs (SomaAptamer_SeerFeature)",
       y = "Log2(FC)",
       colour = "Method") +
  theme_bw() +
  ylim(-3,3) +
  geom_hline(yintercept = 2, colour = "black", linetype = "dashed") +
  geom_hline(yintercept = -2, colour = "black", linetype = "dashed") +
  theme(axis.text.x = element_blank()) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(legend.title = element_text(size = 12)) +
  theme(title = element_text(size = 14))
ggsave("corr_sscombo_KMC_sig_plot.png")
corr_sscombo_KMC_sig_plot

#FC correlation graph version 2 - filtered mouse
corr2_sscombo_KMC_plot = sscombo_KMC_corr_df %>%
  ggplot() +
  geom_point(data = sscombo_KMC_corr_df, aes(x = Soma_FC, y = Seer_FC, color = SigFC_Flag), size = 2, alpha = 0.5) +
  geom_text_repel(aes(x = Soma_FC, y = Seer_FC, label = SOMA_Sig_FC_Prot), max.overlaps = 20, size = 2) +
  geom_text_repel(aes(x = Soma_FC, y = Seer_FC, label = Seer_Sig_FC_Prot), max.overlaps = 20, size = 2) +
  scale_color_manual(values = c("black","turquoise3","purple2","green2")) +
  labs(title = "SomaLogic vs. Seer Fold Change (FC)",
       subtitle = "Murine KMC-PDAC v. KMC Control",
       x = "Log2(Soma_FC)",
       y = "Log2(Seer_FC)") +
  theme_bw() +
  ylim(-3,3) +
  xlim(-3,3) +
  geom_vline(xintercept = 2, colour = "purple4", linetype = "dashed", linewidth = 1) +
  geom_vline(xintercept = -2, colour = "purple4", linetype = "dashed", linewidth = 1) +
  geom_hline(yintercept = 2, colour = "turquoise4", linetype = "dashed", linewidth = 1) +
  geom_hline(yintercept = -2, colour = "turquoise4", linetype = "dashed", linewidth = 1) +
  theme(legend.title = element_text(size = 14)) +
  theme(legend.text = element_text(size = 12)) +
  theme(axis.text.x = element_text(vjust = 0.5, size = 12)) +
  theme(axis.text.y = element_text(vjust = 0.5, size = 12)) +
  theme(axis.title = element_text(vjust = 0.5, size = 14)) +
  theme(title = element_text(size = 14))
ggsave("corr2_sscombo_KMC_plot.png", width = 7.29, height = 4.51, units = "in")
corr2_sscombo_KMC_plot

#FC correlation graph verison 3 - filtered mouse quality score colored - uncomment for quality filtering trials
#qual_sscombo_KMC_corr_df = sscombo_KMC_corr_df %>% filter((Quality == "High"))
#Correlation calculation - low pvals only (either SOMA or Seer) - uncomment for quality filtering trials
# cor_qual_sscombo_KMC = cor(qual_sscombo_KMC_corr_df$Soma_FC,qual_sscombo_KMC_corr_df$Seer_FC,method="pearson")
corr3_sscombo_KMC_plot = sscombo_KMC_corr_df %>%
  mutate(`Mouse Aptamer Quality` = factor(Quality, levels = c("Low","Medium","High"))) %>%
  mutate(`Method w/ Sig FC` = factor(SigFC_Flag, levels = c("Neither Method","Seer Only","SOMA Only","Both Methods"))) %>%
  ggplot() +
  geom_point(aes(x = Soma_FC, y = Seer_FC, color = `Method w/ Sig FC`, shape = `Mouse Aptamer Quality`), size = 2, alpha = 0.5) +
  geom_text_repel(aes(x = Soma_FC, y = Seer_FC, label = SOMA_Sig_FC_Prot), max.overlaps = 20, size = 2) +
  geom_text_repel(aes(x = Soma_FC, y = Seer_FC, label = Seer_Sig_FC_Prot), max.overlaps = 20, size = 2) +
  scale_color_manual(values = c("black","turquoise3","purple2","green2"), drop = FALSE) +
  scale_shape_manual(values = c(4,17,16)) +
  labs(title = "SomaLogic vs. Seer Fold Change (FC)",
       subtitle = "KMC-PDAC v. KMC Control",
       x = "Log2(Soma_FC)",
       y = "Log2(Seer_FC)") +
  theme_bw() +
  ylim(-3,3) +
  xlim(-3,3) +
  geom_vline(xintercept = 2, colour = "purple4", linetype = "dashed", linewidth = 1) +
  geom_vline(xintercept = -2, colour = "purple4", linetype = "dashed", linewidth = 1) +
  geom_hline(yintercept = 2, colour = "turquoise4", linetype = "dashed", linewidth = 1) +
  geom_hline(yintercept = -2, colour = "turquoise4", linetype = "dashed", linewidth = 1) +
  #theme(axis.text.x = element_blank()) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(legend.title = element_text(size = 12)) +
  theme(title = element_text(size = 14))
ggsave("corr3_sscombo_KMC_plot.png", width = 7.29, height = 4.51, units = "in")
corr3_sscombo_KMC_plot

#FC correlation graph verison 4 - filtered mouse p-value (before BH) filtered (SOMA and Seer pval<=0.25)
low_pval_sscombo_KMC_corr_df = sscombo_KMC_corr_df %>% filter((wt_pval <= 0.25) & (SOMA_wt_pval <= 0.25))
#Correlation calculation - low pvals only (either SOMA or Seer)
cor_low_pval_sscombo_KMC = cor(low_pval_sscombo_KMC_corr_df$Soma_FC,low_pval_sscombo_KMC_corr_df$Seer_FC,method="pearson")
corr4_sscombo_KMC_plot = low_pval_sscombo_KMC_corr_df %>%
  ggplot() +
  geom_point(data = low_pval_sscombo_KMC_corr_df, aes(x = Soma_FC, y = Seer_FC, color = SigFC_Flag), size = 2, alpha = 0.5) +
  geom_text_repel(aes(x = Soma_FC, y = Seer_FC, label = SOMA_Sig_FC_Prot), max.overlaps = 20, size = 2) +
  geom_text_repel(aes(x = Soma_FC, y = Seer_FC, label = Seer_Sig_FC_Prot), max.overlaps = 20, size = 2) +
  scale_color_manual(values = c("black","turquoise3","purple2","green2")) +
  labs(title = "SomaLogic vs. Seer Fold Change (FC)",
       subtitle = "KMC-PDAC v. KMC Control",
       x = "Log2(Soma_FC)",
       y = "Log2(Seer_FC)") +
  theme_bw() +
  ylim(-3,3) +
  xlim(-3,3) +
  geom_vline(xintercept = 2, colour = "purple4", linetype = "dashed", linewidth = 1) +
  geom_vline(xintercept = -2, colour = "purple4", linetype = "dashed", linewidth = 1) +
  geom_hline(yintercept = 2, colour = "turquoise4", linetype = "dashed", linewidth = 1) +
  geom_hline(yintercept = -2, colour = "turquoise4", linetype = "dashed", linewidth = 1) +
  #theme(axis.text.x = element_blank()) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(legend.title = element_text(size = 12)) +
  theme(title = element_text(size = 14))
ggsave("corr4_sscombo_KMC_plot.png", width = 7.29, height = 4.51, units = "in")
corr4_sscombo_KMC_plot
```
\newline

**Figure 40: Figures summarizing the results of cross-platform analysis for the murine KMC-PDAC and KMC control data.**

### Protein ID Overlap & Fold Change Comparison - Human Case v. All Control

```{r CC_WiT_Human_MethodComparison}
#Grab SOMA aptamer seq, target, targetfullname, intensity, columns from filt_mouse df (not using normalized data for now...)
soma_CChuman_df = WiT_DE1_human_df %>% dplyr::select(-c(count_Case,count_Control,data_Case,data_Control)) %>%
                                    mutate(UniProt = ifelse(grepl("|", UniProt, fixed = TRUE), gsub("[|].*", "", UniProt), UniProt))
colnames(soma_CChuman_df) = c("AptName","SOMA_wt_pval","SOMA_Diff_Median","SOMA_wt_pval_BH","SOMA_Target","SOMA_TargetFullName","UniProt","SOMA_p_val_sig","SOMA_prot_sig")

#Load Seer KMC-KMC_control data
seer_CChuman_df = read_tsv("/Users/sorianos/Code/SomaLogic/Seer_HumanCC_WiT.tsv")
seer_CChuman_df = seer_CChuman_df %>% dplyr::select(-c(count_Case,count_Control))

#Find number of protein IDs (removing NP duplicates) in seer human data - AFTER all filters
seer_CChuman_proteins = seer_CChuman_df %>% dplyr::select(c(`Protein IDs`,`Gene names`)) %>% unique() #%>% nrow()

#Transform dataset into same format as mouse seer file (seer_KMC_df)
seer_CChuman_df = seer_CChuman_df %>% mutate(UniProt = ifelse(grepl(";", `Protein IDs`), sub(";.*", "", `Protein IDs`), `Protein IDs`)) %>%
                                      mutate(Genes = ifelse(grepl(";", `Gene names`), sub(";.*", "", `Gene names`), `Gene names`)) %>%
                                      mutate(Feature = ifelse(grepl(";", Feature), sub(";.*", "", Feature), Feature)) %>%
                                      dplyr::select(-c("Protein IDs","Gene names"))

#Find number of protein IDs (removing NP duplicates) in seer KMC data
sig_seer_CChuman_proteins = seer_CChuman_df %>% filter(p_val_sig == "Significant") %>% dplyr::select(c(Genes,UniProt)) %>% unique() %>% nrow()

#Merge Soma and Seer df for comparison
sscombo_CChuman_df = merge(soma_CChuman_df,seer_CChuman_df, by = "UniProt") #Intersecting UniProt IDs btwn Soma + Seer

#Generate table of protein overlaps (removing NP replicates)
protID_overlap_CChuman = sscombo_CChuman_df %>% dplyr::select(c(UniProt,SOMA_Target,SOMA_TargetFullName,Genes, `Protein names`)) %>% unique()
colnames(protID_overlap_CChuman) = c("UniProt","SOMA_Target","SOMA_TargetFullName","Seer_Target","Seer_TargetFullName")
#Export table of overlap proteins
write_tsv(protID_overlap_CChuman,"protID_overlap_CChuman.tsv")

sscombo_CChuman_sig_df = sscombo_CChuman_df %>% filter(p_val_sig == "Significant")
sigprotID_CChuman_overlap = sscombo_CChuman_sig_df %>% dplyr::select(c(Genes)) %>% unique() %>% nrow()

#Export list of Seer UniProts - used for SynGO
write(seer_CChuman_df$UniProt,"seer_uniprotIDs_KMC_WiT.tsv")
```

```{r CC_WiT_Human_Alt_MethodComparison, include = FALSE}
#Grab SOMA alt aptamer seq, target, targetfullname, intensity, columns from filt_mouse df (not using normalized data for now...)
soma_CChuman_alt_df = WiT_DE1_human_alt_df %>% dplyr::select(-c(count_Case,count_Control,data_Case, data_Control)) %>%
                                    mutate(UniProt = ifelse(grepl("|", UniProt, fixed = TRUE), gsub("[|].*", "", UniProt), UniProt)) 
colnames(soma_CChuman_alt_df) = c("AptName","SOMA_wt_pval","SOMA_Diff_Median","SOMA_wt_pval_BH","SOMA_Target","SOMA_TargetFullName","UniProt","SOMA_p_val_sig","SOMA_prot_sig")

#Merge Soma alt and Seer df for comparison
sscombo_CChuman_alt_df = merge(soma_CChuman_alt_df,seer_CChuman_df, by="UniProt") #Intersecting UniProt IDs btwn Soma + Seer

#Generate table of protein overlaps (removing NP replicates)
protID_overlap_CChuman_alt = sscombo_CChuman_alt_df %>% dplyr::select(c(UniProt,SOMA_Target,SOMA_TargetFullName,Genes, `Protein names`)) %>% unique()
colnames(protID_overlap_CChuman_alt) = c("UniProt","SOMA_Target","SOMA_TargetFullName","Seer_Target","Seer_TargetFullName")

#Export table of overlap proteins
write_tsv(protID_overlap_CChuman_alt,"protID_overlap_CChuman_alt.tsv")

sscombo_CChuman_alt_sig_df = sscombo_CChuman_alt_df %>% filter(p_val_sig == "Significant")
sigprotID_CChuman_alt_overlap = sscombo_CChuman_alt_sig_df %>% dplyr::select(c(Genes)) %>% unique() %>% nrow()

```

```{r CC_WiT_Human_Corr}
#Calculate per row (single protein Soma vs. Seer NP) correlation coefficient (r) for Soma and Seer fold changes
soma_CChuman_FC = sscombo_CChuman_df$SOMA_Diff_Median
seer_CChuman_FC = sscombo_CChuman_df$diff_median
sscombo_CChuman_corr_df = sscombo_CChuman_df %>% rowwise() %>% mutate(data_FC = list(c(SOMA_Diff_Median,Diff))) %>%
                                          mutate(FC_SD = sd(data_FC), FC_CV = 100*sqrt(2^(log(2)*(FC_SD^2))-1)) %>%
                                          mutate(AptName_SOMATarget = paste(AptName, SOMA_Target, sep = "|")) %>%
                                          mutate(AptName_SeerFeature = paste(AptName_SOMATarget, Feature, sep = "_")) %>%
                                          ungroup() %>%
                                          mutate(SOMA_Sig_FC_Prot = ifelse(SOMA_Diff_Median >= 2 | SOMA_Diff_Median <= -2, AptName_SeerFeature, NA)) %>%
                                          mutate(Seer_Sig_FC_Prot = ifelse(Diff >= 2 | Diff <= -2, AptName_SeerFeature, NA)) %>%
                                          mutate(SigFC_Flag = ifelse(!is.na(SOMA_Sig_FC_Prot) & !is.na(Seer_Sig_FC_Prot),
                                                                      "Both Methods",
                                                                      ifelse(!is.na(SOMA_Sig_FC_Prot),
                                                                             "SOMA Only",
                                                                             ifelse(!is.na(Seer_Sig_FC_Prot),
                                                                                    "Seer Only",
                                                                                    "Neither Method"))))
colnames(sscombo_CChuman_corr_df)[colnames(sscombo_CChuman_corr_df) == "SOMA_Diff_Median"] = "Soma_FC"
colnames(sscombo_CChuman_corr_df)[colnames(sscombo_CChuman_corr_df) == "Diff"] = "Seer_FC"

cor_sscombo_CChuman = cor(sscombo_CChuman_corr_df$Soma_FC,sscombo_CChuman_corr_df$Seer_FC,method="pearson")

#FC correlation graph - filtered human
corr_sscombo_CChuman_plot = sscombo_CChuman_corr_df %>%
  ggplot() +
  geom_point(data = sscombo_CChuman_corr_df, aes(x = AptName_SeerFeature, y = Soma_FC, colour = "SomaLogic"), size = 1) +
  geom_text_repel(aes(x = AptName_SeerFeature, y = Soma_FC,label = SOMA_Sig_FC_Prot), max.overlaps = 10, size = 2) +
  geom_point(data = sscombo_CChuman_corr_df, aes(x = AptName_SeerFeature, y = Seer_FC, colour = "Seer MS"), shape = "triangle", size = 1) +
  geom_text_repel(aes(x = AptName_SeerFeature, y = Seer_FC,label = Seer_Sig_FC_Prot), max.overlaps = 10, size = 2) +
  labs(title = "",
       subtitle = "Human Case v. Control",
       x = "Protein IDs (SomaAptamer_SeerFeature)",
       y = "Log2(FC)",
       colour = "Platform") +
  theme_bw() +
  # theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5)) +
  ylim(-3,3) +
  rremove("x.grid") +
  geom_hline(yintercept = 2, colour = "black", linetype = "dashed") +
  geom_hline(yintercept = -2, colour = "black", linetype = "dashed") +
  theme(axis.text.x = element_blank()) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(legend.title = element_text(size = 12)) +
  theme(title = element_text(size = 14))
ggsave("corr_sscombo_CChuman_plot.png", width = 7.29, height = 4.51, units = "in")
corr_sscombo_CChuman_plot

sscombo_CChuman_corr_sig_df = sscombo_CChuman_corr_df %>% filter(p_val_sig == "Significant")

#FC correlation graph - filtered human, only significant Seer features
corr_sscombo_CChuman_sig_plot = sscombo_CChuman_corr_sig_df %>%
  ggplot() +
  geom_point(aes(x = AptName_SeerFeature, y = Soma_FC, colour = "SomaLogic"), size = 1) +
  geom_text_repel(aes(x = AptName_SeerFeature, y = Soma_FC,label = SOMA_Sig_FC_Prot), max.overlaps = 10, size = 2) +
  geom_point(aes(x = AptName_SeerFeature, y = Seer_FC, colour = "Seer MS"), shape = "triangle", size = 1) +
  geom_text_repel(aes(x = AptName_SeerFeature, y = Seer_FC,label = Seer_Sig_FC_Prot), max.overlaps = 10, size = 2) +
  labs(title = "SomaLogic vs. Seer Fold Change (FC)",
       subtitle = "Human Case v. Control, Seer MS-Significant Features",
       x = "Protein IDs (SomaAptamer_SeerFeature)",
       y = "Log2(FC)",
       colour = "Method") +
  theme_bw() +
  # theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size = 5)) +
  ylim(-3,3) +
  #rremove("x.grid") +
  geom_hline(yintercept = 2, colour = "black", linetype = "dashed") +
  geom_hline(yintercept = -2, colour = "black", linetype = "dashed") +
  theme(axis.text.x = element_blank()) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(legend.title = element_text(size = 12)) +
  theme(title = element_text(size = 14))
ggsave("corr_sscombo_CChuman_sig_plot.png")
corr_sscombo_CChuman_sig_plot

#FC correlation graph verison 2 - filtered mouse
corr2_sscombo_CChuman_plot = sscombo_CChuman_corr_df %>%
  mutate(`Method w/ Sig FC` = factor(SigFC_Flag, levels = c("Neither Method","Seer Only","SOMA Only","Both Methods"))) %>%
  ggplot() +
  geom_point(aes(x = Soma_FC, y = Seer_FC, color = `Method w/ Sig FC`), size = 2, alpha = 0.5) +
  geom_text_repel(aes(x = Soma_FC, y = Seer_FC, label = SOMA_Sig_FC_Prot), max.overlaps = 20, size = 2) +
  geom_text_repel(aes(x = Soma_FC, y = Seer_FC, label = Seer_Sig_FC_Prot), max.overlaps = 20, size = 2) +
  scale_color_manual(values = c("black","turquoise3","purple2","green2"), drop = FALSE) +
  labs(title = "",
       subtitle = "Human Case v. Control",
       x = "Log2(Soma_FC)",
       y = "Log2(Seer_FC)") +
  theme_bw() +
  ylim(-3,3) +
  xlim(-3,3) +
  geom_vline(xintercept = 2, colour = "purple4", linetype = "dashed", linewidth = 1) +
  geom_vline(xintercept = -2, colour = "purple4", linetype = "dashed", linewidth = 1) +
  geom_hline(yintercept = 2, colour = "turquoise4", linetype = "dashed", linewidth = 1) +
  geom_hline(yintercept = -2, colour = "turquoise4", linetype = "dashed", linewidth = 1) +
  theme(legend.title = element_text(size = 14)) +
  theme(legend.text = element_text(size = 12)) +
  theme(axis.text.x = element_text(vjust = 0.5, size = 12)) +
  theme(axis.text.y = element_text(vjust = 0.5, size = 12)) +
  theme(axis.title = element_text(vjust = 0.5, size = 14)) +
  theme(title = element_text(size = 14))
ggsave("corr2_sscombo_CChuman_plot.png", width = 7.29, height = 4.51, units = "in")
corr2_sscombo_CChuman_plot
```
\newline

**Figure 41: Figures summarizing the results of cross-platform analysis for the human Case v. Control comparison data.**

### Raw data Method Correlation - Murine

```{r Raw_Data_Feature_Comparison}
#Load Seer raw data - KMC only
Seer_MUS_proteins_KMC_raw_noSF = read_tsv("~/Code/SomaLogic/Seer_MUS_proteins_KMC.tsv") #No sparsity filter
Seer_MUS_proteins_KMC_raw = read_tsv("~/Code/SomaLogic/Seer_MUS_proteins_KMC_50%Sparsity.tsv") #50% sparsity filter
Seer_MUS_proteins_KMC = Seer_MUS_proteins_KMC_raw %>% dplyr::select(c("Feature","Protein.Ids","Protein.Names","PG.MaxLFQ","Log2_LFQ_Intensity","Group_final","SampleName")) %>%
                        #Removes protein group IDs
                        # filter(grepl(";", `Protein.Ids`) == FALSE) %>%
                        # mutate(UniProt = `Protein.Ids`) %>%
                        # mutate(Proteins = `Protein.Names`) %>%
                        #Keeps first ID in protein group IDs
                        mutate(UniProt = ifelse(grepl(";", `Protein.Ids`), sub(";.*", "", `Protein.Ids`), `Protein.Ids`)) %>%
                        mutate(Proteins = ifelse(grepl(";", `Protein.Names`), sub(";.*", "", `Protein.Names`), `Protein.Names`)) %>%
                        mutate(Feature = ifelse(grepl(";", Feature), sub(";.*", "", Feature), Feature)) %>%
                        dplyr::select(-c("Protein.Ids","Protein.Names"))

#Load Soma raw data
Soma_MUS_proteins_KMC = filt_mouse_df %>% filter(SampleType == "Sample") %>%
                                          filter(eLoD_Flag==TRUE) %>%
                                          filter(Group_final == "KMC - PDAC" | Group_final == "KMC_control") %>%
                                          dplyr::select(c("UniProt","AptName","Target","TargetFullName","Intensity","Quality","Group_final","SampleName.y")) %>%
                                          mutate(Log2_Intensity = log(Intensity,2),
                                                 zscore = (Intensity - mean(Intensity))/sd(Intensity))

#Perform nomenclature pivot from Soma human uniprots to corresponding mouse uniprots
###NOTE: Depending on KEGG server status, the following function may result in a timeout error. If so, run again and continue when successful.
Soma_MUS_proteins_UPs = Soma_MUS_proteins_KMC$UniProt %>% unique()
raw_NCBI_KEGG_anno_df = nomConvert_Human_Mouse(Soma_MUS_proteins_UPs)
###
#Merge mouse uniprots and other mouse annotations to Soma data
raw_Soma_KMC_df = merge(Soma_MUS_proteins_KMC,raw_NCBI_KEGG_anno_df, by.x = "UniProt", by.y = "HOM_UniProt", all.x = TRUE) %>% unique()
#Filter out human/soma genes with no mouse UniProt match
raw_Soma_KMC_df = raw_Soma_KMC_df %>% filter(!is.na(MUS_UniProt)) %>%
                                      mutate(UniProts_MUS_HOM_Sample = paste(MUS_UniProt,UniProt,SampleName.y,sep = "_")) #Added SampleName here - SOMA

#Perform nomenclature pivot from Seer human uniprots to corresponding mouse uniprots
###NOTE: Depending on KEGG server status, the following function may result in a timeout error. If so, run again and continue when successful.
Seer_MUS_proteins_UPs = Seer_MUS_proteins_KMC$UniProt %>% unique()
seer_raw_NCBI_KEGG_anno_df = nomConvert_Mouse_Human(Seer_MUS_proteins_UPs)
###
#Merge mouse uniprots and other mouse annotations to Soma data
raw_Seer_KMC_df = merge(Seer_MUS_proteins_KMC,seer_raw_NCBI_KEGG_anno_df, by.x = "UniProt", by.y = "MUS_UniProt", all.x = TRUE) %>% unique()
#Filter out human/soma genes with no mouse UniProt match
raw_Seer_KMC_df = raw_Seer_KMC_df %>% filter(!is.na(HOM_UniProt)) %>%
                                      mutate(UniProts_MUS_HOM_Sample = paste(UniProt,HOM_UniProt,SampleName,sep = "_"))

#Create df for platform overlap analysis and corr plot
raw_sscombo_KMC = raw_Soma_KMC_df %>% #ONLY KEEP OVERLAP PROTEINS IN INTERSECT OF BOTH DIRECTION PIVOTS (767 unique)
                                      filter(UniProts_MUS_HOM_Sample %in% raw_Seer_KMC_df$UniProts_MUS_HOM_Sample) %>%
                                      mutate(UniProts_Sample_Group = paste(UniProts_MUS_HOM_Sample,Group_final,sep = "_")) %>%
                                      mutate(Soma_Log2_Zscore = (Log2_Intensity - mean(Log2_Intensity))/sd(Log2_Intensity))

raw_Seer_intensities = raw_Seer_KMC_df %>% mutate(NP = ifelse(grepl("NP1", Feature), "NP1",
                                                       ifelse(grepl("NP2", Feature), "NP2",
                                                              ifelse(grepl("NP3", Feature), "NP3",
                                                                     ifelse(grepl("NP4", Feature), "NP4", "NP5"))))) %>%
                                            mutate(Seer_Log2_Intensity = log(PG.MaxLFQ,2),
                                                   Seer_Log2_Zscore = (Seer_Log2_Intensity - mean(Seer_Log2_Intensity))/sd(Seer_Log2_Intensity)) %>%
                                            dplyr::select(c("UniProts_MUS_HOM_Sample","NP","PG.MaxLFQ","Seer_Log2_Intensity","Seer_Log2_Zscore","Group_final")) %>%
                                            mutate(UniProts_Sample_Group = paste(UniProts_MUS_HOM_Sample,Group_final,sep = "_"))

raw_sscombo_KMC = left_join(raw_sscombo_KMC,raw_Seer_intensities,by = "UniProts_Sample_Group")

#Raw Data correlation graph - filtered mouse quality score colored - TAKES UP A LOT OF SPACE
# PDAC Samples: "MurPDAC_P4_AJ_5_PL" "MurPDAC_P4_AJ_6_PL" "MurPDAC_P7_AJ_2_PL" "MurPDAC_P3_AJ_4_PL" "MurPDAC_P7_AJ_3_PL" "MurPDAC_P3_AJ_1_PL"
# CTRL Samples: MurPDAC_P7_AK_1_PL,MurPDAC_P7_AM_1_PL,MurPDAC_P3_AL_5_PL,MurPDAC_P6_AL_7_PL,MurPDAC_P4_AL_4_PL,MurPDAC_P2_AL_3_PL,MurPDAC_P5_AM_5_PL,MurPDAC_P2_AM_4_PL,MurPDAC_P5_AL_6_PL,MurPDAC_P5_AM_6_PL,MurPDAC_P3_AL_2_PL,MurPDAC_P8_AL_1_PL

#Create df for plotting - specify NP, aptamer quality, group, and sample
corr_mag_plot_df = raw_sscombo_KMC %>% filter(NP == "NP2") %>%
                                        filter(!Quality == "Low") %>%
                                        filter(Group_final.x == "KMC_control") %>%
                                        filter(SampleName.y == "MurPDAC_P7_AK_1_PL") %>%
                                        mutate(Soma_Log2_Intensity = Log2_Intensity,
                                               Seer_Log2_Intensity = Seer_Log2_Intensity,
                                               Soma_Log2_ZScore_Intensity = Soma_Log2_Zscore,
                                               Seer_Log2_ZScore_Intensity = Seer_Log2_Zscore,
                                               UniProts_MUS_HOM = str_replace(UniProts_MUS_HOM_Sample.x,"_MurPDAC.*",""))

write.table(corr_mag_plot_df,"corr_mag_NP2_MurPDAC_P7_AK_1_mhQ_plot_v6.tsv")

corr_Log2Intensity_KMCCTRL_NP2_mhQ_MurPDAC_P7_AK_1_PL_plot = corr_mag_plot_df %>%
  ggplot(aes(x = Soma_Log2_Intensity, y = Seer_Log2_Intensity)) +
  geom_point(aes(color = UniProts_MUS_HOM_Sample.x), size = 2, alpha = 0.5) +
  geom_smooth(method = lm, linetype = "dashed", color = "darkblue", fill = "lightblue2") +
  geom_text_repel(aes(x = Soma_Log2_Intensity, y = Seer_Log2_Intensity, label = UniProts_MUS_HOM), max.overlaps = 10, size = 2) +
  labs(title = "SomaLogic vs. Seer Log2 Intensity",
       subtitle = "KMC_control,MurPDAC_P7_AK_1_PL,NP2,Q:Med+High",
       x = "Soma Log2 Intensity",
       y = "Seer Log2 Intensity") +
  theme_bw() +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(title = element_text(size = 14)) +
  theme(legend.position = "none")
ggsave("Murine_RawData_Graphs/corr_Log2Intensity_KMCCTRL_NP2_mhQ_MurPDAC_P7_AK_1_PL_plot_v6.png", width = 7.29, height = 4.51, units = "in")
corr_Log2Intensity_KMCCTRL_NP2_mhQ_MurPDAC_P7_AK_1_PL_plot

#Correlation calculation - murine KMC
cor_corr_NP2_mhQ_MurPDAC_P7_AK_1_PL = cor(corr_mag_plot_df$Soma_Log2_Zscore,corr_mag_plot_df$Seer_Log2_Zscore,method="pearson",use = "pairwise.complete.obs")

#Log2 Zscore correlation graph (same as above, but with Log2 Zscores compared)
corr_Log2Zscore_KMCCTRL_NP2_mhQ_MurPDAC_P7_AK_1_PL_plot = corr_mag_plot_df %>%
  ggplot(aes(x = Soma_Log2_ZScore_Intensity, y = Seer_Log2_ZScore_Intensity)) +
  geom_point(aes(color = UniProts_MUS_HOM_Sample.x), size = 2, alpha = 0.5) +
  geom_smooth(method = lm, linetype = "dashed", color = "darkblue", fill = "lightblue2") +
  geom_text_repel(aes(x = Soma_Log2_ZScore_Intensity, y = Seer_Log2_ZScore_Intensity, label = UniProts_MUS_HOM), max.overlaps = 10, size = 2) +
  labs(title = "SomaLogic vs. Seer Log2 Z-Score Intensity",
       subtitle = "KMC Control,MurPDAC_P7_AK_1_PL,NP2,Q:Med+High",
       x = "Soma Log2 Z-Score Intensity",
       y = "Seer Log2 Z-Score Intensity") +
  theme_bw() +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "darkgrey") +
  geom_vline(xintercept = 0, linetype = "dashed", color = "darkgrey") +
  theme(title = element_text(size = 14)) +
  theme(legend.position = "none")
ggsave("Murine_RawData_Graphs/corr_Log2Zscore_KMCCTRL_NP2_mhQ_MurPDAC_P7_AK_1_PL_plot_v6.png", width = 7.29, height = 4.51, units = "in")
corr_Log2Zscore_KMCCTRL_NP2_mhQ_MurPDAC_P7_AK_1_PL_plot

#Raw data correlation graph - filtered mouse - TAKES UP A LOT OF SPACE
mag_Log2Intensity_KMCCTRL_NP2_mhQ_MurPDAC_P7_AK_1_PL_plot = corr_mag_plot_df %>%
  ggplot() +
  geom_point(aes(x = UniProts_MUS_HOM, y = Soma_Log2_Intensity, colour = "SomaLogic"), size = 1) +
  geom_hline(yintercept = mean(corr_mag_plot_df$Soma_Log2_Intensity), linetype = "dashed", color = "turquoise4") +
  #geom_text_repel(aes(x = UniProts_MUS_HOM, y = Soma_Log2_Intensity,label = UniProts_MUS_HOM), max.overlaps = 8, size = 2) +
  geom_point(aes(x = UniProts_MUS_HOM, y = Seer_Log2_Intensity, colour = "Seer MS"), shape = "triangle", size = 1) +
  geom_hline(yintercept = mean(corr_mag_plot_df$Seer_Log2_Intensity), linetype = "dashed", color = "red4") +
  # geom_text_repel(aes(x = UniProts_MUS_HOM, y = Seer_Log2_Intensity,label = UniProts_MUS_HOM), max.overlaps = 8, size = 2) +
  labs(title = "Log2 Intensity Comparison: SomaLogic (RFU) vs. Seer (MaxLFQ)",
       subtitle = "KMC_control,MurPDAC_P7_AK_1_PL,NP2,Q:Med+High",
       x = "Protein IDs (UniProt_MUS_HOM)",
       y = "Intensity",
       colour = "Platform") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90)) + 
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(legend.title = element_text(size = 12)) +
  theme(title = element_text(size = 14))
ggsave("Murine_RawData_Graphs/mag_Log2Intensity_KMCCTRL_NP2_mhQ_MurPDAC_P7_AK_1_PL_plot_v6.png", width = 30, height = 4.51, units = "in")
mag_Log2Intensity_KMCCTRL_NP2_mhQ_MurPDAC_P7_AK_1_PL_plot

#Raw data correlation graph - filtered mouse - TAKES UP A LOT OF SPACE
mag_ZScore_KMCCTRL_NP2_mhQ_MurPDAC_P7_AK_1_PL_plot = corr_mag_plot_df %>%
  ggplot() +
  geom_point(aes(x = UniProts_MUS_HOM, y = Soma_Log2_ZScore_Intensity, colour = "SomaLogic"), size = 1) +
  geom_hline(yintercept = mean(corr_mag_plot_df$Soma_Log2_ZScore_Intensity), linetype = "dashed", color = "turquoise4") +
  #geom_text_repel(aes(x = UniProts_MUS_HOM, y = Soma_Log2_ZScore_Intensity,label = UniProts_MUS_HOM), max.overlaps = 8, size = 2) +
  geom_point(aes(x = UniProts_MUS_HOM, y = Seer_Log2_ZScore_Intensity, colour = "Seer MS"), shape = "triangle", size = 1) +
  geom_hline(yintercept = mean(corr_mag_plot_df$Seer_Log2_ZScore_Intensity), linetype = "dashed", color = "red4") +
  # geom_text_repel(aes(x = UniProts_MUS_HOM, y = Seer_Log2_ZScore_Intensity,label = UniProts_MUS_HOM), max.overlaps = 8, size = 2) +
  labs(title = "Log2 Z-Score Intensity Comparison: SomaLogic vs. Seer",
       subtitle = "KMC Control,MurPDAC_P7_AK_1_PL,NP2,Q:Med+High",
       x = "Protein IDs (UniProt_MUS_HOM)",
       y = "ZScore",
       colour = "Platform") +
  theme_bw() +
  geom_hline(yintercept = 0, linetype = "dashed", color = "darkgrey") +
  #rremove("x.grid") +
  theme(axis.text.x = element_text(angle = 90)) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(legend.title = element_text(size = 12)) +
  theme(title = element_text(size = 14))
ggsave("Murine_RawData_Graphs/mag_ZScore_KMCCTRL_NP2_mhQ_MurPDAC_P7_AK_1_PL_plot_v6.png", width = 30, height = 4.51, units = "in")
mag_ZScore_KMCCTRL_NP2_mhQ_MurPDAC_P7_AK_1_PL_plot
```
\newline

**Figure 42: Figures summarizing the results of cross-platform analysis for the murine KMC-PDAC and KMC control Log2 intensity and Log2 ZScore intensity data.**

```{r ZScore_WiT, include = FALSE}
#Wilcox test - store pvalue and difference
# ZScore_Soma_KMC_df = Soma_MUS_proteins_KMC %>%
#   dplyr::group_by(AptName,Group_final) %>%
#   summarize(data = list(zscore), count = n(), .groups = "drop") %>%
#   pivot_wider(names_from = Group_final, values_from = c(data,count)) %>%
#   filter(`count_KMC - PDAC` > 1 & count_KMC_control > 1)
#   
# ZScore_Soma_KMC_WiT_res = ZScore_Soma_KMC_df %>%
#                           rowwise() %>%
#                           mutate(zscore_wt_pval = wilcox.test(unlist(`data_KMC - PDAC`), unlist(data_KMC_control))$p.value) %>%
#                           mutate(zscore_Diff = (median(unlist(`data_KMC - PDAC`)) - median(unlist(data_KMC_control)))) %>%
#                           ungroup() %>%
#                           mutate(zscore_wt_pval_BH = p.adjust(zscore_wt_pval, method = "BH"))
# ZScore_Soma_KMC_WiT_res = merge(ZScore_Soma_KMC_WiT_res, partial_apt_anno_df, by = "AptName", all.x = TRUE)
# 
# ZScore_Soma_KMC_WiT_res = ZScore_Soma_KMC_WiT_res %>%
#   mutate(p_val_sig = ifelse(zscore_wt_pval_BH < 0.05, "Significant", "Non"),
#          proteins_sig = ifelse(zscore_wt_pval_BH < 0.05, Target, ""),
#   )
```

```{r Plot_WiT_pval_ZScore_WiT, include = FALSE}
#Plot WiT pvalue distribution - before multiple testing correction
# ZScore_Soma_KMC_WiT_res_pval_plot = ZScore_Soma_KMC_WiT_res %>%
#   ggplot(aes(zscore_wt_pval)) +
#   theme_bw() +
#   theme(panel.grid = element_blank()) +
#   geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
#   labs(x = "p-value",
#        y = "",
#        title = "ZScore P-Value Dist",
#        subtitle = "") +
#   theme(plot.subtitle = element_text(face = "italic"))
# #ggsave("WiT_pval_dist_DE3_mouse_alt_plot.png")
# ZScore_Soma_KMC_WiT_res_pval_plot
```

```{r Plot_WiT_vol_ZScore_WiT, include = FALSE}
#Wilcoxon volcano Plot
# ZScore_Soma_KMC_WiT_volcano_plot = ZScore_Soma_KMC_WiT_res %>%
#   ggplot(aes(x = zscore_Diff, y = -log(zscore_wt_pval_BH, 10), group = p_val_sig)) +
#   geom_point(aes(color = p_val_sig, text = Target), size = 2, alpha = 0.5) +
#   geom_hline(yintercept = -log(0.05, 10), linetype = "dashed", color = "red") +
#   geom_vline(xintercept = 2, linetype = "dashed", color = "red") +
#   geom_vline(xintercept = -2, linetype = "dashed", color = "red") +
#   geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
#   scale_color_manual(values = c("black", "red")) +
#   labs(title = "SomaLogic Murine DE: KMC-PDAC v. KMC Control",
#        x = "Log2(KMC-PDAC-KMC_Control)",
#        y = "-Log10(pvalue)") +
#   theme_bw() +
#   xlim(-5,5) +
#   theme(axis.text.x = element_text(vjust = 0.5, size = 14)) +
#   theme(axis.text.y = element_text(vjust = 0.5, size = 14)) +
#   theme(axis.title = element_text(vjust = 0.5, size = 16)) +
#   theme(title = element_text(size = 16)) +
#   theme(legend.position = "none") +
#   theme(plot.subtitle = element_text(face = "italic"))
# ggsave("ZScore_Soma_KMC_WiT_volcano_plot.png")
# ZScore_Soma_KMC_WiT_volcano_plot
# 
# ZScore_Soma_KMC_WiT_volcano_plotly = ggplotly(ZScore_Soma_KMC_WiT_volcano_plot, tooltip = "Target")
# saveWidget(ZScore_Soma_KMC_WiT_volcano_plotly, "ZScore_Soma_KMC_WiT_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

```{r ZScore_Seer_WiT, include = FALSE}
#Wilcox test - store pvalue and difference
# ZScore_Seer_KMC_df = Seer_MUS_proteins_KMC %>%
#                       dplyr::group_by(Feature,Group_final) %>%
#                       summarize(data = list(zscore), count = n(), .groups = "drop") %>%
#                       pivot_wider(names_from = Group_final, values_from = c(data,count)) %>%
#                       filter(`count_KMC - PDAC` > 1 & count_KMC_control > 1)
#   
# ZScore_Seer_KMC_WiT_res = ZScore_Seer_KMC_df %>%
#                           rowwise() %>%
#                           mutate(zscore_wt_pval = wilcox.test(unlist(`data_KMC - PDAC`), unlist(data_KMC_control))$p.value) %>%
#                           mutate(zscore_Diff = (median(unlist(`data_KMC - PDAC`)) - median(unlist(data_KMC_control)))) %>%
#                           ungroup() %>%
#                           mutate(zscore_wt_pval_BH = p.adjust(zscore_wt_pval, method = "BH"))
# 
# ZScore_Seer_KMC_WiT_anno_df = Seer_MUS_proteins_KMC_raw %>% 
#   mutate(Feature = paste(NP, `Protein.Ids`, sep = "|")) %>%
#   dplyr::select(c("Feature", "Protein.Ids", "Protein.Names", "First.Protein.Description")) %>% 
#   distinct(Feature, .keep_all = TRUE)
# 
# ZScore_Seer_KMC_WiT_res = left_join(ZScore_Seer_KMC_WiT_res, ZScore_Seer_KMC_WiT_anno_df, by = "Feature")
# 
# ZScore_Seer_KMC_WiT_res = ZScore_Seer_KMC_WiT_res %>%
#   mutate(p_val_sig = ifelse(zscore_wt_pval_BH < 0.05, "Significant", "Non"),
#          proteins_sig = ifelse(zscore_wt_pval_BH < 0.05, Protein.Ids, ""),
#   )
```

```{r Plot_WiT_pval_ZScore_Seer_WiT, include = FALSE}
#Plot WiT pvalue distribution - before multiple testing correction
# ZScore_Seer_KMC_WiT_res_pval_plot = ZScore_Seer_KMC_WiT_res %>%
#   ggplot(aes(zscore_wt_pval)) +
#   theme_bw() +
#   theme(panel.grid = element_blank()) +
#   geom_histogram(breaks = seq(0,1,0.05), fill = "#22c6c9", color = "black", alpha = 0.5) +
#   labs(x = "p-value",
#        y = "",
#        title = "ZScore P-Value Dist",
#        subtitle = "") +
#   theme(plot.subtitle = element_text(face = "italic"))
# #ggsave("WiT_pval_dist_DE3_mouse_alt_plot.png")
# ZScore_Seer_KMC_WiT_res_pval_plot
```

```{r Plot_WiT_vol_ZScore_Seer_WiT, include = FALSE}
#Wilcoxon volcano Plot
# ZScore_Seer_KMC_WiT_volcano_plot = ZScore_Seer_KMC_WiT_res %>%
#   ggplot(aes(x = zscore_Diff, y = -log(zscore_wt_pval_BH, 10), group = p_val_sig)) +
#   geom_point(aes(color = p_val_sig, text = Protein.Ids), size = 2, alpha = 0.5) +
#   geom_hline(yintercept = -log(0.05, 10), linetype = "dashed", color = "red") +
#   geom_vline(xintercept = 2, linetype = "dashed", color = "red") +
#   geom_vline(xintercept = -2, linetype = "dashed", color = "red") +
#   geom_text_repel(aes(label = proteins_sig), max.overlaps = 20) +
#   scale_color_manual(values = c("black", "red")) +
#   labs(title = "Seer Murine DE: KMC-PDAC v. KMC Control",
#        x = "Log2(KMC-PDAC-KMC_Control)",
#        y = "-Log10(pvalue)") +
#   theme_bw() +
#   xlim(-5,5) +
#   theme(axis.text.x = element_text(vjust = 0.5, size = 14)) +
#   theme(axis.text.y = element_text(vjust = 0.5, size = 14)) +
#   theme(axis.title = element_text(vjust = 0.5, size = 16)) +
#   theme(title = element_text(size = 16)) +
#   theme(legend.position = "none") +
#   theme(plot.subtitle = element_text(face = "italic"))
# ggsave("ZScore_Seer_KMC_WiT_volcano_plot.png")
# ZScore_Seer_KMC_WiT_volcano_plot
# 
# ZScore_Seer_KMC_WiT_volcano_plotly = ggplotly(ZScore_Seer_KMC_WiT_volcano_plot, tooltip = "Target")
# saveWidget(ZScore_Seer_KMC_WiT_volcano_plotly, "ZScore_Seer_KMC_WiT_volcano_plotly.html", selfcontained = F, libdir = "lib")
```

### Raw data Method Correlation - Human

```{r Human_Raw_Data_Feature_Comparison}
#Load Seer raw human data - no sparsity filter
seer_CChuman_noSF_df = read_tsv("/Users/sorianos/Code/SomaLogic/Seer_HumanCC_raw_noSF.tsv")
#Load Seer raw human data - 50% sparsity filter
seer_CChuman_50SF_df = read_tsv("/Users/sorianos/Code/SomaLogic/Seer_HumanCC_raw_50SF.tsv")
#Load annotation files and adjust columns for merging - with above files, the comment-ed anno below is not necessary
# seer_CChuman_anno_df = read_tsv("/Users/sorianos/Code/SomaLogic/Seer_Human_annotation_NPs_separate.txt")
# seer_CChuman_anno_df = seer_CChuman_anno_df %>% dplyr::select(Condition, Experiment)
# colnames(seer_CChuman_anno_df)[colnames(seer_CChuman_anno_df) == "Experiment"] = "Sample_NP"
seer_CChuman_anno2_df = read_csv("/Users/sorianos/Code/SomaLogic/sample_annotation.csv")
seer_CChuman_anno2_df = seer_CChuman_anno2_df %>% dplyr::select(Subject.ID,Screening.Outcome,Sample.Num) %>%
                                                  mutate(Sample.Num = str_replace(Sample.Num,"S",""))
colnames(seer_CChuman_anno2_df)[colnames(seer_CChuman_anno2_df) == "Sample.Num"] = "Sample"
#Transform dataset into same format as mouse seer file - create features, sample, NP columns
seer_CChuman_df = seer_CChuman_50SF_df %>% mutate(Sample = str_replace(str_replace(substring(Sample_NP, 1, 3),"S0",""),"S",""),
                                             NP = ifelse(grepl("NP1", Sample_NP), "NP1",
                                                          ifelse(grepl("NP2", Sample_NP), "NP2",
                                                                 ifelse(grepl("NP3", Sample_NP), "NP3",
                                                                        ifelse(grepl("NP4", Sample_NP), "NP4", "NP5")))))

#Merge in Sample annotation per Sample.Num
seer_HumanCC_final_df = merge(seer_CChuman_df, seer_CChuman_anno2_df, by = "Sample")

#Adjust human df for comparison to Soma results
Seer_HOM_proteins = seer_HumanCC_final_df %>% dplyr::select(c("Feature","Protein.IDs","Protein.names","Gene.names","Log2_LFQ_Intensity","Condition","Sample_NP","Sample","NP","Subject.ID")) %>%
                                              #Remove protein group IDs
                                              # filter(grepl(";", `Protein.IDs`) == FALSE) %>%
                                              # mutate(UniProt = `Protein.IDs`) %>%
                                              # mutate(Proteins = `Protein.names`) %>%
                                              # mutate(Genes = `Gene.names`) %>%
                                              #Keep 1st proteinID in protein group
                                              mutate(UniProt = ifelse(grepl(";", `Protein.IDs`), sub(";.*", "", `Protein.IDs`), `Protein.IDs`)) %>%
                                              mutate(Proteins = ifelse(grepl(";", `Protein.names`), sub(";.*", "", `Protein.names`), `Protein.names`)) %>%
                                              mutate(Feature = ifelse(grepl(";", Feature), sub(";.*", "", Feature), Feature)) %>%
                                              mutate(Genes = ifelse(grepl(";", `Gene.names`), sub(";.*", "", `Gene.names`), `Gene.names`)) %>%
                                              mutate(Seer_Log2_Intensity = Log2_LFQ_Intensity,
                                                     Seer_Log2_Zscore = (Seer_Log2_Intensity - mean(Seer_Log2_Intensity))/sd(Seer_Log2_Intensity),
                                                     UniProt_SampleID = paste(UniProt,Subject.ID,sep = "_")) %>%
                                              dplyr::select(-c("Protein.IDs","Protein.names","Gene.names","Log2_LFQ_Intensity"))

#Load Soma raw data
Soma_HOM_proteins = filt_human_df %>% filter(SampleType == "Sample") %>%
                                      filter(!is.na(Screening.Outcome)) %>%
                                      filter(eLoD_Flag==TRUE) %>%
                                      dplyr::select(c("UniProt","AptName","Target","TargetFullName","Intensity","Screening.Outcome","Sample_ID")) %>%
                                      mutate(Soma_Log2_Intensity = log(Intensity,2),
                                             Soma_Log2_Zscore = (Soma_Log2_Intensity - mean(Soma_Log2_Intensity))/sd(Soma_Log2_Intensity),
                                             UniProt_SampleID = paste(UniProt,Sample_ID,sep = "_"))

#Combine Seer and Soma raw data by UniProt column
raw_sscombo_human = inner_join(Soma_HOM_proteins,Seer_HOM_proteins, by = "UniProt_SampleID")

#Human case/control samples (15 case, 15 control)
human_samples = raw_sscombo_human %>% dplyr::select(c("Sample_ID","Screening.Outcome")) %>% unique()

#Raw Data correlation graph - filtered human - TAKES UP A LOT OF SPACE
# CASE Samples: RDY0016_30, RDY0016_06, RDY0016_12, RDY0016_17, RDY0016_13, RDY0016_07, RDY0016_18, RDY0016_28, RDY0016_05, RDY0016_26, RDY0016_08, RDY0016_02, RDY0016_09, RDY0016_11, RDY0016_04
# CTRL Samples: RDY0016_29, RDY0016_21, RDY0016_16, RDY0016_27, RDY0016_22, RDY0016_24, RDY0016_23, RDY0016_25, RDY0016_03, RDY0016_31, RDY0016_20, RDY0016_10, RDY0016_01, RDY0016_19, RDY0016_14

#Create df for plotting - specify NP, aptamer quality, group, and sample
corr_mag_human_plot_df = raw_sscombo_human %>% filter(NP == "NP2") %>%
                                                filter(Condition == "Case") %>%
                                                filter(Sample_ID == "RDY0016_04") %>%
                                                mutate(Soma_Log2_Intensity = Soma_Log2_Intensity,
                                                       Seer_Log2_Intensity = Seer_Log2_Intensity,
                                                       Soma_Log2_ZScore_Intensity = Soma_Log2_Zscore,
                                                       Seer_Log2_ZScore_Intensity = Seer_Log2_Zscore) %>%
                                                dplyr::select(-UniProt.y) %>%
                                                mutate(Pairwise_Cor = cor(Soma_Log2_ZScore_Intensity,Seer_Log2_ZScore_Intensity, method = "pearson", use = "pairwise"))

colnames(corr_mag_human_plot_df)[colnames(corr_mag_human_plot_df) == "UniProt.x"] = "UniProt"

write.table(corr_mag_human_plot_df,"corr_mag_NP2_RDY0016_04_plot_v6.tsv")

corr_Log2Intensity_HumanCase_NP2_RDY0016_04_plot = corr_mag_human_plot_df %>%
  ggplot(aes(x = Soma_Log2_Intensity, y = Seer_Log2_Intensity)) +
  geom_point(aes(color = UniProt),size = 2, alpha = 0.5) +
  geom_smooth(method = lm, linetype = "dashed", color = "darkblue", fill = "lightblue2") +
  geom_text_repel(aes(x = Soma_Log2_Intensity, y = Seer_Log2_Intensity, label = UniProt), max.overlaps = 10, size = 2) +
  labs(title = "SomaLogic vs. Seer Log2 Intensity",
       subtitle = "Human Case,RDY0016_04,NP2",
       x = "Soma_Log2Intensity",
       y = "Seer_Log2Intensity") +
  theme_bw() +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(title = element_text(size = 14)) +
  theme(legend.position = "none")
ggsave("HumanCC_RawData_Graphs/corr_Log2Intensity_HumanCC_NP2_RDY0016_04_plot_v6.png", width = 7.29, height = 4.51, units = "in")
corr_Log2Intensity_HumanCase_NP2_RDY0016_04_plot

#Correlation calculation - human
cor_corr_NP2_RDY0016_04 = cor(corr_mag_human_plot_df$Soma_Log2_Zscore,corr_mag_human_plot_df$Seer_Log2_Zscore,method="pearson",use = "pairwise.complete.obs")

#Log2 Zscore correlation graph (same as above, but with Log2 Zscores compared)
corr_Log2Zscore_HumanCase_NP2_RDY0016_04_plot = corr_mag_human_plot_df %>%
  ggplot(aes(x = Soma_Log2_ZScore_Intensity, y = Seer_Log2_ZScore_Intensity)) +
  geom_point(aes(color = UniProt), size = 2, alpha = 0.5) +
  geom_smooth(method = lm, linetype = "dashed", color = "darkblue", fill = "lightblue2") +
  geom_text_repel(aes(x = Soma_Log2_ZScore_Intensity, y = Seer_Log2_ZScore_Intensity, label = UniProt), max.overlaps = 10, size = 2) +
  labs(title = "SomaLogic vs. Seer Log2 Z-Score Intensity",
       subtitle = "Human Case,RDY0016_04,NP2",
       x = "Soma Log2 Z-Score Intensity",
       y = "Seer Log2 Z-Score Intensity") +
  theme_bw() +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "darkgrey") +
  geom_vline(xintercept = 0, linetype = "dashed", color = "darkgrey") +
  theme(title = element_text(size = 14)) +
  theme(legend.position = "none")
ggsave("HumanCC_RawData_Graphs/corr_Log2Zscore_HumanCC_NP2_RDY0016_04_plot_v6.png", width = 7.29, height = 4.51, units = "in")
corr_Log2Zscore_HumanCase_NP2_RDY0016_04_plot

#Raw data correlation graph - filtered human - TAKES UP A LOT OF SPACE
mag_Log2Intensity_HumanCase_NP2_RDY0016_04_plot = corr_mag_human_plot_df %>%
  ggplot() +
  geom_point(aes(x = UniProt, y = Soma_Log2_Intensity, colour = "SomaLogic"), size = 1) +
  geom_hline(yintercept = mean(corr_mag_human_plot_df$Soma_Log2_Intensity), linetype = "dashed", color = "turquoise4") +
  #geom_text_repel(aes(x = UniProt, y = Soma_Log2_Intensity,label = UniProt), max.overlaps = 8, size = 2) +
  geom_point(aes(x = UniProt, y = Seer_Log2_Intensity, colour = "Seer MS"), shape = "triangle", size = 1) +
  geom_hline(yintercept = mean(corr_mag_human_plot_df$Seer_Log2_Intensity), linetype = "dashed", color = "red4") +
  #geom_text_repel(aes(x = UniProt, y = Seer_Log2_Intensity,label = UniProt), max.overlaps = 8, size = 2) +
  labs(title = "Log2 Intensity Comparison: SomaLogic (RFU) vs. Seer (MaxLFQ)",
       subtitle = "Human Case,RDY0016_04,NP2",
       x = "Protein IDs (UniProt)",
       y = "Log2_Intensity",
       colour = "Platform") +
  theme_bw() +
  #rremove("x.grid") +
  theme(axis.text.x = element_text(angle = 90)) +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(legend.title = element_text(size = 12)) +
  theme(title = element_text(size = 14))
ggsave("HumanCC_RawData_Graphs/mag_Log2Intensity_HumanCC_NP2_RDY0016_04_plot_v6.png", width = 26, height = 4.51, units = "in")
mag_Log2Intensity_HumanCase_NP2_RDY0016_04_plot

#Raw data correlation graph - filtered mouse - TAKES UP A LOT OF SPACE
mag_Log2ZScore_HumanCase_NP2_RDY0016_04_plot = corr_mag_human_plot_df %>%
  filter(NP == "NP2") %>%
  filter(Screening.Outcome == "Case") %>%
  filter(Sample_ID == "RDY0016_04") %>%
  mutate(Soma_ZScore = Soma_Log2_Zscore,
         Seer_ZScore = Seer_Log2_Zscore) %>%
  ggplot() +
  geom_point(aes(x = UniProt, y = Soma_ZScore, colour = "SomaLogic"), size = 1) +
  geom_hline(yintercept = mean(corr_mag_human_plot_df$Soma_Log2_ZScore_Intensity), linetype = "dashed", color = "turquoise4") +
  #geom_text_repel(aes(x = UniProt, y = Soma_ZScore,label = UniProt), max.overlaps = 8, size = 2) +
  geom_point(aes(x = UniProt, y = Seer_ZScore, colour = "Seer MS"), shape = "triangle", size = 1) +
  geom_hline(yintercept = mean(corr_mag_human_plot_df$Seer_Log2_ZScore_Intensity), linetype = "dashed", color = "red4") +
  #geom_text_repel(aes(x = UniProt, y = Seer_ZScore,label = UniProt), max.overlaps = 8, size = 2) +
  labs(title = "Log2 ZScore Intensity Comparison: SomaLogic vs. Seer",
       subtitle = "Human Case,RDY0016_04,NP2",
       x = "Protein IDs (UniProt)",
       y = "Log2_Zscore",
       colour = "Platform") +
  theme_bw() +
  #rremove("x.grid") +
  theme(axis.text.x = element_text(angle = 90)) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "darkgrey") +
  theme(axis.title = element_text(vjust = 0.5, size = 12)) +
  theme(legend.title = element_text(size = 12)) +
  theme(title = element_text(size = 14))
ggsave("HumanCC_RawData_Graphs/mag_Log2ZScore_HumanCC_NP2_RDY0016_04_plot_v6.png", width = 26, height = 4.51, units = "in")
mag_Log2ZScore_HumanCase_NP2_RDY0016_04_plot
```
\newline

**Figure 43: Figures summarizing the results of cross-platform analysis for the human Case v. Control Log2 intensity and Log2 ZScore intensity data.**

```{r Log2_Zscore_Histograms}
#Create histogram of murine raw data
murine_Log2Intensity_hist_plot = raw_sscombo_KMC %>%
  pivot_longer(names_to = "Platform",
               values_to = "Log2_Intensity",
               cols = ends_with("Log2_Intensity")) %>%
  mutate(Platform = ifelse(Platform == "Log2_Intensity", "SomaLogic", "Seer")) %>%
  ggplot(aes(x = Log2_Intensity, fill = Platform)) +
  geom_histogram(color = "black", alpha = 0.5, binwidth = 0.25) + #breaks = seq(0,1,0.05)
  labs(x = "",
       y = "Log2 Intensity",
       title = "SomaLogic v. Seer: Log2 Intensity Histogram",
       subtitle = "Murine: KMC-PDAC v. KMC Control") +
  theme_bw() +
  theme(plot.subtitle = element_text(face = "italic"))
ggsave("murine_Log2Intensity_hist_plot.png", width = 7.29, height = 4.51, units = "in")
murine_Log2Intensity_hist_plot

#Create histogram of murine zscore data (confirm zscore transformation - mean=0, sd=1)
murine_Log2ZScore_hist_plot = raw_sscombo_KMC %>%
  pivot_longer(names_to = "Platform",
               values_to = "Log2_Zscore",
               cols = ends_with("Log2_Zscore")) %>%
  mutate(Platform = ifelse(Platform == "Soma_Log2_Zscore", "SomaLogic", "Seer")) %>%
  ggplot(aes(x = Log2_Zscore, fill = Platform)) +
  geom_histogram(color = "black", alpha = 0.5, binwidth = 0.1) + #breaks = seq(0,1,0.05)
  labs(x = "",
       y = "Log2 ZScore Intensity",
       title = "SomaLogic v. Seer: ZScore Intensity Histogram",
       subtitle = "Murine: KMC-PDAC v. KMC Control") +
  theme_bw() +
  theme(plot.subtitle = element_text(face = "italic"))
ggsave("murine_Log2ZScore_hist_plot.png", width = 7.29, height = 4.51, units = "in")
murine_Log2ZScore_hist_plot

#Create histogram of human raw data
human_Log2Intensity_hist_plot = raw_sscombo_human %>%
  pivot_longer(names_to = "Platform",
               values_to = "Log2_Intensity",
               cols = ends_with("Log2_Intensity")) %>%
  mutate(Platform = ifelse(Platform == "Soma_Log2_Intensity", "SomaLogic", "Seer")) %>%
  ggplot(aes(x = Log2_Intensity, fill = Platform)) +
  geom_histogram(color = "black", alpha = 0.5, binwidth = 0.25) +
  labs(x = "",
       y = "Log2 Intensity",
       title = "SomaLogic v. Seer: Log2 Intensity Histogram",
       subtitle = "Human: Case v. Control") +
  theme_bw() +
  theme(plot.subtitle = element_text(face = "italic"))
ggsave("human_Log2Intensity_hist_plot.png", width = 7.29, height = 4.51, units = "in")
human_Log2Intensity_hist_plot

#Create histogram of murine zscore data (confirm zscore transformation - mean=0, sd=1)
human_Log2ZScore_hist_plot = raw_sscombo_human %>%
  pivot_longer(names_to = "Platform",
               values_to = "Log2_Zscore",
               cols = ends_with("Log2_Zscore")) %>%
  mutate(Platform = ifelse(Platform == "Soma_Log2_Zscore", "SomaLogic", "Seer")) %>%
  ggplot(aes(x = Log2_Zscore, fill = Platform)) +
  geom_histogram(color = "black", alpha = 0.5, binwidth = 0.1) +
  labs(x = "",
       y = "Log2 ZScore Intensity",
       title = "SomaLogic v. Seer: ZScore Intensity Histogram",
       subtitle = "Human: Case v. Control") +
  theme_bw() +
  theme(plot.subtitle = element_text(face = "italic"))
ggsave("human_Log2ZScore_hist_plot.png", width = 7.29, height = 4.51, units = "in")
human_Log2ZScore_hist_plot

```
\newline

**Figure 44: Histograms showing distribution of SomaLogic and Seer Log2 and Log2 ZScore intensity data for murine KMC-PDAC and KMC control, and human Case and Control.**

\newline

**Table 5: Unique protein counts for each platform (SomaLogic and Seer MS) for the two main studies compared here (murine KMC-PDAC and KMC Control, human Case and Control) and the number of overlapping protein IDs between platforms in each study. For protein groups in Seer IDs, only the first ID is retained to be able to align IDs between Seer and SomaLogic.**

```{r Protein_Numbers_Table}
#Table showing number of proteins identified in both platforms, and overlapping proteins for both species.
#Murine - Individual

num_SOMA_MUS_proteins = Soma_MUS_proteins_KMC %>% filter(!(UniProt == "")) %>%
                                                  group_by(UniProt) %>%
                                                  summarise(Platform_Species = "SomaLogic_MUS")

num_Seer_MUS_proteins = Seer_MUS_proteins_KMC %>% filter(!(UniProt == "")) %>%
                                                  group_by(UniProt) %>%
                                                  summarise(Platform_Species = "Seer_MUS")

#Murine - Overlap

num_Overlap_MUS_proteins = raw_sscombo_KMC %>% mutate(UniProt_MUS_HOM = str_replace(UniProts_MUS_HOM_Sample.x,"_MurPDAC.*","")) %>%
                                                group_by(UniProt_MUS_HOM) %>%
                                                summarise(Platform_Species = "Overlap_MUS")

#Human - Individual

num_SOMA_HOM_proteins = Soma_HOM_proteins %>% filter(!(UniProt == "")) %>%
                                              group_by(UniProt) %>%
                                              summarise(Platform_Species = "SomaLogic_HOM")

num_Seer_HOM_proteins = Seer_HOM_proteins %>% filter(!(UniProt == "")) %>%
                                              group_by(UniProt) %>%
                                              summarise(Platform_Species = "Seer_HOM")

num_Overlap_HOM_proteins = raw_sscombo_human %>% group_by(UniProt.x) %>%
                                                  summarise(Platform_Species = "Overlap_HOM")

combo_numProteins_df = bind_rows(num_SOMA_MUS_proteins,num_Seer_MUS_proteins,num_SOMA_HOM_proteins,num_Seer_HOM_proteins,num_Overlap_MUS_proteins,num_Overlap_HOM_proteins)

table_numProteins_df = combo_numProteins_df %>% group_by(Platform_Species) %>%
                                                summarise(Protein_Count = n()) %>%
                                                mutate(Study = ifelse(grepl("_MUS", Platform_Species),
                                                                      "Murine KMC-PDAC v. KMC Control",
                                                                      "Human Prostate Cancer Case v. Control")) %>%
                                                mutate(Platform = str_replace(Platform_Species,"_.*","")) %>%
                                                dplyr::select(-Platform_Species) %>%
                                                pivot_wider(names_from = Platform, values_from = Protein_Count) %>%
                                                dplyr::select(Study,Seer,SomaLogic,Overlap)

#Add Initial Seer protein count column
#Murine
table_numProteins_df$Seer_Protein_Count_Initial = length(unique(Seer_MUS_proteins_KMC_raw_noSF$Protein.Ids))
#Human
table_numProteins_df$Seer_Protein_Count_Initial[table_numProteins_df$Study == "Human Prostate Cancer Case v. Control"] = length(unique(seer_CChuman_noSF_df$`Protein.IDs`))
#Add post-50% Sparsity Filter Seer protein count column
#Murine
table_numProteins_df$Seer_Protein_Count_50Sparsity = length(unique(Seer_MUS_proteins_KMC_raw$Protein.Ids))
#Human
table_numProteins_df$Seer_Protein_Count_50Sparsity[table_numProteins_df$Study == "Human Prostate Cancer Case v. Control"] = length(unique(seer_CChuman_50SF_df$`Protein.IDs`))
#Add pre-eLoD filter Somalogic column
#Murine
preeLoD_Soma_protein_count_murKMC = filt_mouse_df %>% filter(SampleType == "Sample") %>%
                                                      filter(Group_final == "KMC - PDAC" | Group_final == "KMC_control")
table_numProteins_df$SomaLogic_Protein_Count_preeLoD = length(unique(preeLoD_Soma_protein_count_murKMC$UniProt))
#Human
preeLoD_Soma_protein_count_human = filt_human_df %>% filter(SampleType == "Sample") %>% 
                                                filter(!is.na(Screening.Outcome))
table_numProteins_df$SomaLogic_Protein_Count_preeLoD[table_numProteins_df$Study == "Human Prostate Cancer Case v. Control"] = length(unique(preeLoD_Soma_protein_count_human$UniProt))

#Arrange table df
table_numProteins_final_df = table_numProteins_df %>% dplyr::select(c("Study","Seer_Protein_Count_Initial","Seer_Protein_Count_50Sparsity","Seer","SomaLogic_Protein_Count_preeLoD","SomaLogic","Overlap"))

#Create kable table
knitr::kable(table_numProteins_final_df, col.names = c("Study","Seer Protein Count Initial","Seer Protein Count 50% Sparsity","Seer Protein Count w/ Alt. Group IDs","SomaLogic Protein Count (pre-eLoD)","SomaLogic Protein Count (post-eLoD)","Overlap Protein Count"))

```